{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "### Mutation Frequency Analysis in CLL Genes\nThis notebook analyzes the association between specific gene mutations and IGHV mutational status in CLL patients."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport matplotlib.pyplot as plt\nimport seaborn as sns\n\n# Load the extracted data\ndata = {\n    'Gene': ['TP53', 'ATM', 'SF3B1', 'NOTCH1', 'BIRC3', 'MYD88', 'PLCG2'],\n    'Mutation_Frequency (%)': [15.5, 19.6, 18.6, 13.4, 9.3, 2.1, 1.0],\n    'Associated_IGHV_Status': ['Unmutated', 'Unmutated', 'Both', 'Unmutated', 'Both', 'Both', 'Unmutated']\n}\ndata_df = pd.DataFrame(data)"}, {"cell_type": "markdown", "metadata": {}, "source": "### Visualization of Mutation Frequencies\nUsing seaborn to create a bar plot representing the mutation frequencies of genes associated with different IGHV statuses."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "plt.figure(figsize=(10,6))\nsns.barplot(x='Gene', y='Mutation_Frequency (%)', hue='Associated_IGHV_Status', data=data_df, palette='viridis')\nplt.title('Gene Mutation Frequencies and IGHV Status in CLL Patients')\nplt.xlabel('Genes')\nplt.ylabel('Mutation Frequency (%)')\nplt.legend(title='IGHV Status')\nplt.tight_layout()\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "### Statistical Analysis\nPerforming chi-square test to determine the significance of the association between gene mutations and IGHV status."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "from scipy.stats import chi2_contingency\n\n# Create contingency table for chi-square test\ntable = pd.crosstab(data_df['Gene'], data_df['Associated_IGHV_Status'])\nchi2, p, dof, ex = chi2_contingency(table)\nprint(f'Chi-square Test p-value: {p}')"}, {"cell_type": "markdown", "metadata": {}, "source": "### Interpretation of Results\nA p-value less than 0.05 indicates a significant association between gene mutations and IGHV status."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "if p < 0.05:\n    print('There is a significant association between gene mutations and IGHV status.')\nelse:\n    print('No significant association found between gene mutations and IGHV status.')"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20Analyzes%20mutation%20frequencies%20and%20their%20association%20with%20IGHV%20status%20in%20CLL%20patients%20using%20the%20provided%20dataset.%0A%0AIntegrate%20larger%20datasets%20and%20include%20survival%20analysis%20to%20enhance%20the%20robustness%20of%20the%20association%20between%20gene%20mutations%20and%20IGHV%20status.%0A%0AGene%20mutations%20immunoglobulin%20heavy-chain%20variable%20region%20chromosomal%20alterations%20chronic%20lymphocytic%20leukemia%20India%0A%0A%23%23%23%20Mutation%20Frequency%20Analysis%20in%20CLL%20Genes%0AThis%20notebook%20analyzes%20the%20association%20between%20specific%20gene%20mutations%20and%20IGHV%20mutational%20status%20in%20CLL%20patients.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20matplotlib.pyplot%20as%20plt%0Aimport%20seaborn%20as%20sns%0A%0A%23%20Load%20the%20extracted%20data%0Adata%20%3D%20%7B%0A%20%20%20%20%27Gene%27%3A%20%5B%27TP53%27%2C%20%27ATM%27%2C%20%27SF3B1%27%2C%20%27NOTCH1%27%2C%20%27BIRC3%27%2C%20%27MYD88%27%2C%20%27PLCG2%27%5D%2C%0A%20%20%20%20%27Mutation_Frequency%20%28%25%29%27%3A%20%5B15.5%2C%2019.6%2C%2018.6%2C%2013.4%2C%209.3%2C%202.1%2C%201.0%5D%2C%0A%20%20%20%20%27Associated_IGHV_Status%27%3A%20%5B%27Unmutated%27%2C%20%27Unmutated%27%2C%20%27Both%27%2C%20%27Unmutated%27%2C%20%27Both%27%2C%20%27Both%27%2C%20%27Unmutated%27%5D%0A%7D%0Adata_df%20%3D%20pd.DataFrame%28data%29%0A%0A%23%23%23%20Visualization%20of%20Mutation%20Frequencies%0AUsing%20seaborn%20to%20create%20a%20bar%20plot%20representing%20the%20mutation%20frequencies%20of%20genes%20associated%20with%20different%20IGHV%20statuses.%0A%0Aplt.figure%28figsize%3D%2810%2C6%29%29%0Asns.barplot%28x%3D%27Gene%27%2C%20y%3D%27Mutation_Frequency%20%28%25%29%27%2C%20hue%3D%27Associated_IGHV_Status%27%2C%20data%3Ddata_df%2C%20palette%3D%27viridis%27%29%0Aplt.title%28%27Gene%20Mutation%20Frequencies%20and%20IGHV%20Status%20in%20CLL%20Patients%27%29%0Aplt.xlabel%28%27Genes%27%29%0Aplt.ylabel%28%27Mutation%20Frequency%20%28%25%29%27%29%0Aplt.legend%28title%3D%27IGHV%20Status%27%29%0Aplt.tight_layout%28%29%0Aplt.show%28%29%0A%0A%23%23%23%20Statistical%20Analysis%0APerforming%20chi-square%20test%20to%20determine%20the%20significance%20of%20the%20association%20between%20gene%20mutations%20and%20IGHV%20status.%0A%0Afrom%20scipy.stats%20import%20chi2_contingency%0A%0A%23%20Create%20contingency%20table%20for%20chi-square%20test%0Atable%20%3D%20pd.crosstab%28data_df%5B%27Gene%27%5D%2C%20data_df%5B%27Associated_IGHV_Status%27%5D%29%0Achi2%2C%20p%2C%20dof%2C%20ex%20%3D%20chi2_contingency%28table%29%0Aprint%28f%27Chi-square%20Test%20p-value%3A%20%7Bp%7D%27%29%0A%0A%23%23%23%20Interpretation%20of%20Results%0AA%20p-value%20less%20than%200.05%20indicates%20a%20significant%20association%20between%20gene%20mutations%20and%20IGHV%20status.%0A%0Aif%20p%20%3C%200.05%3A%0A%20%20%20%20print%28%27There%20is%20a%20significant%20association%20between%20gene%20mutations%20and%20IGHV%20status.%27%29%0Aelse%3A%0A%20%20%20%20print%28%27No%20significant%20association%20found%20between%20gene%20mutations%20and%20IGHV%20status.%27%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Association%20of%20Specific%20Gene%20Mutations%20with%20Immunoglobulin%20Heavy-Chain%20Variable%20Region%20and%20Chromosomal%20Alterations%20in%20Chronic%20Lymphocytic%20Leukemia%20Patients%20in%20India)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "Analyzes mutation frequencies and their association with IGHV status in CLL patients using the provided dataset.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-11", "tags": ["gene mutations CLL India", "biology", "bioinformatics"]}}