{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "## RNA-seq Data Analysis for Ipi1 Mutation in *Candida glabrata*\n\nThis notebook performs a differential gene expression analysis to identify genes affected by the Ipi1 R70H mutation."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Import necessary libraries\nimport pandas as pd\nimport matplotlib.pyplot as plt\nimport seaborn as sns\nfrom statsmodels.stats.multitest import multipletests\n\n# Load the RNA-seq data\n# Assuming data is in GEO Series GSE255839\nrna_seq_data = pd.read_csv('https://biologpt.com/?q=GSE255839')\n"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 1: Data Preprocessing\n\nEnsure that the data is clean and properly formatted for analysis."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Check for missing values\nprint(rna_seq_data.isnull().sum())\n\n# Fill or remove missing values as necessary\nrna_seq_data = rna_seq_data.dropna()\n"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 2: Differential Expression Analysis\n\nIdentify genes that are significantly upregulated or downregulated due to the Ipi1 mutation."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Perform differential expression analysis\nfrom statsmodels.formula.api import ols\n\ngenes = rna_seq_data['gene'].unique()\ndifferential_results = []\nfor gene in genes:\n    gene_data = rna_seq_data[rna_seq_data['gene'] == gene]\n    model = ols('expression ~ mutation', data=gene_data).fit()\n    p_value = model.pvalues['mutation']\n    differential_results.append({'gene': gene, 'p_value': p_value})\n\n# Adjust for multiple testing\nresults_df = pd.DataFrame(differential_results)\nresults_df['adj_p_value'] = multipletests(results_df['p_value'], method='fdr_bh')[1]\n\n# Select significantly differentially expressed genes\nsig_genes = results_df[results_df['adj_p_value'] < 0.05]\nprint(f\"Number of significant genes: {len(sig_genes)}\")\n"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 3: Visualization\n\nVisualize the top differentially expressed genes."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Heatmap of top 20 differentially expressed genes\ntop_genes = sig_genes.nsmallest(20, 'adj_p_value')['gene']\nheatmap_data = rna_seq_data[rna_seq_data['gene'].isin(top_genes)].pivot('gene', 'sample', 'expression')\nsns.heatmap(heatmap_data, cmap='viridis')\nplt.title('Top 20 Differentially Expressed Genes')\nplt.show()\n"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 4: Interpretation\n\nInterpret the biological significance of the differentially expressed genes."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Example: Gene Ontology enrichment (using mock data)\n# This section would typically involve using external libraries or APIs\n# to perform GO enrichment based on the list of significant genes.\n# Here, we'll just print the list.\nprint(sig_genes['gene'].tolist())\n"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20Analyzes%20RNA-seq%20data%20to%20identify%20differentially%20expressed%20genes%20associated%20with%20the%20Ipi1%20mutation%20in%20Candida%20glabrata.%0A%0AIntegrate%20GO%20enrichment%20analysis%20and%20incorporate%20interactive%20visualization%20tools%20like%20Plotly%20for%20enhanced%20data%20exploration.%0A%0AIpi1%20mutation%20multidrug%20resistance%20Candida%20glabrata%20review%0A%0A%23%23%20RNA-seq%20Data%20Analysis%20for%20Ipi1%20Mutation%20in%20%2ACandida%20glabrata%2A%0A%0AThis%20notebook%20performs%20a%20differential%20gene%20expression%20analysis%20to%20identify%20genes%20affected%20by%20the%20Ipi1%20R70H%20mutation.%0A%0A%23%20Import%20necessary%20libraries%0Aimport%20pandas%20as%20pd%0Aimport%20matplotlib.pyplot%20as%20plt%0Aimport%20seaborn%20as%20sns%0Afrom%20statsmodels.stats.multitest%20import%20multipletests%0A%0A%23%20Load%20the%20RNA-seq%20data%0A%23%20Assuming%20data%20is%20in%20GEO%20Series%20GSE255839%0Arna_seq_data%20%3D%20pd.read_csv%28%27https%3A%2F%2Fbiologpt.com%2F%3Fq%3DGSE255839%27%29%0A%0A%0A%23%23%23%20Step%201%3A%20Data%20Preprocessing%0A%0AEnsure%20that%20the%20data%20is%20clean%20and%20properly%20formatted%20for%20analysis.%0A%0A%23%20Check%20for%20missing%20values%0Aprint%28rna_seq_data.isnull%28%29.sum%28%29%29%0A%0A%23%20Fill%20or%20remove%20missing%20values%20as%20necessary%0Arna_seq_data%20%3D%20rna_seq_data.dropna%28%29%0A%0A%0A%23%23%23%20Step%202%3A%20Differential%20Expression%20Analysis%0A%0AIdentify%20genes%20that%20are%20significantly%20upregulated%20or%20downregulated%20due%20to%20the%20Ipi1%20mutation.%0A%0A%23%20Perform%20differential%20expression%20analysis%0Afrom%20statsmodels.formula.api%20import%20ols%0A%0Agenes%20%3D%20rna_seq_data%5B%27gene%27%5D.unique%28%29%0Adifferential_results%20%3D%20%5B%5D%0Afor%20gene%20in%20genes%3A%0A%20%20%20%20gene_data%20%3D%20rna_seq_data%5Brna_seq_data%5B%27gene%27%5D%20%3D%3D%20gene%5D%0A%20%20%20%20model%20%3D%20ols%28%27expression%20~%20mutation%27%2C%20data%3Dgene_data%29.fit%28%29%0A%20%20%20%20p_value%20%3D%20model.pvalues%5B%27mutation%27%5D%0A%20%20%20%20differential_results.append%28%7B%27gene%27%3A%20gene%2C%20%27p_value%27%3A%20p_value%7D%29%0A%0A%23%20Adjust%20for%20multiple%20testing%0Aresults_df%20%3D%20pd.DataFrame%28differential_results%29%0Aresults_df%5B%27adj_p_value%27%5D%20%3D%20multipletests%28results_df%5B%27p_value%27%5D%2C%20method%3D%27fdr_bh%27%29%5B1%5D%0A%0A%23%20Select%20significantly%20differentially%20expressed%20genes%0Asig_genes%20%3D%20results_df%5Bresults_df%5B%27adj_p_value%27%5D%20%3C%200.05%5D%0Aprint%28f%22Number%20of%20significant%20genes%3A%20%7Blen%28sig_genes%29%7D%22%29%0A%0A%0A%23%23%23%20Step%203%3A%20Visualization%0A%0AVisualize%20the%20top%20differentially%20expressed%20genes.%0A%0A%23%20Heatmap%20of%20top%2020%20differentially%20expressed%20genes%0Atop_genes%20%3D%20sig_genes.nsmallest%2820%2C%20%27adj_p_value%27%29%5B%27gene%27%5D%0Aheatmap_data%20%3D%20rna_seq_data%5Brna_seq_data%5B%27gene%27%5D.isin%28top_genes%29%5D.pivot%28%27gene%27%2C%20%27sample%27%2C%20%27expression%27%29%0Asns.heatmap%28heatmap_data%2C%20cmap%3D%27viridis%27%29%0Aplt.title%28%27Top%2020%20Differentially%20Expressed%20Genes%27%29%0Aplt.show%28%29%0A%0A%0A%23%23%23%20Step%204%3A%20Interpretation%0A%0AInterpret%20the%20biological%20significance%20of%20the%20differentially%20expressed%20genes.%0A%0A%23%20Example%3A%20Gene%20Ontology%20enrichment%20%28using%20mock%20data%29%0A%23%20This%20section%20would%20typically%20involve%20using%20external%20libraries%20or%20APIs%0A%23%20to%20perform%20GO%20enrichment%20based%20on%20the%20list%20of%20significant%20genes.%0A%23%20Here%2C%20we%27ll%20just%20print%20the%20list.%0Aprint%28sig_genes%5B%27gene%27%5D.tolist%28%29%29%0A%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Mechanisms%20of%20multidrug%20resistance%20caused%20by%20an%20Ipi1%20mutation%20in%20the%20fungal%20pathogen%20Candida%20glabrata)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "Analyzes RNA-seq data to identify differentially expressed genes associated with the Ipi1 mutation in Candida glabrata.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-11", "tags": ["Ipi1 mutation Candida glabrata", "biology", "bioinformatics"]}}