{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "### RNA-Seq Data Analysis of Myometrial deTARs in Mares with Endometrosis\n\nThis notebook processes RNA-Seq data to identify differentially expressed genes and visualize enriched biological pathways related to endometrosis stages."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Import necessary libraries\nimport pandas as pd\nimport seaborn as sns\nimport matplotlib.pyplot as plt\nfrom sklearn.decomposition import PCA\nimport plotly.express as px\nfrom bioinfokit.analys import gprofiler\n"}, {"cell_type": "markdown", "metadata": {}, "source": "### Load and Prepare Data\n\nLoad the RNA-Seq data and prepare it for analysis."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Load differentially expressed transcriptionally active regions (deTARs)\ndeTARs_IIA_I = pd.read_csv('deTARs_IIA_vs_I.csv')\ndeTARs_IIB_I = pd.read_csv('deTARs_IIB_vs_I.csv')\ndeTARs_III_I = pd.read_csv('deTARs_III_vs_I.csv')\n"}, {"cell_type": "markdown", "metadata": {}, "source": "### Identify Common deTARs Across All Stages\n\nFind the intersection of deTARs across all comparisons to identify common regulatory changes."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Identify common deTARs\ncommon_deTARs = set(deTARs_IIA_I['gene']).intersection(set(deTARs_IIB_I['gene']), set(deTARs_III_I['gene']))\ncommon_deTARs_df = pd.DataFrame(list(common_deTARs), columns=['gene'])\ncommon_deTARs_df.head()\n"}, {"cell_type": "markdown", "metadata": {}, "source": "### Pathway Enrichment Analysis\n\nPerform pathway enrichment to understand the biological processes impacted by the common deTARs."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Perform pathway enrichment using g:Profiler\ngp = gprofiler.GProfiler(return_dataframe=True)\nenriched_paths = gp.profile(organism='hsapiens', query=common_deTARs, sources=['GO:BP', 'KEGG'])\nenriched_paths.head()\n"}, {"cell_type": "markdown", "metadata": {}, "source": "### Visualize Enriched Pathways\n\nCreate a bar plot to visualize the top enriched pathways."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Plot top enriched pathways\ntop_paths = enriched_paths.head(10)\nplt.figure(figsize=(10,6))\nsns.barplot(x='source', y='name', data=top_paths, palette='viridis')\nplt.title('Top Enriched Pathways in Common deTARs')\nplt.xlabel('Source')\nplt.ylabel('Pathway')\nplt.tight_layout()\nplt.show()\n"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20Analyzes%20differentially%20expressed%20genes%20and%20visualizes%20enriched%20pathways%20using%20RNA-seq%20data%20from%20the%20study.%0A%0AIncorporate%20interactive%20visualizations%20and%20integrate%20additional%20datasets%20for%20a%20more%20comprehensive%20pathway%20analysis.%0A%0AMyometrial%20transcriptome%20endometrosis%20mares%20review%0A%0A%23%23%23%20RNA-Seq%20Data%20Analysis%20of%20Myometrial%20deTARs%20in%20Mares%20with%20Endometrosis%0A%0AThis%20notebook%20processes%20RNA-Seq%20data%20to%20identify%20differentially%20expressed%20genes%20and%20visualize%20enriched%20biological%20pathways%20related%20to%20endometrosis%20stages.%0A%0A%23%20Import%20necessary%20libraries%0Aimport%20pandas%20as%20pd%0Aimport%20seaborn%20as%20sns%0Aimport%20matplotlib.pyplot%20as%20plt%0Afrom%20sklearn.decomposition%20import%20PCA%0Aimport%20plotly.express%20as%20px%0Afrom%20bioinfokit.analys%20import%20gprofiler%0A%0A%0A%23%23%23%20Load%20and%20Prepare%20Data%0A%0ALoad%20the%20RNA-Seq%20data%20and%20prepare%20it%20for%20analysis.%0A%0A%23%20Load%20differentially%20expressed%20transcriptionally%20active%20regions%20%28deTARs%29%0AdeTARs_IIA_I%20%3D%20pd.read_csv%28%27deTARs_IIA_vs_I.csv%27%29%0AdeTARs_IIB_I%20%3D%20pd.read_csv%28%27deTARs_IIB_vs_I.csv%27%29%0AdeTARs_III_I%20%3D%20pd.read_csv%28%27deTARs_III_vs_I.csv%27%29%0A%0A%0A%23%23%23%20Identify%20Common%20deTARs%20Across%20All%20Stages%0A%0AFind%20the%20intersection%20of%20deTARs%20across%20all%20comparisons%20to%20identify%20common%20regulatory%20changes.%0A%0A%23%20Identify%20common%20deTARs%0Acommon_deTARs%20%3D%20set%28deTARs_IIA_I%5B%27gene%27%5D%29.intersection%28set%28deTARs_IIB_I%5B%27gene%27%5D%29%2C%20set%28deTARs_III_I%5B%27gene%27%5D%29%29%0Acommon_deTARs_df%20%3D%20pd.DataFrame%28list%28common_deTARs%29%2C%20columns%3D%5B%27gene%27%5D%29%0Acommon_deTARs_df.head%28%29%0A%0A%0A%23%23%23%20Pathway%20Enrichment%20Analysis%0A%0APerform%20pathway%20enrichment%20to%20understand%20the%20biological%20processes%20impacted%20by%20the%20common%20deTARs.%0A%0A%23%20Perform%20pathway%20enrichment%20using%20g%3AProfiler%0Agp%20%3D%20gprofiler.GProfiler%28return_dataframe%3DTrue%29%0Aenriched_paths%20%3D%20gp.profile%28organism%3D%27hsapiens%27%2C%20query%3Dcommon_deTARs%2C%20sources%3D%5B%27GO%3ABP%27%2C%20%27KEGG%27%5D%29%0Aenriched_paths.head%28%29%0A%0A%0A%23%23%23%20Visualize%20Enriched%20Pathways%0A%0ACreate%20a%20bar%20plot%20to%20visualize%20the%20top%20enriched%20pathways.%0A%0A%23%20Plot%20top%20enriched%20pathways%0Atop_paths%20%3D%20enriched_paths.head%2810%29%0Aplt.figure%28figsize%3D%2810%2C6%29%29%0Asns.barplot%28x%3D%27source%27%2C%20y%3D%27name%27%2C%20data%3Dtop_paths%2C%20palette%3D%27viridis%27%29%0Aplt.title%28%27Top%20Enriched%20Pathways%20in%20Common%20deTARs%27%29%0Aplt.xlabel%28%27Source%27%29%0Aplt.ylabel%28%27Pathway%27%29%0Aplt.tight_layout%28%29%0Aplt.show%28%29%0A%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20The%20myometrial%20transcriptome%20changes%20in%20mares%20with%20endometrosis)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "Analyzes differentially expressed genes and visualizes enriched pathways using RNA-seq data from the study.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-01-29", "tags": ["myometrial transcriptome mares", "biology", "bioinformatics"]}}