{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "### Genetic Diversity Analysis of Genebank Collections\nThis notebook analyzes SNP data to assess genetic diversity and identify redundancies within plant germplasm collections."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\nimport matplotlib.pyplot as plt\nimport seaborn as sns\nfrom sklearn.decomposition import PCA\nfrom sklearn.cluster import KMeans\n\n# Load SNP data\nsnp_data = pd.read_csv('genebank_snp_data.csv')\n"}, {"cell_type": "markdown", "metadata": {}, "source": "#### Data Preprocessing\nHandling missing values and normalizing the SNP data."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "snp_data.fillna(snp_data.mean(), inplace=True)\nfrom sklearn.preprocessing import StandardScaler\nscaler = StandardScaler()\nscaled_data = scaler.fit_transform(snp_data.iloc[:,1:])\n"}, {"cell_type": "markdown", "metadata": {}, "source": "#### Principal Component Analysis (PCA)\nReducing dimensionality to visualize genetic diversity."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "pca = PCA(n_components=2)\npc_result = pca.fit_transform(scaled_data)\nplt.figure(figsize=(10,7))\nsns.scatterplot(x=pca_result[:,0], y=pca_result[:,1])\nplt.title('PCA of Genebank SNP Data')\nplt.xlabel('PC1')\nplt.ylabel('PC2')\nplt.show()\n"}, {"cell_type": "markdown", "metadata": {}, "source": "#### Clustering Analysis\nIdentifying genetic clusters to detect redundancies."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "kmeans = KMeans(n_clusters=5)\nclusters = kmeans.fit_predict(scaled_data)\nplt.figure(figsize=(10,7))\nsns.scatterplot(x=pca_result[:,0], y=pca_result[:,1], hue=clusters, palette='Set2')\nplt.title('KMeans Clustering of Genebank Accessions')\nplt.xlabel('PC1')\nplt.ylabel('PC2')\nplt.show()\n"}, {"cell_type": "markdown", "metadata": {}, "source": "#### Identifying Redundant Accessions\nAccessions within the same cluster may be genetically redundant."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Add cluster labels to data\nsnp_data['Cluster'] = clusters\n\n# Identify potential redundancies\nredundant_accessions = snp_data[snp_data.duplicated(['Cluster'], keep=False)]\nredundant_accessions\n"}, {"cell_type": "markdown", "metadata": {}, "source": "#### Conclusion\nThis analysis highlights genetic clusters within the genebank collection, identifying potential redundancies for streamlining germplasm management."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Save the list of redundant accessions\nredundant_accessions.to_csv('redundant_accessions.csv', index=False)"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20Analyze%20genetic%20diversity%20and%20identify%20redundancies%20in%20genebank%20collections%20using%20SNP%20data.%0A%0AIncorporate%20more%20sophisticated%20clustering%20algorithms%20and%20cross-validate%20with%20phenotypic%20data%20to%20enhance%20redundancy%20detection%20accuracy.%0A%0AGenotyping%20strategies%20for%20genebank%20management%0A%0A%23%23%23%20Genetic%20Diversity%20Analysis%20of%20Genebank%20Collections%0AThis%20notebook%20analyzes%20SNP%20data%20to%20assess%20genetic%20diversity%20and%20identify%20redundancies%20within%20plant%20germplasm%20collections.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0Aimport%20matplotlib.pyplot%20as%20plt%0Aimport%20seaborn%20as%20sns%0Afrom%20sklearn.decomposition%20import%20PCA%0Afrom%20sklearn.cluster%20import%20KMeans%0A%0A%23%20Load%20SNP%20data%0Asnp_data%20%3D%20pd.read_csv%28%27genebank_snp_data.csv%27%29%0A%0A%0A%23%23%23%23%20Data%20Preprocessing%0AHandling%20missing%20values%20and%20normalizing%20the%20SNP%20data.%0A%0Asnp_data.fillna%28snp_data.mean%28%29%2C%20inplace%3DTrue%29%0Afrom%20sklearn.preprocessing%20import%20StandardScaler%0Ascaler%20%3D%20StandardScaler%28%29%0Ascaled_data%20%3D%20scaler.fit_transform%28snp_data.iloc%5B%3A%2C1%3A%5D%29%0A%0A%0A%23%23%23%23%20Principal%20Component%20Analysis%20%28PCA%29%0AReducing%20dimensionality%20to%20visualize%20genetic%20diversity.%0A%0Apca%20%3D%20PCA%28n_components%3D2%29%0Apc_result%20%3D%20pca.fit_transform%28scaled_data%29%0Aplt.figure%28figsize%3D%2810%2C7%29%29%0Asns.scatterplot%28x%3Dpca_result%5B%3A%2C0%5D%2C%20y%3Dpca_result%5B%3A%2C1%5D%29%0Aplt.title%28%27PCA%20of%20Genebank%20SNP%20Data%27%29%0Aplt.xlabel%28%27PC1%27%29%0Aplt.ylabel%28%27PC2%27%29%0Aplt.show%28%29%0A%0A%0A%23%23%23%23%20Clustering%20Analysis%0AIdentifying%20genetic%20clusters%20to%20detect%20redundancies.%0A%0Akmeans%20%3D%20KMeans%28n_clusters%3D5%29%0Aclusters%20%3D%20kmeans.fit_predict%28scaled_data%29%0Aplt.figure%28figsize%3D%2810%2C7%29%29%0Asns.scatterplot%28x%3Dpca_result%5B%3A%2C0%5D%2C%20y%3Dpca_result%5B%3A%2C1%5D%2C%20hue%3Dclusters%2C%20palette%3D%27Set2%27%29%0Aplt.title%28%27KMeans%20Clustering%20of%20Genebank%20Accessions%27%29%0Aplt.xlabel%28%27PC1%27%29%0Aplt.ylabel%28%27PC2%27%29%0Aplt.show%28%29%0A%0A%0A%23%23%23%23%20Identifying%20Redundant%20Accessions%0AAccessions%20within%20the%20same%20cluster%20may%20be%20genetically%20redundant.%0A%0A%23%20Add%20cluster%20labels%20to%20data%0Asnp_data%5B%27Cluster%27%5D%20%3D%20clusters%0A%0A%23%20Identify%20potential%20redundancies%0Aredundant_accessions%20%3D%20snp_data%5Bsnp_data.duplicated%28%5B%27Cluster%27%5D%2C%20keep%3DFalse%29%5D%0Aredundant_accessions%0A%0A%0A%23%23%23%23%20Conclusion%0AThis%20analysis%20highlights%20genetic%20clusters%20within%20the%20genebank%20collection%2C%20identifying%20potential%20redundancies%20for%20streamlining%20germplasm%20management.%0A%0A%23%20Save%20the%20list%20of%20redundant%20accessions%0Aredundant_accessions.to_csv%28%27redundant_accessions.csv%27%2C%20index%3DFalse%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Genotyping%20Genebank%20Collections%3A%20Strategic%20Approaches%20and%20Considerations%20for%20Optimal%20Collection%20Management)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "Analyze genetic diversity and identify redundancies in genebank collections using SNP data.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-11", "tags": ["genotyping genebank strategies", "biology", "bioinformatics"]}}