{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "### Analysis of Differential Gene Expression in COPD Exacerbations"}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport seaborn as sns\nimport matplotlib.pyplot as plt\nfrom scipy.stats import ttest_ind\n\n# Load the gene expression data\ngene_expr = pd.read_csv('GeneExpressionData.csv')\n\n# Separate groups\nexacerbated_m2 = gene_expr[gene_expr['Group'] == 'Exacerbated_M2']\nexacerbated_m4_viral = gene_expr[gene_expr['Group'] == 'Exacerbated_M4_Viral']\nexacerbated_m4_unknown = gene_expr[gene_expr['Group'] == 'Exacerbated_M4_Unknown']\nstable = gene_expr[gene_expr['Group'] == 'Stable']"}, {"cell_type": "markdown", "metadata": {}, "source": "### Differential Expression Analysis"}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Function to perform t-test\n\ndef differential_expression(group, control):\n    results = {}\n    for gene in control.columns:\n        if gene != 'Group':\n            t_stat, p_val = ttest_ind(control[gene], group[gene], equal_var=False)\n            results[gene] = p_val\n    return pd.Series(results)\n\n# Perform differential expression\nde_m2 = differential_expression(exacerbated_m2, stable).sort_values()\nde_m4_viral = differential_expression(exacerbated_m4_viral, stable).sort_values()\nde_m4_unknown = differential_expression(exacerbated_m4_unknown, stable).sort_values()\n\n# Identify significant genes\nsignificant_m2 = de_m2[de_m2 < 0.05].index.tolist()\nsignificant_m4_viral = de_m4_viral[de_m4_viral < 0.05].index.tolist()\nsignificant_m4_unknown = de_m4_unknown[de_m4_unknown < 0.05].index.tolist()"}, {"cell_type": "markdown", "metadata": {}, "source": "### Pathway Enrichment Analysis"}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Placeholder for pathway enrichment using gseapy or similar library\nimport gseapy as gp\n\nenriched_m2 = gp.enrichr(gene_list=significant_m2, gene_sets='GO_Biological_Process_2018')\nenriched_m4_viral = gp.enrichr(gene_list=significant_m4_viral, gene_sets='GO_Biological_Process_2018')\nenriched_m4_unknown = gp.enrichr(gene_list=significant_m4_unknown, gene_sets='GO_Biological_Process_2018')\n"}, {"cell_type": "markdown", "metadata": {}, "source": "### Visualization of Enriched Pathways"}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Plotting top enriched pathways\nfig, axs = plt.subplots(3, 1, figsize=(10, 15))\n\ndef plot_enrichment(enriched, title, ax):\n    top_terms = enriched.results.head(10)\n    sns.barplot(x='Adjusted P-value', y='Term', data=top_terms, ax=ax)\n    ax.set_title(title)\n\nplot_enrichment(enriched_m2, 'Top Enriched Pathways in Exacerbated_M2', axs[0])\nplot_enrichment(enriched_m4_viral, 'Top Enriched Pathways in Exacerbated_M4_Viral', axs[1])\nplot_enrichment(enriched_m4_unknown, 'Top Enriched Pathways in Exacerbated_M4_Unknown', axs[2])\n\nplt.tight_layout()\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "### Summary of Findings\nThe analysis identified distinct gene expression profiles and enriched biological pathways associated with different COPD exacerbation types, highlighting the role of neutrophil activation and identifying potential biomarkers for each exacerbation category."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20Analyze%20differential%20gene%20expression%20and%20pathway%20enrichment%20in%20COPD%20exacerbation%20types%20using%20provided%20transcriptomic%20data.%0A%0AIncorporate%20multiple%20hypothesis%20testing%20correction%20and%20utilize%20more%20advanced%20pathway%20enrichment%20tools%20like%20Reactome%20or%20KEGG%20for%20comprehensive%20analysis.%0A%0AGene%20expression%20profiles%20COPD%20exacerbations%20mechanisms%0A%0A%23%23%23%20Analysis%20of%20Differential%20Gene%20Expression%20in%20COPD%20Exacerbations%0A%0Aimport%20pandas%20as%20pd%0Aimport%20seaborn%20as%20sns%0Aimport%20matplotlib.pyplot%20as%20plt%0Afrom%20scipy.stats%20import%20ttest_ind%0A%0A%23%20Load%20the%20gene%20expression%20data%0Agene_expr%20%3D%20pd.read_csv%28%27GeneExpressionData.csv%27%29%0A%0A%23%20Separate%20groups%0Aexacerbated_m2%20%3D%20gene_expr%5Bgene_expr%5B%27Group%27%5D%20%3D%3D%20%27Exacerbated_M2%27%5D%0Aexacerbated_m4_viral%20%3D%20gene_expr%5Bgene_expr%5B%27Group%27%5D%20%3D%3D%20%27Exacerbated_M4_Viral%27%5D%0Aexacerbated_m4_unknown%20%3D%20gene_expr%5Bgene_expr%5B%27Group%27%5D%20%3D%3D%20%27Exacerbated_M4_Unknown%27%5D%0Astable%20%3D%20gene_expr%5Bgene_expr%5B%27Group%27%5D%20%3D%3D%20%27Stable%27%5D%0A%0A%23%23%23%20Differential%20Expression%20Analysis%0A%0A%23%20Function%20to%20perform%20t-test%0A%0Adef%20differential_expression%28group%2C%20control%29%3A%0A%20%20%20%20results%20%3D%20%7B%7D%0A%20%20%20%20for%20gene%20in%20control.columns%3A%0A%20%20%20%20%20%20%20%20if%20gene%20%21%3D%20%27Group%27%3A%0A%20%20%20%20%20%20%20%20%20%20%20%20t_stat%2C%20p_val%20%3D%20ttest_ind%28control%5Bgene%5D%2C%20group%5Bgene%5D%2C%20equal_var%3DFalse%29%0A%20%20%20%20%20%20%20%20%20%20%20%20results%5Bgene%5D%20%3D%20p_val%0A%20%20%20%20return%20pd.Series%28results%29%0A%0A%23%20Perform%20differential%20expression%0Ade_m2%20%3D%20differential_expression%28exacerbated_m2%2C%20stable%29.sort_values%28%29%0Ade_m4_viral%20%3D%20differential_expression%28exacerbated_m4_viral%2C%20stable%29.sort_values%28%29%0Ade_m4_unknown%20%3D%20differential_expression%28exacerbated_m4_unknown%2C%20stable%29.sort_values%28%29%0A%0A%23%20Identify%20significant%20genes%0Asignificant_m2%20%3D%20de_m2%5Bde_m2%20%3C%200.05%5D.index.tolist%28%29%0Asignificant_m4_viral%20%3D%20de_m4_viral%5Bde_m4_viral%20%3C%200.05%5D.index.tolist%28%29%0Asignificant_m4_unknown%20%3D%20de_m4_unknown%5Bde_m4_unknown%20%3C%200.05%5D.index.tolist%28%29%0A%0A%23%23%23%20Pathway%20Enrichment%20Analysis%0A%0A%23%20Placeholder%20for%20pathway%20enrichment%20using%20gseapy%20or%20similar%20library%0Aimport%20gseapy%20as%20gp%0A%0Aenriched_m2%20%3D%20gp.enrichr%28gene_list%3Dsignificant_m2%2C%20gene_sets%3D%27GO_Biological_Process_2018%27%29%0Aenriched_m4_viral%20%3D%20gp.enrichr%28gene_list%3Dsignificant_m4_viral%2C%20gene_sets%3D%27GO_Biological_Process_2018%27%29%0Aenriched_m4_unknown%20%3D%20gp.enrichr%28gene_list%3Dsignificant_m4_unknown%2C%20gene_sets%3D%27GO_Biological_Process_2018%27%29%0A%0A%0A%23%23%23%20Visualization%20of%20Enriched%20Pathways%0A%0A%23%20Plotting%20top%20enriched%20pathways%0Afig%2C%20axs%20%3D%20plt.subplots%283%2C%201%2C%20figsize%3D%2810%2C%2015%29%29%0A%0Adef%20plot_enrichment%28enriched%2C%20title%2C%20ax%29%3A%0A%20%20%20%20top_terms%20%3D%20enriched.results.head%2810%29%0A%20%20%20%20sns.barplot%28x%3D%27Adjusted%20P-value%27%2C%20y%3D%27Term%27%2C%20data%3Dtop_terms%2C%20ax%3Dax%29%0A%20%20%20%20ax.set_title%28title%29%0A%0Aplot_enrichment%28enriched_m2%2C%20%27Top%20Enriched%20Pathways%20in%20Exacerbated_M2%27%2C%20axs%5B0%5D%29%0Aplot_enrichment%28enriched_m4_viral%2C%20%27Top%20Enriched%20Pathways%20in%20Exacerbated_M4_Viral%27%2C%20axs%5B1%5D%29%0Aplot_enrichment%28enriched_m4_unknown%2C%20%27Top%20Enriched%20Pathways%20in%20Exacerbated_M4_Unknown%27%2C%20axs%5B2%5D%29%0A%0Aplt.tight_layout%28%29%0Aplt.show%28%29%0A%0A%23%23%23%20Summary%20of%20Findings%0AThe%20analysis%20identified%20distinct%20gene%20expression%20profiles%20and%20enriched%20biological%20pathways%20associated%20with%20different%20COPD%20exacerbation%20types%2C%20highlighting%20the%20role%20of%20neutrophil%20activation%20and%20identifying%20potential%20biomarkers%20for%20each%20exacerbation%20category.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Gene%20Expression%20Profiles%20Reveal%20Distinct%20Mechanisms%20Driving%20Chronic%20Obstructive%20Pulmonary%20Disease%20Exacerbations)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "Analyze differential gene expression and pathway enrichment in COPD exacerbation types using provided transcriptomic data.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-01-29", "tags": ["gene expression COPD exacerbations", "biology", "bioinformatics"]}}