{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "This section describes steps to download RNA-seq data from BioProject PRJNA1124505, process the high-depth reads, and perform alternative splicing analysis."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import subprocess\n\n# Download dataset using fastq-dump\nsubprocess.run(['fastq-dump', '--split-files', 'SRRXXXXXX'])\n\n# Run Salmon for transcript quantification\nsubprocess.run(['salmon', 'quant', '-i', 'transcript_index', '-l', 'A', '-r', 'reads.fq', '-o', 'quant_output'])"}, {"cell_type": "markdown", "metadata": {}, "source": "Next, the notebook demonstrates differential splicing analysis using the 3D RNA-seq toolkit and visualization of isoform usage differences between WT and mutant samples."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport matplotlib.pyplot as plt\n\n# Load processed data\ndf = pd.read_csv('differential_splicing_results.csv')\n\n# Plot isoform usage differences\nplt.bar(df['Isoform'], df['Usage'], color=['#6A0C76', '#B565A7'])\nplt.xlabel('DGK5 Isoforms')\nplt.ylabel('Relative Usage (%)')\nplt.title('Isoform Usage in WT vs cpl3 Mutant')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "The code block provides a basic pipeline from data acquisition to visualization, suitable for extension to more complex bioinformatics workflows."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Further statistical analysis and refinement of splicing events can be added based on the initial results obtained."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20Code%20downloads%20Arabidopsis%20RNA-seq%20datasets%2C%20performs%20differential%20splicing%20analysis%20using%203D%20RNA-seq%20pipelines%2C%20and%20visualizes%20splicing%20changes.%0A%0AIncorporate%20error%20handling%2C%20integrate%20with%20current%20RNA-seq%20pipeline%20standards%2C%20and%20add%20more%20descriptive%20comments%20for%20reproducibility.%0A%0ACPL3%20alternative%20splicing%20DGK5%20plant%20immunity%20phosphatidic%20acid%0A%0AThis%20section%20describes%20steps%20to%20download%20RNA-seq%20data%20from%20BioProject%20PRJNA1124505%2C%20process%20the%20high-depth%20reads%2C%20and%20perform%20alternative%20splicing%20analysis.%0A%0Aimport%20subprocess%0A%0A%23%20Download%20dataset%20using%20fastq-dump%0Asubprocess.run%28%5B%27fastq-dump%27%2C%20%27--split-files%27%2C%20%27SRRXXXXXX%27%5D%29%0A%0A%23%20Run%20Salmon%20for%20transcript%20quantification%0Asubprocess.run%28%5B%27salmon%27%2C%20%27quant%27%2C%20%27-i%27%2C%20%27transcript_index%27%2C%20%27-l%27%2C%20%27A%27%2C%20%27-r%27%2C%20%27reads.fq%27%2C%20%27-o%27%2C%20%27quant_output%27%5D%29%0A%0ANext%2C%20the%20notebook%20demonstrates%20differential%20splicing%20analysis%20using%20the%203D%20RNA-seq%20toolkit%20and%20visualization%20of%20isoform%20usage%20differences%20between%20WT%20and%20mutant%20samples.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20matplotlib.pyplot%20as%20plt%0A%0A%23%20Load%20processed%20data%0Adf%20%3D%20pd.read_csv%28%27differential_splicing_results.csv%27%29%0A%0A%23%20Plot%20isoform%20usage%20differences%0Aplt.bar%28df%5B%27Isoform%27%5D%2C%20df%5B%27Usage%27%5D%2C%20color%3D%5B%27%236A0C76%27%2C%20%27%23B565A7%27%5D%29%0Aplt.xlabel%28%27DGK5%20Isoforms%27%29%0Aplt.ylabel%28%27Relative%20Usage%20%28%25%29%27%29%0Aplt.title%28%27Isoform%20Usage%20in%20WT%20vs%20cpl3%20Mutant%27%29%0Aplt.show%28%29%0A%0AThe%20code%20block%20provides%20a%20basic%20pipeline%20from%20data%20acquisition%20to%20visualization%2C%20suitable%20for%20extension%20to%20more%20complex%20bioinformatics%20workflows.%0A%0A%23%20Further%20statistical%20analysis%20and%20refinement%20of%20splicing%20events%20can%20be%20added%20based%20on%20the%20initial%20results%20obtained.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Global%20profiling%20of%20CPL3-mediated%20alternative%20splicing%20reveals%20regulatory%20mechanisms%20of%20DGK5%20in%20plant%20immunity%20and%20phosphatidic%20acid%20homeostasis)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "Code downloads Arabidopsis RNA-seq datasets, performs differential splicing analysis using 3D RNA-seq pipelines, and visualizes splicing changes.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-04-02", "tags": ["CPL3 alternative splicing", "biology", "bioinformatics"]}}