{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "This block describes the process of importing the wheat SNP dataset and performing candidate gene validation using GWAS results from the study."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\n# Download dataset containing SNP data and candidate gene lists\nsnp_data = pd.read_csv('https://ngdc.cncb.ac.cn/bioproject/browse/PRJCA030529/variants.csv')\ncandidate_genes = pd.read_csv('candidate_genes.csv')\n\n# Merge and filter the dataset based on significant markers\nmerged_data = pd.merge(snp_data, candidate_genes, on='gene_id', how='inner')\nfiltered_data = merged_data[merged_data['p_value'] < 1e-5]\nprint(filtered_data.head())"}, {"cell_type": "markdown", "metadata": {}, "source": "Next, we perform haplotype analysis and visualize the distribution of superior haplotypes across different breeding periods."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import matplotlib.pyplot as plt\n# Assume filtered_data contains a 'haplotype' column and 'breeding_period'\nhaplotype_counts = filtered_data.groupby(['breeding_period', 'haplotype']).size().unstack(fill_value=0)\nhaplotype_counts.plot(kind='bar', stacked=True, colormap='viridis', figsize=(10,6))\nplt.title('Superior Haplotype Distribution by Breeding Period')\nplt.xlabel('Breeding Period')\nplt.ylabel('Count')\nplt.legend(title='Haplotype', bbox_to_anchor=(1.05, 1), loc='upper left')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "This notebook provides a basic pipeline to process the genome-wide data, cross-reference candidate genes, and visualize relevant haplotype distributions to validate the study's findings."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20Downloads%20and%20processes%20the%20wheat%20genome-wide%20dataset%20to%20validate%20candidate%20gene%20associations%20and%20haplotype%20structures.%0A%0AInclude%20more%20advanced%20statistical%20models%2C%20integrate%20environmental%20data%2C%20and%20validate%20with%20independent%20datasets.%0A%0AGenome-wide%20analysis%20genetic%20basis%20agronomical%20traits%20wheat%20breeding%20Henan%20Province%0A%0AThis%20block%20describes%20the%20process%20of%20importing%20the%20wheat%20SNP%20dataset%20and%20performing%20candidate%20gene%20validation%20using%20GWAS%20results%20from%20the%20study.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0A%23%20Download%20dataset%20containing%20SNP%20data%20and%20candidate%20gene%20lists%0Asnp_data%20%3D%20pd.read_csv%28%27https%3A%2F%2Fngdc.cncb.ac.cn%2Fbioproject%2Fbrowse%2FPRJCA030529%2Fvariants.csv%27%29%0Acandidate_genes%20%3D%20pd.read_csv%28%27candidate_genes.csv%27%29%0A%0A%23%20Merge%20and%20filter%20the%20dataset%20based%20on%20significant%20markers%0Amerged_data%20%3D%20pd.merge%28snp_data%2C%20candidate_genes%2C%20on%3D%27gene_id%27%2C%20how%3D%27inner%27%29%0Afiltered_data%20%3D%20merged_data%5Bmerged_data%5B%27p_value%27%5D%20%3C%201e-5%5D%0Aprint%28filtered_data.head%28%29%29%0A%0ANext%2C%20we%20perform%20haplotype%20analysis%20and%20visualize%20the%20distribution%20of%20superior%20haplotypes%20across%20different%20breeding%20periods.%0A%0Aimport%20matplotlib.pyplot%20as%20plt%0A%23%20Assume%20filtered_data%20contains%20a%20%27haplotype%27%20column%20and%20%27breeding_period%27%0Ahaplotype_counts%20%3D%20filtered_data.groupby%28%5B%27breeding_period%27%2C%20%27haplotype%27%5D%29.size%28%29.unstack%28fill_value%3D0%29%0Ahaplotype_counts.plot%28kind%3D%27bar%27%2C%20stacked%3DTrue%2C%20colormap%3D%27viridis%27%2C%20figsize%3D%2810%2C6%29%29%0Aplt.title%28%27Superior%20Haplotype%20Distribution%20by%20Breeding%20Period%27%29%0Aplt.xlabel%28%27Breeding%20Period%27%29%0Aplt.ylabel%28%27Count%27%29%0Aplt.legend%28title%3D%27Haplotype%27%2C%20bbox_to_anchor%3D%281.05%2C%201%29%2C%20loc%3D%27upper%20left%27%29%0Aplt.show%28%29%0A%0AThis%20notebook%20provides%20a%20basic%20pipeline%20to%20process%20the%20genome-wide%20data%2C%20cross-reference%20candidate%20genes%2C%20and%20visualize%20relevant%20haplotype%20distributions%20to%20validate%20the%20study%27s%20findings.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Genome-wide%20analysis%20reveals%20the%20genetic%20basis%20of%20key%20agronomical%20traits%20and%20modern%20wheat%20breeding%20in%20Henan%20Province)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "Downloads and processes the wheat genome-wide dataset to validate candidate gene associations and haplotype structures.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-04-02", "tags": ["genetic basis agronomical traits", "biology", "bioinformatics"]}}