{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "This section describes how to download raw sequence data from the ENA based on accession numbers provided by the EMO BON release. It then preprocesses the data and calculates diversity indices using BioPython and SciPy."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import os\nimport requests\nfrom Bio import SeqIO\nimport numpy as np\nfrom scipy.stats import entropy\n\n# Example: Download a sample FASTA file from ENA (placeholder accession number)\nurl = 'https://www.ebi.ac.uk/ena/browser/api/fasta/your_accession_number'\nresponse = requests.get(url)\nwith open('sample.fasta', 'w') as out_file:\n    out_file.write(response.text)\n\n# Parse sequences\nsequences = list(SeqIO.parse('sample.fasta', 'fasta'))\nprint('Number of sequences:', len(sequences))\n\n# Calculate nucleotide diversity (example on sequence lengths)\nlengths = np.array([len(rec.seq) for rec in sequences])\ndiversity_index = entropy(np.bincount(lengths))\nprint('Diversity Index:', diversity_index)"}, {"cell_type": "markdown", "metadata": {}, "source": "The code above illustrates the downloading of sequencing data, parsing with BioPython, and calculating a rudimentary diversity index using SciPy's entropy to summarize sequence length distribution. This pipeline can be expanded to include taxonomic assignments and advanced diversity metrics."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Further analysis steps could include mapping reads to reference databases, assembling contigs, and more sophisticated diversity and network analyses using pandas and scikit-learn.\nimport pandas as pd\n\n# Create a sample DataFrame of sequence features\ndata = {'SequenceID': [rec.id for rec in sequences], 'Length': [len(rec.seq) for rec in sequences]}\ndf = pd.DataFrame(data)\nprint(df.head())"}, {"cell_type": "markdown", "metadata": {}, "source": "This notebook section is designed to act as a starting point for processing the EMO BON dataset and can be modified for comprehensive statistical analysis and visualization of metagenomic diversity."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20The%20code%20downloads%20and%20processes%20EMO%20BON%20datasets%20to%20analyze%20diversity%20metrics%20using%20shotgun%20metagenomics%20data.%0A%0AInclude%20real%20ENA%20accession%20links%20and%20integrate%20taxonomic%20classification%20pipelines%20for%20improved%20ecological%20interpretation.%0A%0AEuropean%20marine%20omics%20biodiversity%20observation%20network%20EMO%20BON%20shotgun%20metagenomics%20data%20water%20sediment%20samples%0A%0AThis%20section%20describes%20how%20to%20download%20raw%20sequence%20data%20from%20the%20ENA%20based%20on%20accession%20numbers%20provided%20by%20the%20EMO%20BON%20release.%20It%20then%20preprocesses%20the%20data%20and%20calculates%20diversity%20indices%20using%20BioPython%20and%20SciPy.%0A%0Aimport%20os%0Aimport%20requests%0Afrom%20Bio%20import%20SeqIO%0Aimport%20numpy%20as%20np%0Afrom%20scipy.stats%20import%20entropy%0A%0A%23%20Example%3A%20Download%20a%20sample%20FASTA%20file%20from%20ENA%20%28placeholder%20accession%20number%29%0Aurl%20%3D%20%27https%3A%2F%2Fwww.ebi.ac.uk%2Fena%2Fbrowser%2Fapi%2Ffasta%2Fyour_accession_number%27%0Aresponse%20%3D%20requests.get%28url%29%0Awith%20open%28%27sample.fasta%27%2C%20%27w%27%29%20as%20out_file%3A%0A%20%20%20%20out_file.write%28response.text%29%0A%0A%23%20Parse%20sequences%0Asequences%20%3D%20list%28SeqIO.parse%28%27sample.fasta%27%2C%20%27fasta%27%29%29%0Aprint%28%27Number%20of%20sequences%3A%27%2C%20len%28sequences%29%29%0A%0A%23%20Calculate%20nucleotide%20diversity%20%28example%20on%20sequence%20lengths%29%0Alengths%20%3D%20np.array%28%5Blen%28rec.seq%29%20for%20rec%20in%20sequences%5D%29%0Adiversity_index%20%3D%20entropy%28np.bincount%28lengths%29%29%0Aprint%28%27Diversity%20Index%3A%27%2C%20diversity_index%29%0A%0AThe%20code%20above%20illustrates%20the%20downloading%20of%20sequencing%20data%2C%20parsing%20with%20BioPython%2C%20and%20calculating%20a%20rudimentary%20diversity%20index%20using%20SciPy%27s%20entropy%20to%20summarize%20sequence%20length%20distribution.%20This%20pipeline%20can%20be%20expanded%20to%20include%20taxonomic%20assignments%20and%20advanced%20diversity%20metrics.%0A%0A%23%20Further%20analysis%20steps%20could%20include%20mapping%20reads%20to%20reference%20databases%2C%20assembling%20contigs%2C%20and%20more%20sophisticated%20diversity%20and%20network%20analyses%20using%20pandas%20and%20scikit-learn.%0Aimport%20pandas%20as%20pd%0A%0A%23%20Create%20a%20sample%20DataFrame%20of%20sequence%20features%0Adata%20%3D%20%7B%27SequenceID%27%3A%20%5Brec.id%20for%20rec%20in%20sequences%5D%2C%20%27Length%27%3A%20%5Blen%28rec.seq%29%20for%20rec%20in%20sequences%5D%7D%0Adf%20%3D%20pd.DataFrame%28data%29%0Aprint%28df.head%28%29%29%0A%0AThis%20notebook%20section%20is%20designed%20to%20act%20as%20a%20starting%20point%20for%20processing%20the%20EMO%20BON%20dataset%20and%20can%20be%20modified%20for%20comprehensive%20statistical%20analysis%20and%20visualization%20of%20metagenomic%20diversity.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20First%20release%20of%20the%20European%20marine%20omics%20biodiversity%20observation%20network%20%28EMO%20BON%29%20shotgun%20metagenomics%20data%20from%20water%20and%20sediment%20samples)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "The code downloads and processes EMO BON datasets to analyze diversity metrics using shotgun metagenomics data.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-27", "tags": ["EMO BON metagenomics data", "biology", "bioinformatics"]}}