{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "The notebook describes the process of loading 16S rRNA amplicon data, performing quality control, and calculating Shannon diversity indices across different feeding regimes using QIIME2 and relevant Python libraries."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import qiime2\nimport pandas as pd\nimport numpy as np\nfrom skbio.diversity.alpha import shannon\n# Load dataset\nmetadata = pd.read_csv('metadata.csv')\ndata = pd.read_csv('feature-table.csv')\n# Calculate Shannon diversity\nmetadata['shannon'] = data.apply(lambda row: shannon(row.values), axis=1)\nprint(metadata[['sample_id', 'shannon']].head())"}, {"cell_type": "markdown", "metadata": {}, "source": "This code is useful for analyzing how different feeding parameters affect gut microbiome diversity by directly computing Shannon diversity from microbial abundances."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# The following section plots the diversity data\nimport plotly.express as px\nfig = px.bar(metadata, x='sample_id', y='shannon', title='Shannon Diversity across Samples')\nfig.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20The%20Python%20code%20downloads%20and%20processes%2016S%20rRNA%20sequencing%20data%20to%20quantify%20Shannon%20diversity%20scores%20under%20varying%20feeding%20parameters.%0A%0AInclude%20error%20handling%2C%20integrate%20metadata%20filtering%2C%20and%20add%20statistical%20tests%20to%20compare%20diversity%20across%20feeding%20regimens.%0A%0AOptimizing%20microbial%20intake%20gut%20microbiome%20diversity%0A%0AThe%20notebook%20describes%20the%20process%20of%20loading%2016S%20rRNA%20amplicon%20data%2C%20performing%20quality%20control%2C%20and%20calculating%20Shannon%20diversity%20indices%20across%20different%20feeding%20regimes%20using%20QIIME2%20and%20relevant%20Python%20libraries.%0A%0Aimport%20qiime2%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0Afrom%20skbio.diversity.alpha%20import%20shannon%0A%23%20Load%20dataset%0Ametadata%20%3D%20pd.read_csv%28%27metadata.csv%27%29%0Adata%20%3D%20pd.read_csv%28%27feature-table.csv%27%29%0A%23%20Calculate%20Shannon%20diversity%0Ametadata%5B%27shannon%27%5D%20%3D%20data.apply%28lambda%20row%3A%20shannon%28row.values%29%2C%20axis%3D1%29%0Aprint%28metadata%5B%5B%27sample_id%27%2C%20%27shannon%27%5D%5D.head%28%29%29%0A%0AThis%20code%20is%20useful%20for%20analyzing%20how%20different%20feeding%20parameters%20affect%20gut%20microbiome%20diversity%20by%20directly%20computing%20Shannon%20diversity%20from%20microbial%20abundances.%0A%0A%23%20The%20following%20section%20plots%20the%20diversity%20data%0Aimport%20plotly.express%20as%20px%0Afig%20%3D%20px.bar%28metadata%2C%20x%3D%27sample_id%27%2C%20y%3D%27shannon%27%2C%20title%3D%27Shannon%20Diversity%20across%20Samples%27%29%0Afig.show%28%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Optimizing%20microbial%20intake%20helps%20to%20maintain%20the%20gut%20microbiome%20diversity.)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "The Python code downloads and processes 16S rRNA sequencing data to quantify Shannon diversity scores under varying feeding parameters.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-15", "tags": ["microbial intake optimization", "biology", "bioinformatics"]}}