{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "This notebook imports the extracted sample data to build visualizations that compare lineage prevalence across pandemic phases."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport plotly.express as px\n\n# Define the dataset based on the extracted data\ndata = [\n    {'lineage': 'B.1.1.33', 'samples': 46, 'percentage': 52.9},\n    {'lineage': 'B.1.1.28.2', 'samples': 15, 'percentage': 17.2},\n    {'lineage': 'B.1.1.28.1', 'samples': 9, 'percentage': 10.3},\n    {'lineage': 'BQ.1.1', 'samples': 60, 'percentage': 60.6},\n    {'lineage': 'BE.9', 'samples': 9, 'percentage': 9.1},\n    {'lineage': 'BQ.1.18', 'samples': 7, 'percentage': 7.1}\n]\n\n# Create dataframe\ndf = pd.DataFrame(data)\n\n# Plotting sample counts grouped by phase (assumed grouping)\nfig = px.bar(df, x='lineage', y='samples', text='percentage',\n             title='SARS-CoV-2 Lineage Sample Distribution')\nfig.update_traces(texttemplate='%{text}%', textposition='outside')\nfig.show()\n"}, {"cell_type": "markdown", "metadata": {}, "source": "The above bar graph illustrates the comparative sample counts between pre-Omicron (B.1.1.x) and post-Omicron (BQ.1.1, BE.9, BQ.1.18) lineages, thus visually supporting the paper's conclusions."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import numpy as np\n\n# Adding a log-transformed scaling for further analysis\ndf['log_samples'] = np.log(df['samples'] + 1)\nfig2 = px.line(df, x='lineage', y='log_samples', title='Log-Transformed Sample Counts')\nfig2.show()\n"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20The%20code%20conducts%20phylogenetic%20visualization%20and%20statistical%20analysis%20of%20lineage%20data%2C%20comparing%20pre-%20and%20post-Omicron%20samples.%0A%0AInclude%20statistical%20comparisons%20and%20integrate%20geographic%20metadata%20to%20refine%20dispersion%20analysis.%0A%0AComparative%20analysis%20SARS-CoV-2%20lineages%20B.1.1.33%20BQ.1.1%20Omicron%20variant%20pandemic%20phases%0A%0AThis%20notebook%20imports%20the%20extracted%20sample%20data%20to%20build%20visualizations%20that%20compare%20lineage%20prevalence%20across%20pandemic%20phases.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20plotly.express%20as%20px%0A%0A%23%20Define%20the%20dataset%20based%20on%20the%20extracted%20data%0Adata%20%3D%20%5B%0A%20%20%20%20%7B%27lineage%27%3A%20%27B.1.1.33%27%2C%20%27samples%27%3A%2046%2C%20%27percentage%27%3A%2052.9%7D%2C%0A%20%20%20%20%7B%27lineage%27%3A%20%27B.1.1.28.2%27%2C%20%27samples%27%3A%2015%2C%20%27percentage%27%3A%2017.2%7D%2C%0A%20%20%20%20%7B%27lineage%27%3A%20%27B.1.1.28.1%27%2C%20%27samples%27%3A%209%2C%20%27percentage%27%3A%2010.3%7D%2C%0A%20%20%20%20%7B%27lineage%27%3A%20%27BQ.1.1%27%2C%20%27samples%27%3A%2060%2C%20%27percentage%27%3A%2060.6%7D%2C%0A%20%20%20%20%7B%27lineage%27%3A%20%27BE.9%27%2C%20%27samples%27%3A%209%2C%20%27percentage%27%3A%209.1%7D%2C%0A%20%20%20%20%7B%27lineage%27%3A%20%27BQ.1.18%27%2C%20%27samples%27%3A%207%2C%20%27percentage%27%3A%207.1%7D%0A%5D%0A%0A%23%20Create%20dataframe%0Adf%20%3D%20pd.DataFrame%28data%29%0A%0A%23%20Plotting%20sample%20counts%20grouped%20by%20phase%20%28assumed%20grouping%29%0Afig%20%3D%20px.bar%28df%2C%20x%3D%27lineage%27%2C%20y%3D%27samples%27%2C%20text%3D%27percentage%27%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20title%3D%27SARS-CoV-2%20Lineage%20Sample%20Distribution%27%29%0Afig.update_traces%28texttemplate%3D%27%25%7Btext%7D%25%27%2C%20textposition%3D%27outside%27%29%0Afig.show%28%29%0A%0A%0AThe%20above%20bar%20graph%20illustrates%20the%20comparative%20sample%20counts%20between%20pre-Omicron%20%28B.1.1.x%29%20and%20post-Omicron%20%28BQ.1.1%2C%20BE.9%2C%20BQ.1.18%29%20lineages%2C%20thus%20visually%20supporting%20the%20paper%27s%20conclusions.%0A%0Aimport%20numpy%20as%20np%0A%0A%23%20Adding%20a%20log-transformed%20scaling%20for%20further%20analysis%0Adf%5B%27log_samples%27%5D%20%3D%20np.log%28df%5B%27samples%27%5D%20%2B%201%29%0Afig2%20%3D%20px.line%28df%2C%20x%3D%27lineage%27%2C%20y%3D%27log_samples%27%2C%20title%3D%27Log-Transformed%20Sample%20Counts%27%29%0Afig2.show%28%29%0A%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Comparative%20spatial%E2%80%93temporal%20analysis%20of%20SARS-CoV-2%20lineages%20B.1.1.33%20and%20BQ.1.1%20Omicron%20variant%20across%20pandemic%20phases)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "The code conducts phylogenetic visualization and statistical analysis of lineage data, comparing pre- and post-Omicron samples.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-04-03", "tags": ["SARS-CoV-2 B.1.1.33 variant", "biology", "bioinformatics"]}}