{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "This section downloads and preprocesses the relevant scRNA-seq and ATAC-seq datasets, performs clustering, and integrates gene expression with chromatin accessibility profiles."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nimport anndata\nimport matplotlib.pyplot as plt\n\n# Load the single-cell RNA-seq dataset\nadata = sc.read_h5ad('path_to_scRNAseq_data.h5ad')\n# Preprocess the data\nsc.pp.filter_cells(adata, min_genes=200)\nsc.pp.normalize_total(adata, target_sum=1e4)\nsc.pp.log1p(adata)\nsc.tl.pca(adata, svd_solver='arpack')\nsc.pp.neighbors(adata, n_neighbors=10, n_pcs=40)\nsc.tl.umap(adata)\n\n# Plot UMAP visualization\nsc.pl.umap(adata, color=['HDAC1', 'TSLP'], save='_pTh2_cells.png')"}, {"cell_type": "markdown", "metadata": {}, "source": "Next, the code integrates ATAC-seq signals by aligning chromatin accessibility peaks with gene expression data, thereby highlighting regions modulated by HDAC1."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\n\n# Load ATAC-seq peak data\natac_peaks = pd.read_csv('path_to_ATAC_seq_data.tsv', sep='\\t')\n\n# Merge with gene expression based on genomic coordinates / gene identifiers (example merge)\nmerged_data = pd.merge(adata.obs, atac_peaks, left_on='gene_id', right_on='gene_id', how='inner')\n\n# Visualize the correlation between ATAC accessibility and gene expression\nplt.figure(figsize=(8,6))\nplt.scatter(merged_data['expression_level'], merged_data['accessibility_score'], c='purple', alpha=0.6)\nplt.xlabel('Gene Expression Level')\nplt.ylabel('Chromatin Accessibility Score')\nplt.title('Correlation between Gene Expression and ATAC-seq Peaks')\nplt.savefig('correlation_plot.png')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "This notebook demonstrates a pipeline integrating scRNA-seq and ATAC-seq datasets, providing insights into HDAC1-mediated regulation in pTh2 cells."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20The%20Python%20code%20integrates%20single-cell%20RNA-seq%20and%20ATAC-seq%20data%20to%20identify%20differential%20chromatin%20regions%20and%20gene%20expression%20signatures%20in%20pTh2%20cells.%0A%0AInclude%20direct%20links%20to%20real%20dataset%20accession%20codes%20and%20further%20optimize%20integration%20algorithms%20for%20multi-omic%20data%20fusion.%0A%0ASingle-cell%20chromatin%20accessibility%20Th2%20cells%20allergic%20asthma%20regulatory%20programs%0A%0AThis%20section%20downloads%20and%20preprocesses%20the%20relevant%20scRNA-seq%20and%20ATAC-seq%20datasets%2C%20performs%20clustering%2C%20and%20integrates%20gene%20expression%20with%20chromatin%20accessibility%20profiles.%0A%0Aimport%20scanpy%20as%20sc%0Aimport%20anndata%0Aimport%20matplotlib.pyplot%20as%20plt%0A%0A%23%20Load%20the%20single-cell%20RNA-seq%20dataset%0Aadata%20%3D%20sc.read_h5ad%28%27path_to_scRNAseq_data.h5ad%27%29%0A%23%20Preprocess%20the%20data%0Asc.pp.filter_cells%28adata%2C%20min_genes%3D200%29%0Asc.pp.normalize_total%28adata%2C%20target_sum%3D1e4%29%0Asc.pp.log1p%28adata%29%0Asc.tl.pca%28adata%2C%20svd_solver%3D%27arpack%27%29%0Asc.pp.neighbors%28adata%2C%20n_neighbors%3D10%2C%20n_pcs%3D40%29%0Asc.tl.umap%28adata%29%0A%0A%23%20Plot%20UMAP%20visualization%0Asc.pl.umap%28adata%2C%20color%3D%5B%27HDAC1%27%2C%20%27TSLP%27%5D%2C%20save%3D%27_pTh2_cells.png%27%29%0A%0ANext%2C%20the%20code%20integrates%20ATAC-seq%20signals%20by%20aligning%20chromatin%20accessibility%20peaks%20with%20gene%20expression%20data%2C%20thereby%20highlighting%20regions%20modulated%20by%20HDAC1.%0A%0Aimport%20pandas%20as%20pd%0A%0A%23%20Load%20ATAC-seq%20peak%20data%0Aatac_peaks%20%3D%20pd.read_csv%28%27path_to_ATAC_seq_data.tsv%27%2C%20sep%3D%27%5Ct%27%29%0A%0A%23%20Merge%20with%20gene%20expression%20based%20on%20genomic%20coordinates%20%2F%20gene%20identifiers%20%28example%20merge%29%0Amerged_data%20%3D%20pd.merge%28adata.obs%2C%20atac_peaks%2C%20left_on%3D%27gene_id%27%2C%20right_on%3D%27gene_id%27%2C%20how%3D%27inner%27%29%0A%0A%23%20Visualize%20the%20correlation%20between%20ATAC%20accessibility%20and%20gene%20expression%0Aplt.figure%28figsize%3D%288%2C6%29%29%0Aplt.scatter%28merged_data%5B%27expression_level%27%5D%2C%20merged_data%5B%27accessibility_score%27%5D%2C%20c%3D%27purple%27%2C%20alpha%3D0.6%29%0Aplt.xlabel%28%27Gene%20Expression%20Level%27%29%0Aplt.ylabel%28%27Chromatin%20Accessibility%20Score%27%29%0Aplt.title%28%27Correlation%20between%20Gene%20Expression%20and%20ATAC-seq%20Peaks%27%29%0Aplt.savefig%28%27correlation_plot.png%27%29%0Aplt.show%28%29%0A%0AThis%20notebook%20demonstrates%20a%20pipeline%20integrating%20scRNA-seq%20and%20ATAC-seq%20datasets%2C%20providing%20insights%20into%20HDAC1-mediated%20regulation%20in%20pTh2%20cells.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Single-cell%20and%20chromatin%20accessibility%20profiling%20reveals%20regulatory%20programs%20of%20pathogenic%20Th2%20cells%20in%20allergic%20asthma)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "The Python code integrates single-cell RNA-seq and ATAC-seq data to identify differential chromatin regions and gene expression signatures in pTh2 cells.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-29", "tags": ["single-cell profiling Th2 cells", "biology", "bioinformatics"]}}