{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "This section describes the analysis workflow using real datasets (e.g., from GEO datasets GSE178341, GSE232525) to reproduce the paper's findings."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nimport pandas as pd\n\n# Load single-cell RNA-seq data\nadata = sc.read_h5ad('path_to_scRNAseq_data.h5ad')\n\n# Preprocessing steps\nsc.pp.filter_cells(adata, min_genes=500)\nsc.pp.filter_genes(adata, min_cells=10)\nsc.pp.normalize_total(adata, target_sum=1e4)\nsc.pp.log1p(adata)\n\n# Identify CAF clusters\nsc.pp.highly_variable_genes(adata)\nsc.tl.pca(adata)\nsc.pp.neighbors(adata)\nsc.tl.umap(adata)\nsc.tl.leiden(adata, resolution=0.5)\n\n# Visualize clusters\nsc.pl.umap(adata, color=['leiden'], save='_clusters.png')\n\n# Further annotate clusters filtering CAFs using known marker genes\ncaf_markers = ['FAP', 'SFRP2']\nadata.obs['CAF'] = adata.obs['leiden'].apply(lambda x: 'CAF' if 'identified_cluster' in x else 'Other')\nsc.pl.umap(adata, color=['CAF'])\n\n# Integration with spatial data can be performed similarly using squidpy\nimport squidpy as sq\nspatial = sq.datasets.visium_hne_adata()\n\n# Further detailed analysis can include cell-cell communication analysis using CellChat equivalent in Python.\n\n# Export results for further analysis\nadata.write('processed_scRNAseq_data.h5ad')"}, {"cell_type": "markdown", "metadata": {}, "source": "The above blocks constitute a simplified reproducible workflow. In practice, one would integrate spatial transcriptomics data and perform communication network analysis to mirror the paper's approach."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import matplotlib.pyplot as plt\nplt.figure(figsize=(8,6))\nsc.pl.umap(adata, color=['CAF'], show=False)\nplt.title('CAF Distribution in scRNA-seq Data')\nplt.savefig('CAF_Distribution.png')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "This interactive notebook demonstrates the utility of cutting-edge bioinformatics tools (scanpy, squidpy) to analyze heterogeneous cell populations in colorectal cancer, highlighting the clinical relevance of metabolic subtyping and CAF targeting."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20The%20code%20performs%20integrative%20analysis%20of%20single-cell%20and%20spatial%20transcriptomics%20data%20to%20identify%20CAF%20subpopulations%20and%20correlate%20their%20distribution%20with%20metabolic%20CRC%20subtypes.%0A%0AIncorporate%20spatial%20coordinates%20from%20spatial%20transcriptomics%2C%20integrate%20patient%20survival%20data%2C%20and%20perform%20detailed%20cell-cell%20interaction%20analysis%20with%20more%20advanced%20tools.%0A%0ASingle-cell%20profiling%20cancer-associated%20fibroblasts%20colorectal%20cancer%20metabolic%20subtypes%0A%0AThis%20section%20describes%20the%20analysis%20workflow%20using%20real%20datasets%20%28e.g.%2C%20from%20GEO%20datasets%20GSE178341%2C%20GSE232525%29%20to%20reproduce%20the%20paper%27s%20findings.%0A%0Aimport%20scanpy%20as%20sc%0Aimport%20pandas%20as%20pd%0A%0A%23%20Load%20single-cell%20RNA-seq%20data%0Aadata%20%3D%20sc.read_h5ad%28%27path_to_scRNAseq_data.h5ad%27%29%0A%0A%23%20Preprocessing%20steps%0Asc.pp.filter_cells%28adata%2C%20min_genes%3D500%29%0Asc.pp.filter_genes%28adata%2C%20min_cells%3D10%29%0Asc.pp.normalize_total%28adata%2C%20target_sum%3D1e4%29%0Asc.pp.log1p%28adata%29%0A%0A%23%20Identify%20CAF%20clusters%0Asc.pp.highly_variable_genes%28adata%29%0Asc.tl.pca%28adata%29%0Asc.pp.neighbors%28adata%29%0Asc.tl.umap%28adata%29%0Asc.tl.leiden%28adata%2C%20resolution%3D0.5%29%0A%0A%23%20Visualize%20clusters%0Asc.pl.umap%28adata%2C%20color%3D%5B%27leiden%27%5D%2C%20save%3D%27_clusters.png%27%29%0A%0A%23%20Further%20annotate%20clusters%20filtering%20CAFs%20using%20known%20marker%20genes%0Acaf_markers%20%3D%20%5B%27FAP%27%2C%20%27SFRP2%27%5D%0Aadata.obs%5B%27CAF%27%5D%20%3D%20adata.obs%5B%27leiden%27%5D.apply%28lambda%20x%3A%20%27CAF%27%20if%20%27identified_cluster%27%20in%20x%20else%20%27Other%27%29%0Asc.pl.umap%28adata%2C%20color%3D%5B%27CAF%27%5D%29%0A%0A%23%20Integration%20with%20spatial%20data%20can%20be%20performed%20similarly%20using%20squidpy%0Aimport%20squidpy%20as%20sq%0Aspatial%20%3D%20sq.datasets.visium_hne_adata%28%29%0A%0A%23%20Further%20detailed%20analysis%20can%20include%20cell-cell%20communication%20analysis%20using%20CellChat%20equivalent%20in%20Python.%0A%0A%23%20Export%20results%20for%20further%20analysis%0Aadata.write%28%27processed_scRNAseq_data.h5ad%27%29%0A%0AThe%20above%20blocks%20constitute%20a%20simplified%20reproducible%20workflow.%20In%20practice%2C%20one%20would%20integrate%20spatial%20transcriptomics%20data%20and%20perform%20communication%20network%20analysis%20to%20mirror%20the%20paper%27s%20approach.%0A%0Aimport%20matplotlib.pyplot%20as%20plt%0Aplt.figure%28figsize%3D%288%2C6%29%29%0Asc.pl.umap%28adata%2C%20color%3D%5B%27CAF%27%5D%2C%20show%3DFalse%29%0Aplt.title%28%27CAF%20Distribution%20in%20scRNA-seq%20Data%27%29%0Aplt.savefig%28%27CAF_Distribution.png%27%29%0Aplt.show%28%29%0A%0AThis%20interactive%20notebook%20demonstrates%20the%20utility%20of%20cutting-edge%20bioinformatics%20tools%20%28scanpy%2C%20squidpy%29%20to%20analyze%20heterogeneous%20cell%20populations%20in%20colorectal%20cancer%2C%20highlighting%20the%20clinical%20relevance%20of%20metabolic%20subtyping%20and%20CAF%20targeting.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Single-cell%20and%20spatial-resolved%20profiling%20reveals%20cancer-associated%20fibroblast%20heterogeneity%20in%20colorectal%20cancer%20metabolic%20subtypes)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "The code performs integrative analysis of single-cell and spatial transcriptomics data to identify CAF subpopulations and correlate their distribution with metabolic CRC subtypes.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-24", "tags": ["single-cell profiling colorectal cancer", "biology", "bioinformatics"]}}