{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "This notebook downloads the bone marrow scRNA-seq dataset and performs quality control, normalization, clustering, and identification of the RAB13+ EC subpopulation using Scanpy."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nimport pandas as pd\n\n# Download dataset (assuming dataset identifier URL_0 corresponds to a publicly available dataset)\nadata = sc.read_h5ad('path_to_bm_scRNAseq_data.h5ad')\n\n# Basic preprocessing\nsc.pp.filter_cells(adata, min_genes=200)\nsc.pp.filter_genes(adata, min_cells=3)\nadata.var['mt'] = adata.var_names.str.startswith('MT-')\nsc.pp.calculate_qc_metrics(adata, qc_vars=['mt'], percent_top=None, log1p=False, inplace=True)\n\n# Normalize, log transform and identify HVGs\nsc.pp.normalize_total(adata, target_sum=1e4)\nsc.pp.log1p(adata)\nsc.pp.highly_variable_genes(adata, min_mean=0.0125, max_mean=3, min_disp=0.5)\nadata = adata[:, adata.var.highly_variable]\n\n# PCA and clustering\nsc.tl.pca(adata, svd_solver='arpack')\nsc.pp.neighbors(adata, n_neighbors=10, n_pcs=40)\nsc.tl.umap(adata)\nsc.tl.leiden(adata, resolution=0.5)\n\n# Identify RAB13+ cluster by marker gene expression\nsc.tl.rank_genes_groups(adata, 'leiden', method='t-test')\n\n# Visualize clusters and marker expression\nsc.pl.umap(adata, color=['leiden', 'RAB13'], save='_bm_clusters.png')"}, {"cell_type": "markdown", "metadata": {}, "source": "The following steps enable further investigation of ligand-receptor interactions between the identified RAB13+ EC and MSC populations using existing cell communication analysis tools."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Example: Compute cell-cell communication using Squidpy\nimport squidpy as sq\n\n# Convert Scanpy AnnData to Squidpy format and compute neighbors\nsq.gr.spatial_neighbors(adata, coord_type='grid')\nsq.gr.nhood_enrichment(adata, cluster_key='leiden')\n\n# Plot enrichment results\nsq.pl.nhood_enrichment(adata, key='nhood_enrichment', cmap='viridis', save='_nhood_enrichment.png')"}, {"cell_type": "markdown", "metadata": {}, "source": "These analyses help delineate the molecular interactions and networks that underpin age-related alterations in the bone marrow microenvironment."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20The%20code%20processes%20scRNA-seq%20data%20from%20bone%20marrow%20samples%20to%20identify%20and%20visualize%20different%20endothelial%20and%20mesenchymal%20subpopulations%20and%20their%20communication%20networks.%0A%0AIntegrate%20metadata%20on%20age%20groups%20and%20use%20additional%20ligand-receptor%20databases%20to%20refine%20intercellular%20communication%20analysis.%0A%0ASingle-cell%20RNA%20profiling%20RAB13%20endothelial%20mesenchymal%20cells%20aged%20bone%20marrow%0A%0AThis%20notebook%20downloads%20the%20bone%20marrow%20scRNA-seq%20dataset%20and%20performs%20quality%20control%2C%20normalization%2C%20clustering%2C%20and%20identification%20of%20the%20RAB13%2B%20EC%20subpopulation%20using%20Scanpy.%0A%0Aimport%20scanpy%20as%20sc%0Aimport%20pandas%20as%20pd%0A%0A%23%20Download%20dataset%20%28assuming%20dataset%20identifier%20URL_0%20corresponds%20to%20a%20publicly%20available%20dataset%29%0Aadata%20%3D%20sc.read_h5ad%28%27path_to_bm_scRNAseq_data.h5ad%27%29%0A%0A%23%20Basic%20preprocessing%0Asc.pp.filter_cells%28adata%2C%20min_genes%3D200%29%0Asc.pp.filter_genes%28adata%2C%20min_cells%3D3%29%0Aadata.var%5B%27mt%27%5D%20%3D%20adata.var_names.str.startswith%28%27MT-%27%29%0Asc.pp.calculate_qc_metrics%28adata%2C%20qc_vars%3D%5B%27mt%27%5D%2C%20percent_top%3DNone%2C%20log1p%3DFalse%2C%20inplace%3DTrue%29%0A%0A%23%20Normalize%2C%20log%20transform%20and%20identify%20HVGs%0Asc.pp.normalize_total%28adata%2C%20target_sum%3D1e4%29%0Asc.pp.log1p%28adata%29%0Asc.pp.highly_variable_genes%28adata%2C%20min_mean%3D0.0125%2C%20max_mean%3D3%2C%20min_disp%3D0.5%29%0Aadata%20%3D%20adata%5B%3A%2C%20adata.var.highly_variable%5D%0A%0A%23%20PCA%20and%20clustering%0Asc.tl.pca%28adata%2C%20svd_solver%3D%27arpack%27%29%0Asc.pp.neighbors%28adata%2C%20n_neighbors%3D10%2C%20n_pcs%3D40%29%0Asc.tl.umap%28adata%29%0Asc.tl.leiden%28adata%2C%20resolution%3D0.5%29%0A%0A%23%20Identify%20RAB13%2B%20cluster%20by%20marker%20gene%20expression%0Asc.tl.rank_genes_groups%28adata%2C%20%27leiden%27%2C%20method%3D%27t-test%27%29%0A%0A%23%20Visualize%20clusters%20and%20marker%20expression%0Asc.pl.umap%28adata%2C%20color%3D%5B%27leiden%27%2C%20%27RAB13%27%5D%2C%20save%3D%27_bm_clusters.png%27%29%0A%0AThe%20following%20steps%20enable%20further%20investigation%20of%20ligand-receptor%20interactions%20between%20the%20identified%20RAB13%2B%20EC%20and%20MSC%20populations%20using%20existing%20cell%20communication%20analysis%20tools.%0A%0A%23%20Example%3A%20Compute%20cell-cell%20communication%20using%20Squidpy%0Aimport%20squidpy%20as%20sq%0A%0A%23%20Convert%20Scanpy%20AnnData%20to%20Squidpy%20format%20and%20compute%20neighbors%0Asq.gr.spatial_neighbors%28adata%2C%20coord_type%3D%27grid%27%29%0Asq.gr.nhood_enrichment%28adata%2C%20cluster_key%3D%27leiden%27%29%0A%0A%23%20Plot%20enrichment%20results%0Asq.pl.nhood_enrichment%28adata%2C%20key%3D%27nhood_enrichment%27%2C%20cmap%3D%27viridis%27%2C%20save%3D%27_nhood_enrichment.png%27%29%0A%0AThese%20analyses%20help%20delineate%20the%20molecular%20interactions%20and%20networks%20that%20underpin%20age-related%20alterations%20in%20the%20bone%20marrow%20microenvironment.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Single-cell%20transcriptional%20profiling%20reveals%20a%20novel%20RAB13%2B%20endothelial%20subpopulation%20and%20profibrotic%20mesenchymal%20cells%20in%20the%20aged%20human%20bone%20marrow)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "The code processes scRNA-seq data from bone marrow samples to identify and visualize different endothelial and mesenchymal subpopulations and their communication networks.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-26", "tags": ["RAB13 endothelial cells aged", "biology", "bioinformatics"]}}