{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "We begin by loading the protein sequences of the 62 candidate genes from Fraxinus species. The goal is to calculate and visualize the pairwise sequence identity, which helps in identifying convergent mutations."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import numpy as np\nimport pandas as pd\nfrom Bio import SeqIO\n\n# Load candidate gene sequences from a FASTA file\nsequences = list(SeqIO.parse('candidate_genes.fasta', 'fasta'))\n\n# Initialize an empty similarity matrix\ndef pairwise_identity(seq1, seq2):\n    matches = sum(a == b for a, b in zip(seq1, seq2))\n    return matches / min(len(seq1), len(seq2))\n\nn = len(sequences)\nsimilarity_matrix = np.zeros((n, n))\nfor i in range(n):\n    for j in range(i, n):\n        sim = pairwise_identity(str(sequences[i].seq), str(sequences[j].seq))\n        similarity_matrix[i, j] = sim\n        similarity_matrix[j, i] = sim\n\n# Create a DataFrame with sequence IDs as index and column names\nids = [s.id for s in sequences]\ndf = pd.DataFrame(similarity_matrix, index=ids, columns=ids)\ndf.to_csv('similarity_matrix.csv')"}, {"cell_type": "markdown", "metadata": {}, "source": "Next, we use seaborn and matplotlib to generate a heatmap for visualizing the similarity matrix, which can offer insights into convergence patterns."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import seaborn as sns\nimport matplotlib.pyplot as plt\n\n# Load the similarity matrix\ndf = pd.read_csv('similarity_matrix.csv', index_col=0)\n\nplt.figure(figsize=(10,8))\nax = sns.heatmap(df, cmap='viridis', annot=True, fmt='.2f')\nplt.title('Pairwise Sequence Identity of Candidate Genes')\nplt.savefig('heatmap.png')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "This notebook provides a straightforward pipeline for assessing molecular convergence through sequence similarity analysis among candidate genes."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# End of analysis notebook"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20The%20code%20performs%20pairwise%20sequence%20identity%20analysis%20and%20generates%20a%20heatmap%20from%20candidate%20gene%20protein%20sequences%20to%20visually%20assess%20convergence.%0A%0AEnhance%20the%20pipeline%20by%20integrating%20maximum%20likelihood%20based%20models%20%28e.g.%2C%20using%20PAML%29%20for%20a%20more%20robust%20analysis%20of%20convergent%20evolution.%0A%0AMolecular%20convergence%20analyses%20candidate%20genes%20ash%20dieback%20susceptibility%0A%0AWe%20begin%20by%20loading%20the%20protein%20sequences%20of%20the%2062%20candidate%20genes%20from%20Fraxinus%20species.%20The%20goal%20is%20to%20calculate%20and%20visualize%20the%20pairwise%20sequence%20identity%2C%20which%20helps%20in%20identifying%20convergent%20mutations.%0A%0Aimport%20numpy%20as%20np%0Aimport%20pandas%20as%20pd%0Afrom%20Bio%20import%20SeqIO%0A%0A%23%20Load%20candidate%20gene%20sequences%20from%20a%20FASTA%20file%0Asequences%20%3D%20list%28SeqIO.parse%28%27candidate_genes.fasta%27%2C%20%27fasta%27%29%29%0A%0A%23%20Initialize%20an%20empty%20similarity%20matrix%0Adef%20pairwise_identity%28seq1%2C%20seq2%29%3A%0A%20%20%20%20matches%20%3D%20sum%28a%20%3D%3D%20b%20for%20a%2C%20b%20in%20zip%28seq1%2C%20seq2%29%29%0A%20%20%20%20return%20matches%20%2F%20min%28len%28seq1%29%2C%20len%28seq2%29%29%0A%0An%20%3D%20len%28sequences%29%0Asimilarity_matrix%20%3D%20np.zeros%28%28n%2C%20n%29%29%0Afor%20i%20in%20range%28n%29%3A%0A%20%20%20%20for%20j%20in%20range%28i%2C%20n%29%3A%0A%20%20%20%20%20%20%20%20sim%20%3D%20pairwise_identity%28str%28sequences%5Bi%5D.seq%29%2C%20str%28sequences%5Bj%5D.seq%29%29%0A%20%20%20%20%20%20%20%20similarity_matrix%5Bi%2C%20j%5D%20%3D%20sim%0A%20%20%20%20%20%20%20%20similarity_matrix%5Bj%2C%20i%5D%20%3D%20sim%0A%0A%23%20Create%20a%20DataFrame%20with%20sequence%20IDs%20as%20index%20and%20column%20names%0Aids%20%3D%20%5Bs.id%20for%20s%20in%20sequences%5D%0Adf%20%3D%20pd.DataFrame%28similarity_matrix%2C%20index%3Dids%2C%20columns%3Dids%29%0Adf.to_csv%28%27similarity_matrix.csv%27%29%0A%0ANext%2C%20we%20use%20seaborn%20and%20matplotlib%20to%20generate%20a%20heatmap%20for%20visualizing%20the%20similarity%20matrix%2C%20which%20can%20offer%20insights%20into%20convergence%20patterns.%0A%0Aimport%20seaborn%20as%20sns%0Aimport%20matplotlib.pyplot%20as%20plt%0A%0A%23%20Load%20the%20similarity%20matrix%0Adf%20%3D%20pd.read_csv%28%27similarity_matrix.csv%27%2C%20index_col%3D0%29%0A%0Aplt.figure%28figsize%3D%2810%2C8%29%29%0Aax%20%3D%20sns.heatmap%28df%2C%20cmap%3D%27viridis%27%2C%20annot%3DTrue%2C%20fmt%3D%27.2f%27%29%0Aplt.title%28%27Pairwise%20Sequence%20Identity%20of%20Candidate%20Genes%27%29%0Aplt.savefig%28%27heatmap.png%27%29%0Aplt.show%28%29%0A%0AThis%20notebook%20provides%20a%20straightforward%20pipeline%20for%20assessing%20molecular%20convergence%20through%20sequence%20similarity%20analysis%20among%20candidate%20genes.%0A%0A%23%20End%20of%20analysis%20notebook%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Molecular%20convergence%20analyses%20identify%20candidate%20genes%20for%20low%20susceptibility%20to%20the%20ash%20dieback%20pathogen)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "The code performs pairwise sequence identity analysis and generates a heatmap from candidate gene protein sequences to visually assess convergence.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-16", "tags": ["candidate genes ash dieback", "biology", "bioinformatics"]}}