{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "Analysis of single-cell RNA sequencing data to identify differential gene expression between tTreg and pTreg populations."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nadata = sc.read_h5ad('VAT_Treg_cells.h5ad')\nsc.pp.normalize_total(adata, target_sum=1e4)\nsc.pp.log1p(adata)\nsc.tl.pca(adata)\nsc.pl.pca(adata, color='subset')\nsc.tl.umap(adata)\nsc.pl.umap(adata, color=['subset', 'Foxp3', 'IFN_gamma'])"}, {"cell_type": "markdown", "metadata": {}, "source": "This notebook provides a pipeline to visualize and interpret the shifts in Treg cell profiles with obesity."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Further comparative analysis\nsc.tl.rank_genes_groups(adata, 'subset', method='wilcoxon')\nsc.pl.rank_genes_groups(adata, n_genes=20, sharey=False)"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20The%20code%20integrates%20scRNA-seq%20datasets%20to%20distinguish%20Treg%20cell%20subsets%20based%20on%20their%20transcriptomic%20profiles%2C%20aiding%20in%20determining%20functional%20shifts%20in%20obesity.%0A%0AIncorporate%20integration%20with%20TCR%20sequencing%20data%20and%20include%20batch-correction%20methods%20for%20multi-timepoint%20datasets.%0A%0AFunctional%20dichotomy%20of%20Foxp3%20Treg%20cell%20subsets%20in%20visceral%20adipose%20tissue%20of%20lean%20vs%20obese%20mice%0A%0AAnalysis%20of%20single-cell%20RNA%20sequencing%20data%20to%20identify%20differential%20gene%20expression%20between%20tTreg%20and%20pTreg%20populations.%0A%0Aimport%20scanpy%20as%20sc%0Aadata%20%3D%20sc.read_h5ad%28%27VAT_Treg_cells.h5ad%27%29%0Asc.pp.normalize_total%28adata%2C%20target_sum%3D1e4%29%0Asc.pp.log1p%28adata%29%0Asc.tl.pca%28adata%29%0Asc.pl.pca%28adata%2C%20color%3D%27subset%27%29%0Asc.tl.umap%28adata%29%0Asc.pl.umap%28adata%2C%20color%3D%5B%27subset%27%2C%20%27Foxp3%27%2C%20%27IFN_gamma%27%5D%29%0A%0AThis%20notebook%20provides%20a%20pipeline%20to%20visualize%20and%20interpret%20the%20shifts%20in%20Treg%20cell%20profiles%20with%20obesity.%0A%0A%23%20Further%20comparative%20analysis%0Asc.tl.rank_genes_groups%28adata%2C%20%27subset%27%2C%20method%3D%27wilcoxon%27%29%0Asc.pl.rank_genes_groups%28adata%2C%20n_genes%3D20%2C%20sharey%3DFalse%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Functional%20Dichotomy%20of%20Developmental%20Foxp3%20Treg%20Cell%20Subsets%20in%20the%20Visceral%20Adipose%20Tissue%20of%20Lean%20and%20Obese%20Mice)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "The code integrates scRNA-seq datasets to distinguish Treg cell subsets based on their transcriptomic profiles, aiding in determining functional shifts in obesity.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-19", "tags": ["Foxp3 Treg cell subsets", "biology", "bioinformatics"]}}