{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "This section describes the process of obtaining sequences from UniProt and performing multiple sequence alignment using Clustal Omega to identify conserved regions within the LOV and bZIP domains."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import requests\nfrom Bio import SeqIO\nfrom Bio.Align.Applications import ClustalOmegaCommandline\nimport os\n\n# Download sequences from UniProt using their IDs\nuniprot_ids = ['A8QW55']  # example ID for Aureochrome1\nsequences = []\nfor uid in uniprot_ids:\n    url = f'https://www.uniprot.org/uniprot/{uid}.fasta'\n    response = requests.get(url)\n    sequences.append(response.text)\n\n# Save sequences to file\nwith open('sequences.fasta', 'w') as f:\n    for seq in sequences:\n        f.write(seq)\n\n# Run Clustal Omega alignment using a locally installed binary\nclustalomega_cline = ClustalOmegaCommandline(infile='sequences.fasta', outfile='aligned.fasta', verbose=True, auto=True)\nstdout, stderr = clustalomega_cline()\n\n# Parse and display the alignment\nfrom Bio import AlignIO\nalignment = AlignIO.read('aligned.fasta', 'fasta')\nprint(alignment)"}, {"cell_type": "markdown", "metadata": {}, "source": "The code above is useful for identifying conserved residues that are critical for domain function when designing synthetic linkers."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20The%20Python%20code%20downloads%20and%20analyzes%20domain%20sequence%20alignments%20to%20identify%20conserved%20residues%2C%20aiding%20in%20the%20design%20of%20synthetic%20linkers%20for%20photoreceptor%20engineering.%0A%0AInclude%20error%20handling%2C%20integrate%20visualization%20using%20Plotly%2C%20and%20expand%20dataset%20to%20include%20all%20Aureochrome%20variants.%0A%0AModularity%20of%20Thalassiosira%20oceanica%20Aureochrome1%20photoreceptor%20transcription%20factor%0A%0AThis%20section%20describes%20the%20process%20of%20obtaining%20sequences%20from%20UniProt%20and%20performing%20multiple%20sequence%20alignment%20using%20Clustal%20Omega%20to%20identify%20conserved%20regions%20within%20the%20LOV%20and%20bZIP%20domains.%0A%0Aimport%20requests%0Afrom%20Bio%20import%20SeqIO%0Afrom%20Bio.Align.Applications%20import%20ClustalOmegaCommandline%0Aimport%20os%0A%0A%23%20Download%20sequences%20from%20UniProt%20using%20their%20IDs%0Auniprot_ids%20%3D%20%5B%27A8QW55%27%5D%20%20%23%20example%20ID%20for%20Aureochrome1%0Asequences%20%3D%20%5B%5D%0Afor%20uid%20in%20uniprot_ids%3A%0A%20%20%20%20url%20%3D%20f%27https%3A%2F%2Fwww.uniprot.org%2Funiprot%2F%7Buid%7D.fasta%27%0A%20%20%20%20response%20%3D%20requests.get%28url%29%0A%20%20%20%20sequences.append%28response.text%29%0A%0A%23%20Save%20sequences%20to%20file%0Awith%20open%28%27sequences.fasta%27%2C%20%27w%27%29%20as%20f%3A%0A%20%20%20%20for%20seq%20in%20sequences%3A%0A%20%20%20%20%20%20%20%20f.write%28seq%29%0A%0A%23%20Run%20Clustal%20Omega%20alignment%20using%20a%20locally%20installed%20binary%0Aclustalomega_cline%20%3D%20ClustalOmegaCommandline%28infile%3D%27sequences.fasta%27%2C%20outfile%3D%27aligned.fasta%27%2C%20verbose%3DTrue%2C%20auto%3DTrue%29%0Astdout%2C%20stderr%20%3D%20clustalomega_cline%28%29%0A%0A%23%20Parse%20and%20display%20the%20alignment%0Afrom%20Bio%20import%20AlignIO%0Aalignment%20%3D%20AlignIO.read%28%27aligned.fasta%27%2C%20%27fasta%27%29%0Aprint%28alignment%29%0A%0AThe%20code%20above%20is%20useful%20for%20identifying%20conserved%20residues%20that%20are%20critical%20for%20domain%20function%20when%20designing%20synthetic%20linkers.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Exploring%20the%20Modularity%20ofThalassiosira%20oceanicaAureochrome1%20Photoreceptor-Transcription%20Factor%3A%20Characterization%20and%20Component%20Engineering)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "The Python code downloads and analyzes domain sequence alignments to identify conserved residues, aiding in the design of synthetic linkers for photoreceptor engineering.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-19", "tags": ["Thalassiosira oceanica review", "biology", "bioinformatics"]}}