{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "We begin by loading the necessary libraries and downloading the genomic datasets of CCr and Gpe as provided by the study metadata."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\nfrom Bio import SeqIO\n\n# Assume genomes are available as FASTA files from provided accession numbers\nccr_genome = list(SeqIO.parse('CCr_genome.fasta', 'fasta'))\ngpe_genome = list(SeqIO.parse('Gpe_genome.fasta', 'fasta'))\n\n# Load KEGG pathway mapping for biosynthetic genes\nkegg_data = pd.read_csv('kegg_biosynthesis_pathways.csv')\n\nprint('CCr and Gpe genomes loaded. KEGG data loaded.')"}, {"cell_type": "markdown", "metadata": {}, "source": "Next, we perform a simplistic gene presence/absence analysis for amino acid biosynthesis pathways in both genomes."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "def analyze_pathways(genome, kegg_df):\n    genes_in_genome = {record.id for record in genome}\n    pathway_presence = kegg_df.copy()\n    pathway_presence['Present'] = pathway_presence['Gene'].apply(lambda gene: gene in genes_in_genome)\n    return pathway_presence\n\nccr_pathways = analyze_pathways(ccr_genome, kegg_data)\ngpe_pathways = analyze_pathways(gpe_genome, kegg_data)\n\nprint('Pathway analysis complete.')\nprint(ccr_pathways.head())\nprint(gpe_pathways.head())"}, {"cell_type": "markdown", "metadata": {}, "source": "Finally, we visualize the differences in pathway completeness between CCr and Gpe to support the hypothesis of complementary biosynthetic roles."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import plotly.express as px\n\n# Combine data for visualization\nccr_pathways['Symbiont'] = 'CCr'\ngpe_pathways['Symbiont'] = 'Gpe'\ncombined = pd.concat([ccr_pathways, gpe_pathways], ignore_index=True)\n\nfig = px.bar(combined, x='Pathway', y='Present', color='Symbiont', barmode='group',\n             title='Comparative Biosynthetic Pathway Presence in CCr and Gpe')\nfig.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "This notebook thus provides a framework for examining the genomic basis for biosynthetic division of labor between symbionts."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20genomic%20datasets%20and%20analyzes%20amino%20acid%20biosynthesis%20pathways%20in%20symbiont%20genomes%20using%20pathway%20enrichment%20analysis.%0A%0AInclude%20functional%20annotation%20data%20and%20experimental%20proteomics%20datasets%20to%20enhance%20pathway%20validation.%0A%0ABacterial%20endosymbionts%20genomes%20Trioza%20apicalis%20biosynthetic%20capabilities%0A%0AWe%20begin%20by%20loading%20the%20necessary%20libraries%20and%20downloading%20the%20genomic%20datasets%20of%20CCr%20and%20Gpe%20as%20provided%20by%20the%20study%20metadata.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0Afrom%20Bio%20import%20SeqIO%0A%0A%23%20Assume%20genomes%20are%20available%20as%20FASTA%20files%20from%20provided%20accession%20numbers%0Accr_genome%20%3D%20list%28SeqIO.parse%28%27CCr_genome.fasta%27%2C%20%27fasta%27%29%29%0Agpe_genome%20%3D%20list%28SeqIO.parse%28%27Gpe_genome.fasta%27%2C%20%27fasta%27%29%29%0A%0A%23%20Load%20KEGG%20pathway%20mapping%20for%20biosynthetic%20genes%0Akegg_data%20%3D%20pd.read_csv%28%27kegg_biosynthesis_pathways.csv%27%29%0A%0Aprint%28%27CCr%20and%20Gpe%20genomes%20loaded.%20KEGG%20data%20loaded.%27%29%0A%0ANext%2C%20we%20perform%20a%20simplistic%20gene%20presence%2Fabsence%20analysis%20for%20amino%20acid%20biosynthesis%20pathways%20in%20both%20genomes.%0A%0Adef%20analyze_pathways%28genome%2C%20kegg_df%29%3A%0A%20%20%20%20genes_in_genome%20%3D%20%7Brecord.id%20for%20record%20in%20genome%7D%0A%20%20%20%20pathway_presence%20%3D%20kegg_df.copy%28%29%0A%20%20%20%20pathway_presence%5B%27Present%27%5D%20%3D%20pathway_presence%5B%27Gene%27%5D.apply%28lambda%20gene%3A%20gene%20in%20genes_in_genome%29%0A%20%20%20%20return%20pathway_presence%0A%0Accr_pathways%20%3D%20analyze_pathways%28ccr_genome%2C%20kegg_data%29%0Agpe_pathways%20%3D%20analyze_pathways%28gpe_genome%2C%20kegg_data%29%0A%0Aprint%28%27Pathway%20analysis%20complete.%27%29%0Aprint%28ccr_pathways.head%28%29%29%0Aprint%28gpe_pathways.head%28%29%29%0A%0AFinally%2C%20we%20visualize%20the%20differences%20in%20pathway%20completeness%20between%20CCr%20and%20Gpe%20to%20support%20the%20hypothesis%20of%20complementary%20biosynthetic%20roles.%0A%0Aimport%20plotly.express%20as%20px%0A%0A%23%20Combine%20data%20for%20visualization%0Accr_pathways%5B%27Symbiont%27%5D%20%3D%20%27CCr%27%0Agpe_pathways%5B%27Symbiont%27%5D%20%3D%20%27Gpe%27%0Acombined%20%3D%20pd.concat%28%5Bccr_pathways%2C%20gpe_pathways%5D%2C%20ignore_index%3DTrue%29%0A%0Afig%20%3D%20px.bar%28combined%2C%20x%3D%27Pathway%27%2C%20y%3D%27Present%27%2C%20color%3D%27Symbiont%27%2C%20barmode%3D%27group%27%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20title%3D%27Comparative%20Biosynthetic%20Pathway%20Presence%20in%20CCr%20and%20Gpe%27%29%0Afig.show%28%29%0A%0AThis%20notebook%20thus%20provides%20a%20framework%20for%20examining%20the%20genomic%20basis%20for%20biosynthetic%20division%20of%20labor%20between%20symbionts.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Genomes%20of%20the%20Bacterial%20Endosymbionts%20of%20Carrot%20Psyllid%20Trioza%20apicalis%20Suggest%20Complementary%20Biosynthetic%20Capabilities)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads genomic datasets and analyzes amino acid biosynthesis pathways in symbiont genomes using pathway enrichment analysis.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-03", "tags": ["Trioza apicalis endosymbionts", "biology", "bioinformatics"]}}