{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "Below is a step-by-step Jupyter notebook outline to analyze the Pf liver stage RNA-seq data. This notebook leverages scanpy for differential expression analysis and visualization."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nimport pandas as pd\nimport matplotlib.pyplot as plt\n\n# Download and read the dataset from GEO GSE220039 (assuming the data are already processed into an AnnData object)\nadata = sc.read_h5ad('GSE220039_processed.h5ad')\n\n# Quality Control and normalization\nsc.pp.filter_cells(adata, min_genes=200)\nsc.pp.filter_genes(adata, min_cells=3)\nsc.pp.normalize_total(adata, target_sum=1e4)\nsc.pp.log1p(adata)\n\n# Identify highly variable genes\nsc.pp.highly_variable_genes(adata, min_mean=0.0125, max_mean=3, min_disp=0.5)\nadata = adata[:, adata.var.highly_variable]\n\n# Scale the data and perform PCA\nsc.pp.scale(adata, max_value=10)\nsc.tl.pca(adata, svd_solver='arpack')\n\n# Compute neighborhood graph\nsc.pp.neighbors(adata, n_neighbors=10, n_pcs=40)\nsc.tl.umap(adata)\n\n# Differential Expression analysis between early and late liver stage groups\n# Assuming 'stage' is a categorical variable in adata.obs with values 'early' and 'late'\nsc.tl.rank_genes_groups(adata, 'stage', method='t-test')\n\n# Plot the top markers\nsc.pl.rank_genes_groups(adata, n_genes=20, sharey=False)\n\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "The above code performs basic quality control and normalization, followed by PCA, UMAP visualization, and differential gene expression analysis to identify key genes that distinguish early versus late liver stage development in Pf."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Further visualization: Heatmap of top differentially expressed genes\nimport seaborn as sns\n\ntop_genes = adata.uns['rank_genes_groups']['names'][0][:20]\ndata_top = adata[:, top_genes].to_df()\n\n# Create a heatmap\nplt.figure(figsize=(10,8))\nsns.heatmap(data_top.T, cmap='viridis')\nplt.title('Top Differentially Expressed Genes in Pf Liver Stages')\nplt.xlabel('Cells')\nplt.ylabel('Genes')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "This notebook can serve as a foundation for deeper bioinformatics analysis. Adjust parameters and include additional metadata (such as parasitic stage markers) to refine the analysis further."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Save the results of differential expression for further exploration\ndeg_results = pd.DataFrame(adata.uns['rank_genes_groups']['names']).T\n\n# Save to CSV\ndeg_results.to_csv('Pf_liver_stage_DEG_results.csv', index=False)\nprint('Differential expression results saved to Pf_liver_stage_DEG_results.csv')"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20Pf%20liver%20stage%20RNA-seq%20data%20from%20GEO%20GSE220039%2C%20performs%20quality%20control%2C%20normalization%2C%20and%20identifies%20differentially%20expressed%20genes%20using%20scanpy.%0A%0AInclude%20specific%20sample%20metadata%20and%20replicate%20information%20to%20refine%20clustering%20and%20differential%20analysis.%0A%0AGenome-wide%20gene%20expression%20malaria%20liver%20stage%20humanized%20mice%0A%0ABelow%20is%20a%20step-by-step%20Jupyter%20notebook%20outline%20to%20analyze%20the%20Pf%20liver%20stage%20RNA-seq%20data.%20This%20notebook%20leverages%20scanpy%20for%20differential%20expression%20analysis%20and%20visualization.%0A%0Aimport%20scanpy%20as%20sc%0Aimport%20pandas%20as%20pd%0Aimport%20matplotlib.pyplot%20as%20plt%0A%0A%23%20Download%20and%20read%20the%20dataset%20from%20GEO%20GSE220039%20%28assuming%20the%20data%20are%20already%20processed%20into%20an%20AnnData%20object%29%0Aadata%20%3D%20sc.read_h5ad%28%27GSE220039_processed.h5ad%27%29%0A%0A%23%20Quality%20Control%20and%20normalization%0Asc.pp.filter_cells%28adata%2C%20min_genes%3D200%29%0Asc.pp.filter_genes%28adata%2C%20min_cells%3D3%29%0Asc.pp.normalize_total%28adata%2C%20target_sum%3D1e4%29%0Asc.pp.log1p%28adata%29%0A%0A%23%20Identify%20highly%20variable%20genes%0Asc.pp.highly_variable_genes%28adata%2C%20min_mean%3D0.0125%2C%20max_mean%3D3%2C%20min_disp%3D0.5%29%0Aadata%20%3D%20adata%5B%3A%2C%20adata.var.highly_variable%5D%0A%0A%23%20Scale%20the%20data%20and%20perform%20PCA%0Asc.pp.scale%28adata%2C%20max_value%3D10%29%0Asc.tl.pca%28adata%2C%20svd_solver%3D%27arpack%27%29%0A%0A%23%20Compute%20neighborhood%20graph%0Asc.pp.neighbors%28adata%2C%20n_neighbors%3D10%2C%20n_pcs%3D40%29%0Asc.tl.umap%28adata%29%0A%0A%23%20Differential%20Expression%20analysis%20between%20early%20and%20late%20liver%20stage%20groups%0A%23%20Assuming%20%27stage%27%20is%20a%20categorical%20variable%20in%20adata.obs%20with%20values%20%27early%27%20and%20%27late%27%0Asc.tl.rank_genes_groups%28adata%2C%20%27stage%27%2C%20method%3D%27t-test%27%29%0A%0A%23%20Plot%20the%20top%20markers%0Asc.pl.rank_genes_groups%28adata%2C%20n_genes%3D20%2C%20sharey%3DFalse%29%0A%0Aplt.show%28%29%0A%0AThe%20above%20code%20performs%20basic%20quality%20control%20and%20normalization%2C%20followed%20by%20PCA%2C%20UMAP%20visualization%2C%20and%20differential%20gene%20expression%20analysis%20to%20identify%20key%20genes%20that%20distinguish%20early%20versus%20late%20liver%20stage%20development%20in%20Pf.%0A%0A%23%20Further%20visualization%3A%20Heatmap%20of%20top%20differentially%20expressed%20genes%0Aimport%20seaborn%20as%20sns%0A%0Atop_genes%20%3D%20adata.uns%5B%27rank_genes_groups%27%5D%5B%27names%27%5D%5B0%5D%5B%3A20%5D%0Adata_top%20%3D%20adata%5B%3A%2C%20top_genes%5D.to_df%28%29%0A%0A%23%20Create%20a%20heatmap%0Aplt.figure%28figsize%3D%2810%2C8%29%29%0Asns.heatmap%28data_top.T%2C%20cmap%3D%27viridis%27%29%0Aplt.title%28%27Top%20Differentially%20Expressed%20Genes%20in%20Pf%20Liver%20Stages%27%29%0Aplt.xlabel%28%27Cells%27%29%0Aplt.ylabel%28%27Genes%27%29%0Aplt.show%28%29%0A%0AThis%20notebook%20can%20serve%20as%20a%20foundation%20for%20deeper%20bioinformatics%20analysis.%20Adjust%20parameters%20and%20include%20additional%20metadata%20%28such%20as%20parasitic%20stage%20markers%29%20to%20refine%20the%20analysis%20further.%0A%0A%23%20Save%20the%20results%20of%20differential%20expression%20for%20further%20exploration%0Adeg_results%20%3D%20pd.DataFrame%28adata.uns%5B%27rank_genes_groups%27%5D%5B%27names%27%5D%29.T%0A%0A%23%20Save%20to%20CSV%0Adeg_results.to_csv%28%27Pf_liver_stage_DEG_results.csv%27%2C%20index%3DFalse%29%0Aprint%28%27Differential%20expression%20results%20saved%20to%20Pf_liver_stage_DEG_results.csv%27%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Genome-wide%20gene%20expression%20profiles%20throughout%20human%20malaria%20parasite%20liver%20stage%20development%20in%20humanized%20mice)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads Pf liver stage RNA-seq data from GEO GSE220039, performs quality control, normalization, and identifies differentially expressed genes using scanpy.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-12", "tags": ["gene expression malaria liver stage", "biology", "bioinformatics"]}}