{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "Below, we describe the process of downloading real spatial transcriptomics datasets from the study and performing preprocessing and visualization using Scanpy and Squidpy. This enables an in-depth exploration of gene expression patterns across different infection groups."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nimport squidpy as sq\n\n# Download spatial transcriptomics data (real dataset accession not provided, use placeholder)\nadata = sc.read_visium(path='path_to_dataset')\n\n# Preprocess the data\nsc.pp.filter_cells(adata, min_genes=200)\nsc.pp.filter_genes(adata, min_cells=3)\nsc.pp.normalize_total(adata, target_sum=1e4)\nsc.pp.log1p(adata)\n\n# Perform clustering\nsc.tl.pca(adata, svd_solver='arpack')\nsc.pp.neighbors(adata, n_neighbors=10, n_pcs=40)\nsc.tl.umap(adata)\nsc.tl.leiden(adata, resolution=0.5)\n\n# Plot spatial gene expression for a key marker (e.g., CD4)\nsc.pl.spatial(adata, color='CD4', spot_size=1.5, title='Spatial Expression of CD4')"}, {"cell_type": "markdown", "metadata": {}, "source": "The above code demonstrates data loading, filtering, normalization, clustering, and spatial visualization of CD4 expression. This tailored pipeline can help identify spatial patterns of immune cell markers across different lung regions in various infection conditions."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Further analysis: Differential gene expression between clusters\nsc.tl.rank_genes_groups(adata, 'leiden', method='t-test')\nsc.pl.rank_genes_groups(adata, n_genes=20, sharey=False, title='Differential Expression per Cluster')"}, {"cell_type": "markdown", "metadata": {}, "source": "This additional step identifies key differentially expressed genes among spatial clusters, which can be correlated with the transcriptional signatures observed in coinfection."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Save processed data for further analysis\nadata.write('processed_spatial_data.h5ad')\nprint('Data processing complete and saved.')"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20and%20processes%20spatial%20transcriptomics%20data%20to%20visualize%20gene%20expression%20differences%20across%20lung%20regions%20in%20infection%20groups.%0A%0AInclude%20direct%20links%20to%20validated%20spatial%20transcriptomics%20datasets%20and%20integrate%20multi-modal%20data%20analysis%20for%20enhanced%20insights.%0A%0ASpatial%20transcriptomics%20HIV%20tuberculosis%20coinfection%20immune%20response%20morphological%20changes%0A%0ABelow%2C%20we%20describe%20the%20process%20of%20downloading%20real%20spatial%20transcriptomics%20datasets%20from%20the%20study%20and%20performing%20preprocessing%20and%20visualization%20using%20Scanpy%20and%20Squidpy.%20This%20enables%20an%20in-depth%20exploration%20of%20gene%20expression%20patterns%20across%20different%20infection%20groups.%0A%0Aimport%20scanpy%20as%20sc%0Aimport%20squidpy%20as%20sq%0A%0A%23%20Download%20spatial%20transcriptomics%20data%20%28real%20dataset%20accession%20not%20provided%2C%20use%20placeholder%29%0Aadata%20%3D%20sc.read_visium%28path%3D%27path_to_dataset%27%29%0A%0A%23%20Preprocess%20the%20data%0Asc.pp.filter_cells%28adata%2C%20min_genes%3D200%29%0Asc.pp.filter_genes%28adata%2C%20min_cells%3D3%29%0Asc.pp.normalize_total%28adata%2C%20target_sum%3D1e4%29%0Asc.pp.log1p%28adata%29%0A%0A%23%20Perform%20clustering%0Asc.tl.pca%28adata%2C%20svd_solver%3D%27arpack%27%29%0Asc.pp.neighbors%28adata%2C%20n_neighbors%3D10%2C%20n_pcs%3D40%29%0Asc.tl.umap%28adata%29%0Asc.tl.leiden%28adata%2C%20resolution%3D0.5%29%0A%0A%23%20Plot%20spatial%20gene%20expression%20for%20a%20key%20marker%20%28e.g.%2C%20CD4%29%0Asc.pl.spatial%28adata%2C%20color%3D%27CD4%27%2C%20spot_size%3D1.5%2C%20title%3D%27Spatial%20Expression%20of%20CD4%27%29%0A%0AThe%20above%20code%20demonstrates%20data%20loading%2C%20filtering%2C%20normalization%2C%20clustering%2C%20and%20spatial%20visualization%20of%20CD4%20expression.%20This%20tailored%20pipeline%20can%20help%20identify%20spatial%20patterns%20of%20immune%20cell%20markers%20across%20different%20lung%20regions%20in%20various%20infection%20conditions.%0A%0A%23%20Further%20analysis%3A%20Differential%20gene%20expression%20between%20clusters%0Asc.tl.rank_genes_groups%28adata%2C%20%27leiden%27%2C%20method%3D%27t-test%27%29%0Asc.pl.rank_genes_groups%28adata%2C%20n_genes%3D20%2C%20sharey%3DFalse%2C%20title%3D%27Differential%20Expression%20per%20Cluster%27%29%0A%0AThis%20additional%20step%20identifies%20key%20differentially%20expressed%20genes%20among%20spatial%20clusters%2C%20which%20can%20be%20correlated%20with%20the%20transcriptional%20signatures%20observed%20in%20coinfection.%0A%0A%23%20Save%20processed%20data%20for%20further%20analysis%0Aadata.write%28%27processed_spatial_data.h5ad%27%29%0Aprint%28%27Data%20processing%20complete%20and%20saved.%27%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Spatial%20transcriptomic%20analysis%20of%20HIV%20and%20tuberculosis%20coinfection%20in%20a%20humanized%20mouse%20model%20reveals%20specific%20transcription%20patterns%2C%20immune%20responses%20and%20early%20morphological%20alteration%20signaling)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads and processes spatial transcriptomics data to visualize gene expression differences across lung regions in infection groups.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-04", "tags": ["spatial transcriptomics HIV TB", "biology", "bioinformatics"]}}