{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "### Step 1: Import Libraries\nImport necessary libraries for data analysis and visualization."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\nimport seaborn as sns\nimport matplotlib.pyplot as plt"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 2: Load Data\nLoad the gene expression dataset relevant to tuberculosis."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "data = pd.read_csv('path_to_hc_tNGS_data.csv')\n# Display the first few rows of the dataset\nprint(data.head())"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 3: Data Preprocessing\nPreprocess the data to handle missing values and normalize the expression levels."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "data.fillna(0, inplace=True)\n# Normalize the data\nnormalized_data = (data - data.mean()) / data.std()"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 4: Analysis\nPerform statistical analysis to identify significant biomarkers."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "from scipy.stats import ttest_ind\n# Assuming 'ATB' and 'LTBI' are the two groups in the dataset\natb_data = normalized_data[normalized_data['group'] == 'ATB']\nltbi_data = normalized_data[normalized_data['group'] == 'LTBI']\nstat, p_value = ttest_ind(atb_data, ltbi_data)\nprint('P-value:', p_value)"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20analyzes%20gene%20expression%20data%20to%20identify%20biomarkers%20that%20differentiate%20ATB%20from%20LTBI%20using%20hc-tNGS%20data.%0A%0AInclude%20additional%20datasets%20for%20more%20robust%20analysis%20and%20validation%20of%20biomarkers.%0A%0Ahc-tNGS%20differentiate%20active%20latent%20tuberculosis%0A%0A%23%23%23%20Step%201%3A%20Import%20Libraries%0AImport%20necessary%20libraries%20for%20data%20analysis%20and%20visualization.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0Aimport%20seaborn%20as%20sns%0Aimport%20matplotlib.pyplot%20as%20plt%0A%0A%23%23%23%20Step%202%3A%20Load%20Data%0ALoad%20the%20gene%20expression%20dataset%20relevant%20to%20tuberculosis.%0A%0Adata%20%3D%20pd.read_csv%28%27path_to_hc_tNGS_data.csv%27%29%0A%23%20Display%20the%20first%20few%20rows%20of%20the%20dataset%0Aprint%28data.head%28%29%29%0A%0A%23%23%23%20Step%203%3A%20Data%20Preprocessing%0APreprocess%20the%20data%20to%20handle%20missing%20values%20and%20normalize%20the%20expression%20levels.%0A%0Adata.fillna%280%2C%20inplace%3DTrue%29%0A%23%20Normalize%20the%20data%0Anormalized_data%20%3D%20%28data%20-%20data.mean%28%29%29%20%2F%20data.std%28%29%0A%0A%23%23%23%20Step%204%3A%20Analysis%0APerform%20statistical%20analysis%20to%20identify%20significant%20biomarkers.%0A%0Afrom%20scipy.stats%20import%20ttest_ind%0A%23%20Assuming%20%27ATB%27%20and%20%27LTBI%27%20are%20the%20two%20groups%20in%20the%20dataset%0Aatb_data%20%3D%20normalized_data%5Bnormalized_data%5B%27group%27%5D%20%3D%3D%20%27ATB%27%5D%0Altbi_data%20%3D%20normalized_data%5Bnormalized_data%5B%27group%27%5D%20%3D%3D%20%27LTBI%27%5D%0Astat%2C%20p_value%20%3D%20ttest_ind%28atb_data%2C%20ltbi_data%29%0Aprint%28%27P-value%3A%27%2C%20p_value%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Potential%20of%20hc-tNGS%20to%20differentiate%20active%20from%20latent%20tuberculosis)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code analyzes gene expression data to identify biomarkers that differentiate ATB from LTBI using hc-tNGS data.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-24", "tags": ["hc-tNGS tuberculosis differentiation", "biology", "bioinformatics"]}}