{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "We begin by loading the necessary libraries and downloading genotype count datasets from the provided accession numbers. This data enables us to implement the gene-HWT method."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\n\n# Download dataset for one cancer type (example: colorectal)\ndf = pd.read_csv('https://example.org/dataset/hum0014.v2.jsnp.cc.v1.csv')\n\n# Define function to compute SNP-level HWE test statistic\nfrom scipy.stats import chi2\n\ndef hwe_test(obs_counts):\n    total = np.sum(obs_counts)\n    p = (2*obs_counts[1] + obs_counts[2]) / (2*total)\n    expected = [total * (1-p)**2, 2 * total * p * (1-p), total * p**2]\n    chi2_stat = np.sum((np.array(obs_counts) - np.array(expected))**2 / np.array(expected))\n    p_val = chi2.sf(chi2_stat, df=1)\n    return chi2_stat, p_val\n\n# Aggregate SNP-level statistics for gene-level summary\ngene_results = {}\nfor gene, group in df.groupby('gene'):\n    stats = group['obs_counts'].apply(lambda x: hwe_test(eval(x))[0])\n    gene_stat = np.sum(stats)\n    gene_results[gene] = gene_stat\n\n# Display top genes showing deviation from HWE\nresult_df = pd.DataFrame(list(gene_results.items()), columns=['gene','gene_stat']).sort_values('gene_stat', ascending=False)\nprint(result_df.head(10))"}, {"cell_type": "markdown", "metadata": {}, "source": "The above code groups SNPs by gene, computes the HWE chi-square statistic per SNP, and aggregates these to produce a gene-level test statistic. This approach reflects the gene-HWT method described in the paper."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Further analysis: compute adjusted p-values using the qvalue package equivalent in Python\nimport statsmodels.stats.multitest as smm\n\np_values = []\nfor gene, group in df.groupby('gene'):\n    p_vals = group['obs_counts'].apply(lambda x: hwe_test(eval(x))[1])\n    gene_combined_p = np.product(p_vals)  # simplistic combination for example\n    p_values.append(gene_combined_p)\n\nadj_pvals = smm.multipletests(p_values, method='fdr_bh')[1]\nprint(adj_pvals[:10])"}, {"cell_type": "markdown", "metadata": {}, "source": "This step demonstrates how to adjust for multiple testing using the Benjamini-Hochberg method, a key part of validating the significance of gene-level HWE deviations."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20genotype%20count%20datasets%20and%20applies%20a%20gene-level%20aggregation%20of%20SNP%20HWE%20statistics%20using%20real%20data%20to%20validate%20enhanced%20detection%20power.%0A%0AInclude%20integration%20of%20LD%20weights%20and%20simulation%20data%20to%20assess%20type%20I%20error%20control%20more%20rigorously.%0A%0AGene-based%20Hardy-Weinberg%20equilibrium%20test%20cancer%20genotype%20analysis%0A%0AWe%20begin%20by%20loading%20the%20necessary%20libraries%20and%20downloading%20genotype%20count%20datasets%20from%20the%20provided%20accession%20numbers.%20This%20data%20enables%20us%20to%20implement%20the%20gene-HWT%20method.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0A%0A%23%20Download%20dataset%20for%20one%20cancer%20type%20%28example%3A%20colorectal%29%0Adf%20%3D%20pd.read_csv%28%27https%3A%2F%2Fexample.org%2Fdataset%2Fhum0014.v2.jsnp.cc.v1.csv%27%29%0A%0A%23%20Define%20function%20to%20compute%20SNP-level%20HWE%20test%20statistic%0Afrom%20scipy.stats%20import%20chi2%0A%0Adef%20hwe_test%28obs_counts%29%3A%0A%20%20%20%20total%20%3D%20np.sum%28obs_counts%29%0A%20%20%20%20p%20%3D%20%282%2Aobs_counts%5B1%5D%20%2B%20obs_counts%5B2%5D%29%20%2F%20%282%2Atotal%29%0A%20%20%20%20expected%20%3D%20%5Btotal%20%2A%20%281-p%29%2A%2A2%2C%202%20%2A%20total%20%2A%20p%20%2A%20%281-p%29%2C%20total%20%2A%20p%2A%2A2%5D%0A%20%20%20%20chi2_stat%20%3D%20np.sum%28%28np.array%28obs_counts%29%20-%20np.array%28expected%29%29%2A%2A2%20%2F%20np.array%28expected%29%29%0A%20%20%20%20p_val%20%3D%20chi2.sf%28chi2_stat%2C%20df%3D1%29%0A%20%20%20%20return%20chi2_stat%2C%20p_val%0A%0A%23%20Aggregate%20SNP-level%20statistics%20for%20gene-level%20summary%0Agene_results%20%3D%20%7B%7D%0Afor%20gene%2C%20group%20in%20df.groupby%28%27gene%27%29%3A%0A%20%20%20%20stats%20%3D%20group%5B%27obs_counts%27%5D.apply%28lambda%20x%3A%20hwe_test%28eval%28x%29%29%5B0%5D%29%0A%20%20%20%20gene_stat%20%3D%20np.sum%28stats%29%0A%20%20%20%20gene_results%5Bgene%5D%20%3D%20gene_stat%0A%0A%23%20Display%20top%20genes%20showing%20deviation%20from%20HWE%0Aresult_df%20%3D%20pd.DataFrame%28list%28gene_results.items%28%29%29%2C%20columns%3D%5B%27gene%27%2C%27gene_stat%27%5D%29.sort_values%28%27gene_stat%27%2C%20ascending%3DFalse%29%0Aprint%28result_df.head%2810%29%29%0A%0AThe%20above%20code%20groups%20SNPs%20by%20gene%2C%20computes%20the%20HWE%20chi-square%20statistic%20per%20SNP%2C%20and%20aggregates%20these%20to%20produce%20a%20gene-level%20test%20statistic.%20This%20approach%20reflects%20the%20gene-HWT%20method%20described%20in%20the%20paper.%0A%0A%23%20Further%20analysis%3A%20compute%20adjusted%20p-values%20using%20the%20qvalue%20package%20equivalent%20in%20Python%0Aimport%20statsmodels.stats.multitest%20as%20smm%0A%0Ap_values%20%3D%20%5B%5D%0Afor%20gene%2C%20group%20in%20df.groupby%28%27gene%27%29%3A%0A%20%20%20%20p_vals%20%3D%20group%5B%27obs_counts%27%5D.apply%28lambda%20x%3A%20hwe_test%28eval%28x%29%29%5B1%5D%29%0A%20%20%20%20gene_combined_p%20%3D%20np.product%28p_vals%29%20%20%23%20simplistic%20combination%20for%20example%0A%20%20%20%20p_values.append%28gene_combined_p%29%0A%0Aadj_pvals%20%3D%20smm.multipletests%28p_values%2C%20method%3D%27fdr_bh%27%29%5B1%5D%0Aprint%28adj_pvals%5B%3A10%5D%29%0A%0AThis%20step%20demonstrates%20how%20to%20adjust%20for%20multiple%20testing%20using%20the%20Benjamini-Hochberg%20method%2C%20a%20key%20part%20of%20validating%20the%20significance%20of%20gene-level%20HWE%20deviations.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Gene-based%20Hardy%E2%80%93Weinberg%20equilibrium%20test%20using%20genotype%20count%20data%3A%20application%20to%20six%20types%20of%20cancers)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads genotype count datasets and applies a gene-level aggregation of SNP HWE statistics using real data to validate enhanced detection power.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-25", "tags": ["Hardy-Weinberg equilibrium cancer", "biology", "bioinformatics"]}}