{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "Below, we load a real scRNA-seq dataset, preprocess it, and apply PCA, t-SNE, UMAP, and a placeholder for scVI to compare their embeddings."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nimport anndata\nimport pandas as pd\n\n# Load a real scRNA-seq dataset (PBMC 3k dataset)\nadata = sc.datasets.pbmc3k()\n\n# Preprocess the data\nsc.pp.normalize_total(adata, target_sum=1e4)\nsc.pp.log1p(adata)\nsc.pp.highly_variable_genes(adata, min_mean=0.0125, max_mean=3, min_disp=0.5)\nadata = adata[:, adata.var.highly_variable]\n\n# Apply PCA\nsc.tl.pca(adata, svd_solver='arpack')\n\n# Apply t-SNE\nsc.tl.tsne(adata, n_pcs=50)\n\n# Compute UMAP\nsc.pp.neighbors(adata, n_neighbors=10, n_pcs=50)\nsc.tl.umap(adata)\n\n# Store embeddings\nembeddings = pd.DataFrame({\n    'pca_1': adata.obsm['X_pca'][:,0],\n    'pca_2': adata.obsm['X_pca'][:,1],\n    'tsne_1': adata.obsm['X_tsne'][:,0],\n    'tsne_2': adata.obsm['X_tsne'][:,1],\n    'umap_1': adata.obsm['X_umap'][:,0],\n    'umap_2': adata.obsm['X_umap'][:,1]\n})\nprint(embeddings.head())\n"}, {"cell_type": "markdown", "metadata": {}, "source": "The above steps preprocess data and compute embeddings using PCA, t-SNE, and UMAP. In a complete pipeline, scVI would be applied similarly after setting up an anndata object for scVI processing. Comparisons among these embeddings can reveal how each method captures potential temporal structures."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import plotly.express as px\n\n# Visualize t-SNE embedding\nfig_tsne = px.scatter(embeddings, x='tsne_1', y='tsne_2', title='t-SNE Embedding')\nfig_tsne.show()\n\n# Visualize UMAP embedding\nfig_umap = px.scatter(embeddings, x='umap_1', y='umap_2', title='UMAP Embedding')\nfig_umap.show()\n"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20processes%20a%20real%20PBMC%20scRNA-seq%20dataset%20using%20multiple%20dimensionality%20reduction%20methods%20and%20provides%20a%20comparative%20visualization%20to%20benchmark%20temporal%20patterns.%0A%0AIntegrate%20real%20scVI%20code%20and%20synthetic%20VAE-generated%20benchmark%20data%20to%20better%20assess%20temporal%20dynamics.%0A%0ADimensionality%20reduction%20discrepancies%20time-series%20single-cell%20RNA-seq%0A%0ABelow%2C%20we%20load%20a%20real%20scRNA-seq%20dataset%2C%20preprocess%20it%2C%20and%20apply%20PCA%2C%20t-SNE%2C%20UMAP%2C%20and%20a%20placeholder%20for%20scVI%20to%20compare%20their%20embeddings.%0A%0Aimport%20scanpy%20as%20sc%0Aimport%20anndata%0Aimport%20pandas%20as%20pd%0A%0A%23%20Load%20a%20real%20scRNA-seq%20dataset%20%28PBMC%203k%20dataset%29%0Aadata%20%3D%20sc.datasets.pbmc3k%28%29%0A%0A%23%20Preprocess%20the%20data%0Asc.pp.normalize_total%28adata%2C%20target_sum%3D1e4%29%0Asc.pp.log1p%28adata%29%0Asc.pp.highly_variable_genes%28adata%2C%20min_mean%3D0.0125%2C%20max_mean%3D3%2C%20min_disp%3D0.5%29%0Aadata%20%3D%20adata%5B%3A%2C%20adata.var.highly_variable%5D%0A%0A%23%20Apply%20PCA%0Asc.tl.pca%28adata%2C%20svd_solver%3D%27arpack%27%29%0A%0A%23%20Apply%20t-SNE%0Asc.tl.tsne%28adata%2C%20n_pcs%3D50%29%0A%0A%23%20Compute%20UMAP%0Asc.pp.neighbors%28adata%2C%20n_neighbors%3D10%2C%20n_pcs%3D50%29%0Asc.tl.umap%28adata%29%0A%0A%23%20Store%20embeddings%0Aembeddings%20%3D%20pd.DataFrame%28%7B%0A%20%20%20%20%27pca_1%27%3A%20adata.obsm%5B%27X_pca%27%5D%5B%3A%2C0%5D%2C%0A%20%20%20%20%27pca_2%27%3A%20adata.obsm%5B%27X_pca%27%5D%5B%3A%2C1%5D%2C%0A%20%20%20%20%27tsne_1%27%3A%20adata.obsm%5B%27X_tsne%27%5D%5B%3A%2C0%5D%2C%0A%20%20%20%20%27tsne_2%27%3A%20adata.obsm%5B%27X_tsne%27%5D%5B%3A%2C1%5D%2C%0A%20%20%20%20%27umap_1%27%3A%20adata.obsm%5B%27X_umap%27%5D%5B%3A%2C0%5D%2C%0A%20%20%20%20%27umap_2%27%3A%20adata.obsm%5B%27X_umap%27%5D%5B%3A%2C1%5D%0A%7D%29%0Aprint%28embeddings.head%28%29%29%0A%0A%0AThe%20above%20steps%20preprocess%20data%20and%20compute%20embeddings%20using%20PCA%2C%20t-SNE%2C%20and%20UMAP.%20In%20a%20complete%20pipeline%2C%20scVI%20would%20be%20applied%20similarly%20after%20setting%20up%20an%20anndata%20object%20for%20scVI%20processing.%20Comparisons%20among%20these%20embeddings%20can%20reveal%20how%20each%20method%20captures%20potential%20temporal%20structures.%0A%0Aimport%20plotly.express%20as%20px%0A%0A%23%20Visualize%20t-SNE%20embedding%0Afig_tsne%20%3D%20px.scatter%28embeddings%2C%20x%3D%27tsne_1%27%2C%20y%3D%27tsne_2%27%2C%20title%3D%27t-SNE%20Embedding%27%29%0Afig_tsne.show%28%29%0A%0A%23%20Visualize%20UMAP%20embedding%0Afig_umap%20%3D%20px.scatter%28embeddings%2C%20x%3D%27umap_1%27%2C%20y%3D%27umap_2%27%2C%20title%3D%27UMAP%20Embedding%27%29%0Afig_umap.show%28%29%0A%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Evaluating%20discrepancies%20in%20dimensionality%20reduction%20for%20time-series%20single-cell%20RNA-sequencing%20data)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code processes a real PBMC scRNA-seq dataset using multiple dimensionality reduction methods and provides a comparative visualization to benchmark temporal patterns.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-24", "tags": ["RNA-seq dimensionality reduction", "biology", "bioinformatics"]}}