{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "## Differential Gene Expression Analysis of Mosquito Species Under Interspecific Competition\n\nThis notebook analyzes RNA-seq data to identify genes associated with competitive interactions between Aedes aegypti and Anopheles species."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\nimport matplotlib.pyplot as plt\nimport seaborn as sns\nfrom scipy import stats\ndf = pd.read_csv('Aedes_Anopheles_competition_data.csv')\n"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 1: Load and Inspect Data\n\nWe will start by loading the dataset and inspecting its structure to understand the variables involved in the analysis."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "print(df.head())\nprint(df.describe())\nprint(df.info())\n"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 2: Normalize Gene Expression Data\n\nNormalization is crucial for accurate comparison of gene expression levels across samples."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Assuming gene expression counts are in 'counts' column\nfrom sklearn.preprocessing import MinMaxScaler\nscaler = MinMaxScaler()\ndf['normalized_counts'] = scaler.fit_transform(df[['counts']])\n"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 3: Differential Expression Analysis\n\nIdentify genes that are differentially expressed between species under competitive conditions."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "group_a = df[df['species'] == 'Aedes aegypti']['normalized_counts']\ngroup_b = df[df['species'] == 'Anopheles']['normalized_counts']\nt_stat, p_val = stats.ttest_ind(group_a, group_b)\nprint(f'T-test statistic: {t_stat}, p-value: {p_val}')\n"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 4: Visualization of Differential Expression\n\nVisualize the differences in gene expression between the two species."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "sns.boxplot(x='species', y='normalized_counts', data=df)\nplt.title('Gene Expression Comparison')\nplt.show()\n"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 5: Interpretation of Results\n\nInterpret the statistical results to understand the biological significance of differentially expressed genes."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "if p_val < 0.05:\n    print('Significant differential expression observed.')\nelse:\n    print('No significant differential expression.')\n"}, {"cell_type": "markdown", "metadata": {}, "source": "## Conclusion\n\nThe analysis highlights key genes that may be responsible for competitive advantages in Aedes aegypti over Anopheles species."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Further analysis can be performed as needed\n"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20performs%20differential%20gene%20expression%20analysis%20on%20Aedes%20aegypti%20and%20Anopheles%20species%20under%20interspecific%20competition%20using%20RNA-seq%20data%20from%20relevant%20studies.%0A%0AIncorporate%20batch%20effect%20correction%20methods%20and%20utilize%20more%20advanced%20statistical%20models%20for%20more%20accurate%20differential%20expression%20analysis.%0A%0AInterspecific%20competition%20Aedes%20aegypti%20malaria%20vectors%20Tanzania%0A%0A%23%23%20Differential%20Gene%20Expression%20Analysis%20of%20Mosquito%20Species%20Under%20Interspecific%20Competition%0A%0AThis%20notebook%20analyzes%20RNA-seq%20data%20to%20identify%20genes%20associated%20with%20competitive%20interactions%20between%20Aedes%20aegypti%20and%20Anopheles%20species.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0Aimport%20matplotlib.pyplot%20as%20plt%0Aimport%20seaborn%20as%20sns%0Afrom%20scipy%20import%20stats%0Adf%20%3D%20pd.read_csv%28%27Aedes_Anopheles_competition_data.csv%27%29%0A%0A%0A%23%23%23%20Step%201%3A%20Load%20and%20Inspect%20Data%0A%0AWe%20will%20start%20by%20loading%20the%20dataset%20and%20inspecting%20its%20structure%20to%20understand%20the%20variables%20involved%20in%20the%20analysis.%0A%0Aprint%28df.head%28%29%29%0Aprint%28df.describe%28%29%29%0Aprint%28df.info%28%29%29%0A%0A%0A%23%23%23%20Step%202%3A%20Normalize%20Gene%20Expression%20Data%0A%0ANormalization%20is%20crucial%20for%20accurate%20comparison%20of%20gene%20expression%20levels%20across%20samples.%0A%0A%23%20Assuming%20gene%20expression%20counts%20are%20in%20%27counts%27%20column%0Afrom%20sklearn.preprocessing%20import%20MinMaxScaler%0Ascaler%20%3D%20MinMaxScaler%28%29%0Adf%5B%27normalized_counts%27%5D%20%3D%20scaler.fit_transform%28df%5B%5B%27counts%27%5D%5D%29%0A%0A%0A%23%23%23%20Step%203%3A%20Differential%20Expression%20Analysis%0A%0AIdentify%20genes%20that%20are%20differentially%20expressed%20between%20species%20under%20competitive%20conditions.%0A%0Agroup_a%20%3D%20df%5Bdf%5B%27species%27%5D%20%3D%3D%20%27Aedes%20aegypti%27%5D%5B%27normalized_counts%27%5D%0Agroup_b%20%3D%20df%5Bdf%5B%27species%27%5D%20%3D%3D%20%27Anopheles%27%5D%5B%27normalized_counts%27%5D%0At_stat%2C%20p_val%20%3D%20stats.ttest_ind%28group_a%2C%20group_b%29%0Aprint%28f%27T-test%20statistic%3A%20%7Bt_stat%7D%2C%20p-value%3A%20%7Bp_val%7D%27%29%0A%0A%0A%23%23%23%20Step%204%3A%20Visualization%20of%20Differential%20Expression%0A%0AVisualize%20the%20differences%20in%20gene%20expression%20between%20the%20two%20species.%0A%0Asns.boxplot%28x%3D%27species%27%2C%20y%3D%27normalized_counts%27%2C%20data%3Ddf%29%0Aplt.title%28%27Gene%20Expression%20Comparison%27%29%0Aplt.show%28%29%0A%0A%0A%23%23%23%20Step%205%3A%20Interpretation%20of%20Results%0A%0AInterpret%20the%20statistical%20results%20to%20understand%20the%20biological%20significance%20of%20differentially%20expressed%20genes.%0A%0Aif%20p_val%20%3C%200.05%3A%0A%20%20%20%20print%28%27Significant%20differential%20expression%20observed.%27%29%0Aelse%3A%0A%20%20%20%20print%28%27No%20significant%20differential%20expression.%27%29%0A%0A%0A%23%23%20Conclusion%0A%0AThe%20analysis%20highlights%20key%20genes%20that%20may%20be%20responsible%20for%20competitive%20advantages%20in%20Aedes%20aegypti%20over%20Anopheles%20species.%0A%0A%23%20Further%20analysis%20can%20be%20performed%20as%20needed%0A%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20The%20Interspecific%20Competition%20Between%20Larvae%20of%20Aedes%20aegypti%20and%20Major%20African%20Malaria%20Vectors%20in%20a%20Semi-Field%20System%20in%20Tanzania)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code performs differential gene expression analysis on Aedes aegypti and Anopheles species under interspecific competition using RNA-seq data from relevant studies.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-12", "tags": ["Aedes aegypti larvae competition", "biology", "bioinformatics"]}}