{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "Below is a step-by-step Jupyter notebook segment that loads an FFPE scRNAseq dataset, performs quality control, and identifies clusters expressing FOXJ1 using the Scanpy library."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nimport pandas as pd\n\n# Load dataset (replace with actual data link from GEO or similar repository)\ndata_url = 'https://www.ebi.ac.uk/arrayexpress/files/E-MTAB-10372/E-MTAB-10372.json'\nadata = sc.read_json(data_url)\n\n# Preprocessing: filtering and normalization\nsc.pp.filter_cells(adata, min_genes=200)\nsc.pp.filter_genes(adata, min_cells=3)\nsc.pp.normalize_total(adata, target_sum=1e4)\nsc.pp.log1p(adata)\n\n# Identify highly variable genes\nsc.pp.highly_variable_genes(adata, min_mean=0.0125, max_mean=3, min_disp=0.5)\nadata = adata[:, adata.var.highly_variable]\n\n# Perform PCA and clustering\nsc.tl.pca(adata, svd_solver='arpack')\nsc.pp.neighbors(adata, n_neighbors=10, n_pcs=40)\nsc.tl.umap(adata)\nsc.tl.leiden(adata, resolution=0.5)\n\n# Visualization of clusters and FOXJ1 expression\nsc.pl.umap(adata, color=['leiden', 'FOXJ1'], save='_FOXJ1_expression.png')"}, {"cell_type": "markdown", "metadata": {}, "source": "This notebook cell demonstrates data filtering, normalization, dimensionality reduction, clustering, and UMAP visualization to detect FOXJ1-expressing clusters, which are critical for pinpointing MCC subpopulations."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Additional analysis: Identify marker genes in FOXJ1-high clusters\nsc.tl.rank_genes_groups(adata, 'leiden', method='t-test')\nsc.pl.rank_genes_groups(adata, n_genes=20, sharey=False, save='_cluster_markers.png')"}, {"cell_type": "markdown", "metadata": {}, "source": "The second analysis step rank orders genes by cluster specificity, aiding in the confirmation of multi-ciliated cell markers in the identified subpopulation."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20and%20processes%20real%20scRNAseq%20data%20from%20FFPE%20samples%2C%20aligning%20expression%20profiles%20with%20phenotypic%20markers%20to%20validate%20cell%20population%20heterogeneity.%0A%0AInclude%20direct%20links%20to%20validated%20scRNAseq%20datasets%20and%20integrate%20spatial%20transcriptomics%20data%20for%20enhanced%20tissue%20context.%0A%0ASingle-Cell%20RNA%20Sequencing%20FFPE%20Multi-Ciliary%20Cells%20Breast%20Cancer%0A%0ABelow%20is%20a%20step-by-step%20Jupyter%20notebook%20segment%20that%20loads%20an%20FFPE%20scRNAseq%20dataset%2C%20performs%20quality%20control%2C%20and%20identifies%20clusters%20expressing%20FOXJ1%20using%20the%20Scanpy%20library.%0A%0Aimport%20scanpy%20as%20sc%0Aimport%20pandas%20as%20pd%0A%0A%23%20Load%20dataset%20%28replace%20with%20actual%20data%20link%20from%20GEO%20or%20similar%20repository%29%0Adata_url%20%3D%20%27https%3A%2F%2Fwww.ebi.ac.uk%2Farrayexpress%2Ffiles%2FE-MTAB-10372%2FE-MTAB-10372.json%27%0Aadata%20%3D%20sc.read_json%28data_url%29%0A%0A%23%20Preprocessing%3A%20filtering%20and%20normalization%0Asc.pp.filter_cells%28adata%2C%20min_genes%3D200%29%0Asc.pp.filter_genes%28adata%2C%20min_cells%3D3%29%0Asc.pp.normalize_total%28adata%2C%20target_sum%3D1e4%29%0Asc.pp.log1p%28adata%29%0A%0A%23%20Identify%20highly%20variable%20genes%0Asc.pp.highly_variable_genes%28adata%2C%20min_mean%3D0.0125%2C%20max_mean%3D3%2C%20min_disp%3D0.5%29%0Aadata%20%3D%20adata%5B%3A%2C%20adata.var.highly_variable%5D%0A%0A%23%20Perform%20PCA%20and%20clustering%0Asc.tl.pca%28adata%2C%20svd_solver%3D%27arpack%27%29%0Asc.pp.neighbors%28adata%2C%20n_neighbors%3D10%2C%20n_pcs%3D40%29%0Asc.tl.umap%28adata%29%0Asc.tl.leiden%28adata%2C%20resolution%3D0.5%29%0A%0A%23%20Visualization%20of%20clusters%20and%20FOXJ1%20expression%0Asc.pl.umap%28adata%2C%20color%3D%5B%27leiden%27%2C%20%27FOXJ1%27%5D%2C%20save%3D%27_FOXJ1_expression.png%27%29%0A%0AThis%20notebook%20cell%20demonstrates%20data%20filtering%2C%20normalization%2C%20dimensionality%20reduction%2C%20clustering%2C%20and%20UMAP%20visualization%20to%20detect%20FOXJ1-expressing%20clusters%2C%20which%20are%20critical%20for%20pinpointing%20MCC%20subpopulations.%0A%0A%23%20Additional%20analysis%3A%20Identify%20marker%20genes%20in%20FOXJ1-high%20clusters%0Asc.tl.rank_genes_groups%28adata%2C%20%27leiden%27%2C%20method%3D%27t-test%27%29%0Asc.pl.rank_genes_groups%28adata%2C%20n_genes%3D20%2C%20sharey%3DFalse%2C%20save%3D%27_cluster_markers.png%27%29%0A%0AThe%20second%20analysis%20step%20rank%20orders%20genes%20by%20cluster%20specificity%2C%20aiding%20in%20the%20confirmation%20of%20multi-ciliated%20cell%20markers%20in%20the%20identified%20subpopulation.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Single-Cell%20RNA%20Sequencing%20on%20Formalin-Fixed%20and%20Paraffin-Embedded%20%28FFPE%29%20Tissue%20Identified%20Multi-Ciliary%20Cells%20in%20Breast%20Cancer)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads and processes real scRNAseq data from FFPE samples, aligning expression profiles with phenotypic markers to validate cell population heterogeneity.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-25", "tags": ["Single-Cell RNA Sequencing FFPE", "biology", "bioinformatics"]}}