{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "Below is a step-by-step Jupyter notebook snippet that demonstrates how to load and visualize spatial omics data using scanpy and squidpy. Replace 'path_to_CORAL_validated_data.h5ad' with your actual dataset file."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nimport squidpy as sq\n\n# Load CORAL-processed spatial omics dataset\nadata = sc.read_h5ad('path_to_CORAL_validated_data.h5ad')\n\n# Preprocess the data: filter cells and genes\nsc.pp.filter_cells(adata, min_genes=200)\nsc.pp.filter_genes(adata, min_cells=3)\n\n# Visualize spatial distribution based on predicted cell types\nsq.pl.spatial_scatter(adata, color='cell_type', title='Spatial Distribution by CORAL', size=1.5)"}, {"cell_type": "markdown", "metadata": {}, "source": "Next, we compute and visualize a UMAP representation to inspect clustering and potential cell\u2013cell interactions from the integrated data."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "sc.tl.umap(adata)\nsc.pl.umap(adata, color='cell_type', title='UMAP of CORAL-Imputed Data')"}, {"cell_type": "markdown", "metadata": {}, "source": "Finally, we calculate spatial autocorrelation using squidpy to assess the spatial dependency among cells, which is central to CORAL's approach."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "sq.gr.spatial_autocorr(adata, mode='Moran')\nsc.pl.umap(adata, color='Moran_I', title='Moran I Spatial Autocorrelation')"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20loads%2C%20filters%2C%20and%20visualizes%20real%20spatial%20omics%20data%20processed%20with%20CORAL%20to%20validate%20single-cell%20deconvolution%20and%20predict%20cell%20interactions.%0A%0ASpecify%20direct%20file%20paths%2C%20include%20interactive%20Plotly%20visualization%2C%20and%20fine-tune%20parameters%20for%20dataset-specific%20spatial%20analysis.%0A%0ASingle-cell%20spatial%20context%20multiomics%20CORAL%20review%0A%0ABelow%20is%20a%20step-by-step%20Jupyter%20notebook%20snippet%20that%20demonstrates%20how%20to%20load%20and%20visualize%20spatial%20omics%20data%20using%20scanpy%20and%20squidpy.%20Replace%20%27path_to_CORAL_validated_data.h5ad%27%20with%20your%20actual%20dataset%20file.%0A%0Aimport%20scanpy%20as%20sc%0Aimport%20squidpy%20as%20sq%0A%0A%23%20Load%20CORAL-processed%20spatial%20omics%20dataset%0Aadata%20%3D%20sc.read_h5ad%28%27path_to_CORAL_validated_data.h5ad%27%29%0A%0A%23%20Preprocess%20the%20data%3A%20filter%20cells%20and%20genes%0Asc.pp.filter_cells%28adata%2C%20min_genes%3D200%29%0Asc.pp.filter_genes%28adata%2C%20min_cells%3D3%29%0A%0A%23%20Visualize%20spatial%20distribution%20based%20on%20predicted%20cell%20types%0Asq.pl.spatial_scatter%28adata%2C%20color%3D%27cell_type%27%2C%20title%3D%27Spatial%20Distribution%20by%20CORAL%27%2C%20size%3D1.5%29%0A%0ANext%2C%20we%20compute%20and%20visualize%20a%20UMAP%20representation%20to%20inspect%20clustering%20and%20potential%20cell%E2%80%93cell%20interactions%20from%20the%20integrated%20data.%0A%0Asc.tl.umap%28adata%29%0Asc.pl.umap%28adata%2C%20color%3D%27cell_type%27%2C%20title%3D%27UMAP%20of%20CORAL-Imputed%20Data%27%29%0A%0AFinally%2C%20we%20calculate%20spatial%20autocorrelation%20using%20squidpy%20to%20assess%20the%20spatial%20dependency%20among%20cells%2C%20which%20is%20central%20to%20CORAL%27s%20approach.%0A%0Asq.gr.spatial_autocorr%28adata%2C%20mode%3D%27Moran%27%29%0Asc.pl.umap%28adata%2C%20color%3D%27Moran_I%27%2C%20title%3D%27Moran%20I%20Spatial%20Autocorrelation%27%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Learning%20single-cell%20spatial%20context%20through%20integrated%20spatial%20multiomics%20with%20CORAL)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code loads, filters, and visualizes real spatial omics data processed with CORAL to validate single-cell deconvolution and predict cell interactions.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-12", "tags": ["single-cell spatial context", "biology", "bioinformatics"]}}