{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "Step 1: Download and preprocess the RNA-seq dataset from the provided repository. Include quality control and normalization implementations."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\nfrom edgeR import DGEList, calcNormFactors\n# Load dataset\nrna_seq_data = pd.read_csv('path_to_dataset.csv')\n# Preprocessing steps\ndge = DGEList(counts=rna_seq_data)\ndge = calcNormFactors(dge)\nprint('Data normalized for differential expression analysis')"}, {"cell_type": "markdown", "metadata": {}, "source": "Step 2: Perform differential expression analysis and extract significantly altered genes."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import statsmodels.api as sm\n# Assuming design matrix and contrasts are defined\nresult = sm.OLS(dge.counts, design_matrix).fit()\ndeg_results = result.summary()\nprint(deg_results)"}, {"cell_type": "markdown", "metadata": {}, "source": "Step 3: Conduct Gene Ontology enrichment using topGO or similar tools to contextualize the altered biological pathways."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "from gseapy import enrichr\n# Prepare list of significant DEGs\nde_gene_list = list(deg_results[deg_results['pvalue'] < 0.05].index)\n# Enrichment analysis\nenr = enrichr(gene_list=de_gene_list, gene_sets=['GO_Biological_Process_2021'], description='EDC_Gene_Enrichment', outdir='test/enrichr_kegg', cutoff=0.05)\nprint(enr.results.head())"}, {"cell_type": "markdown", "metadata": {}, "source": "Step 4: Visualize the top enriched terms using interactive Plotly bar charts."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import plotly.express as px\nfig = px.bar(enr.results, x='Term', y='Combined Score', title='Top Enriched GO Terms')\nfig.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20analyzes%20RNA-seq%20differential%20expression%20and%20gene%20ontology%20enrichment%20from%20hMSC%20data%20following%20EDC%20exposure%2C%20facilitating%20in-depth%20pathway%20analysis.%0A%0AIncorporate%20more%20robust%20statistical%20normalization%20and%20incorporate%20real-time%20data%20linkage%20from%20the%20pertinent%20transcriptomics%20dataset.%0A%0AEndocrine%20disruptors%2C%20birth%20weight%2C%20adipogenesis%2C%20diabetes%2C%20transcriptional%20changes%0A%0AStep%201%3A%20Download%20and%20preprocess%20the%20RNA-seq%20dataset%20from%20the%20provided%20repository.%20Include%20quality%20control%20and%20normalization%20implementations.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0Afrom%20edgeR%20import%20DGEList%2C%20calcNormFactors%0A%23%20Load%20dataset%0Arna_seq_data%20%3D%20pd.read_csv%28%27path_to_dataset.csv%27%29%0A%23%20Preprocessing%20steps%0Adge%20%3D%20DGEList%28counts%3Drna_seq_data%29%0Adge%20%3D%20calcNormFactors%28dge%29%0Aprint%28%27Data%20normalized%20for%20differential%20expression%20analysis%27%29%0A%0AStep%202%3A%20Perform%20differential%20expression%20analysis%20and%20extract%20significantly%20altered%20genes.%0A%0Aimport%20statsmodels.api%20as%20sm%0A%23%20Assuming%20design%20matrix%20and%20contrasts%20are%20defined%0Aresult%20%3D%20sm.OLS%28dge.counts%2C%20design_matrix%29.fit%28%29%0Adeg_results%20%3D%20result.summary%28%29%0Aprint%28deg_results%29%0A%0AStep%203%3A%20Conduct%20Gene%20Ontology%20enrichment%20using%20topGO%20or%20similar%20tools%20to%20contextualize%20the%20altered%20biological%20pathways.%0A%0Afrom%20gseapy%20import%20enrichr%0A%23%20Prepare%20list%20of%20significant%20DEGs%0Ade_gene_list%20%3D%20list%28deg_results%5Bdeg_results%5B%27pvalue%27%5D%20%3C%200.05%5D.index%29%0A%23%20Enrichment%20analysis%0Aenr%20%3D%20enrichr%28gene_list%3Dde_gene_list%2C%20gene_sets%3D%5B%27GO_Biological_Process_2021%27%5D%2C%20description%3D%27EDC_Gene_Enrichment%27%2C%20outdir%3D%27test%2Fenrichr_kegg%27%2C%20cutoff%3D0.05%29%0Aprint%28enr.results.head%28%29%29%0A%0AStep%204%3A%20Visualize%20the%20top%20enriched%20terms%20using%20interactive%20Plotly%20bar%20charts.%0A%0Aimport%20plotly.express%20as%20px%0Afig%20%3D%20px.bar%28enr.results%2C%20x%3D%27Term%27%2C%20y%3D%27Combined%20Score%27%2C%20title%3D%27Top%20Enriched%20GO%20Terms%27%29%0Afig.show%28%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20A%20mixture%20of%20endocrine%20disrupting%20chemicals%20linked%20to%20lower%20birth%20weight%20induces%20adipogenesis%20and%20transcriptional%20changes%20related%20to%20birth%20weight%20alterations%20and%20diabetes)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code analyzes RNA-seq differential expression and gene ontology enrichment from hMSC data following EDC exposure, facilitating in-depth pathway analysis.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-24", "tags": ["endocrine disruptors birth weight", "biology", "bioinformatics"]}}