{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "### Data Retrieval and Preprocessing\nIn this notebook, we import complete plastome datasets of Phryma leptostachya from GenBank. We then perform alignment using Biopython and preprocess data for BEAST analysis."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import os\nfrom Bio import Entrez, SeqIO\n\nEntrez.email = 'your.email@example.com'\nhandle = Entrez.esearch(db='nuccore', term='Phryma leptostachya[Organism] AND complete plastome', retmax=10)\nrecord = Entrez.read(handle)\nids = record['IdList']\n\nsequences = []\nfor seq_id in ids:\n    handle = Entrez.efetch(db='nuccore', id=seq_id, rettype='fasta', retmode='text')\n    seq_record = SeqIO.read(handle, 'fasta')\n    sequences.append(seq_record)\n\n# Save sequences to file\nSeqIO.write(sequences, 'phryma_plastomes.fasta', 'fasta')\nprint('Downloaded and saved complete plastome sequences.')"}, {"cell_type": "markdown", "metadata": {}, "source": "### Phylogenetic Analysis Setup\nPrepare aligned sequences for divergence analysis using BEAST. Align sequences using MAFFT and convert the alignment to NEXUS format for BEAST."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import subprocess\n\n# Running MAFFT for multiple sequence alignment\nsubprocess.run(['mafft', '--auto', 'phryma_plastomes.fasta'], stdout=open('aligned_phryma.fasta', 'w'))\n\nprint('Alignment completed, ready for BEAST analysis.')"}, {"cell_type": "markdown", "metadata": {}, "source": "### Discussion\nThis pipeline facilitates the reconstruction of phylogenetic relationships and divergence time estimation using robust statistical methods, integrating modern bioinformatics tools."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Additional code to parse BEAST output and visualize the tree would be implemented here.\n# This is a skeleton for a full bioinformatics workflow based on the provided datasets."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20complete%20plastome%20datasets%20and%20performs%20phylogenetic%20and%20divergence%20time%20analyses%20using%20BEAST%20and%20RASP.%0A%0AInclude%20error%20handling%20and%20integrate%20visualization%20libraries%20like%20Plotly%20to%20render%20phylogenetic%20trees%20interactively.%0A%0APhryma%20leptostachya%20phylogenetics%20biogeography%20plastome%20review%0A%0A%23%23%23%20Data%20Retrieval%20and%20Preprocessing%0AIn%20this%20notebook%2C%20we%20import%20complete%20plastome%20datasets%20of%20Phryma%20leptostachya%20from%20GenBank.%20We%20then%20perform%20alignment%20using%20Biopython%20and%20preprocess%20data%20for%20BEAST%20analysis.%0A%0Aimport%20os%0Afrom%20Bio%20import%20Entrez%2C%20SeqIO%0A%0AEntrez.email%20%3D%20%27your.email%40example.com%27%0Ahandle%20%3D%20Entrez.esearch%28db%3D%27nuccore%27%2C%20term%3D%27Phryma%20leptostachya%5BOrganism%5D%20AND%20complete%20plastome%27%2C%20retmax%3D10%29%0Arecord%20%3D%20Entrez.read%28handle%29%0Aids%20%3D%20record%5B%27IdList%27%5D%0A%0Asequences%20%3D%20%5B%5D%0Afor%20seq_id%20in%20ids%3A%0A%20%20%20%20handle%20%3D%20Entrez.efetch%28db%3D%27nuccore%27%2C%20id%3Dseq_id%2C%20rettype%3D%27fasta%27%2C%20retmode%3D%27text%27%29%0A%20%20%20%20seq_record%20%3D%20SeqIO.read%28handle%2C%20%27fasta%27%29%0A%20%20%20%20sequences.append%28seq_record%29%0A%0A%23%20Save%20sequences%20to%20file%0ASeqIO.write%28sequences%2C%20%27phryma_plastomes.fasta%27%2C%20%27fasta%27%29%0Aprint%28%27Downloaded%20and%20saved%20complete%20plastome%20sequences.%27%29%0A%0A%23%23%23%20Phylogenetic%20Analysis%20Setup%0APrepare%20aligned%20sequences%20for%20divergence%20analysis%20using%20BEAST.%20Align%20sequences%20using%20MAFFT%20and%20convert%20the%20alignment%20to%20NEXUS%20format%20for%20BEAST.%0A%0Aimport%20subprocess%0A%0A%23%20Running%20MAFFT%20for%20multiple%20sequence%20alignment%0Asubprocess.run%28%5B%27mafft%27%2C%20%27--auto%27%2C%20%27phryma_plastomes.fasta%27%5D%2C%20stdout%3Dopen%28%27aligned_phryma.fasta%27%2C%20%27w%27%29%29%0A%0Aprint%28%27Alignment%20completed%2C%20ready%20for%20BEAST%20analysis.%27%29%0A%0A%23%23%23%20Discussion%0AThis%20pipeline%20facilitates%20the%20reconstruction%20of%20phylogenetic%20relationships%20and%20divergence%20time%20estimation%20using%20robust%20statistical%20methods%2C%20integrating%20modern%20bioinformatics%20tools.%0A%0A%23%20Additional%20code%20to%20parse%20BEAST%20output%20and%20visualize%20the%20tree%20would%20be%20implemented%20here.%0A%23%20This%20is%20a%20skeleton%20for%20a%20full%20bioinformatics%20workflow%20based%20on%20the%20provided%20datasets.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Revisiting%20Phryma%20leptostachya%20L.%3A%20phylogenetic%20relationships%20and%20biogeographical%20patterns%20from%20complete%20plastome)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads complete plastome datasets and performs phylogenetic and divergence time analyses using BEAST and RASP.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-19", "tags": ["Phryma leptostachya phylogeny", "biology", "bioinformatics"]}}