{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "Import and preprocess RNA-seq datasets from wild tomato species; compare expression profiles of transcription factors like NAC29."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport matplotlib.pyplot as plt\n\n# Download dataset from NCBI GEO using accession GSE288242\n# Process expression data for selected transcription factors\n# Plot expression levels over different species\n\ndf = pd.read_csv('gene_expression_data.csv')\nselected_genes = df[df['gene'].isin(['NAC29', 'ethylene_response_factor'])]\n\nplt.figure(figsize=(10,6))\nfor species in selected_genes['species'].unique():\n    subset = selected_genes[selected_genes['species'] == species]\n    plt.plot(subset['time_point'], subset['expression'], label=species)\nplt.title('Expression Profiles of NAC29 Across Wild Tomato Species')\nplt.xlabel('Time Point')\nplt.ylabel('Expression Level')\nplt.legend()\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "This analysis visualizes the temporal expression changes and facilitates cross-species comparisons of key transcription factors influencing QDR."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20and%20visualizes%20comparative%20transcriptomic%20datasets%20to%20identify%20regulatory%20network%20rewiring%20in%20QDR%20across%20species.%0A%0AInclude%20integration%20of%20epigenetic%20and%20chromatin%20accessibility%20datasets%20for%20a%20multi-layer%20regulatory%20analysis.%0A%0ACo-optation%20transcription%20factors%20quantitative%20disease%20resistance%20evolution%0A%0AImport%20and%20preprocess%20RNA-seq%20datasets%20from%20wild%20tomato%20species%3B%20compare%20expression%20profiles%20of%20transcription%20factors%20like%20NAC29.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20matplotlib.pyplot%20as%20plt%0A%0A%23%20Download%20dataset%20from%20NCBI%20GEO%20using%20accession%20GSE288242%0A%23%20Process%20expression%20data%20for%20selected%20transcription%20factors%0A%23%20Plot%20expression%20levels%20over%20different%20species%0A%0Adf%20%3D%20pd.read_csv%28%27gene_expression_data.csv%27%29%0Aselected_genes%20%3D%20df%5Bdf%5B%27gene%27%5D.isin%28%5B%27NAC29%27%2C%20%27ethylene_response_factor%27%5D%29%5D%0A%0Aplt.figure%28figsize%3D%2810%2C6%29%29%0Afor%20species%20in%20selected_genes%5B%27species%27%5D.unique%28%29%3A%0A%20%20%20%20subset%20%3D%20selected_genes%5Bselected_genes%5B%27species%27%5D%20%3D%3D%20species%5D%0A%20%20%20%20plt.plot%28subset%5B%27time_point%27%5D%2C%20subset%5B%27expression%27%5D%2C%20label%3Dspecies%29%0Aplt.title%28%27Expression%20Profiles%20of%20NAC29%20Across%20Wild%20Tomato%20Species%27%29%0Aplt.xlabel%28%27Time%20Point%27%29%0Aplt.ylabel%28%27Expression%20Level%27%29%0Aplt.legend%28%29%0Aplt.show%28%29%0A%0AThis%20analysis%20visualizes%20the%20temporal%20expression%20changes%20and%20facilitates%20cross-species%20comparisons%20of%20key%20transcription%20factors%20influencing%20QDR.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20co-optation%20of%20transcription%20factors%20drives%20evolution%20of%20quantitative%20disease%20resistance)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads and visualizes comparative transcriptomic datasets to identify regulatory network rewiring in QDR across species.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-19", "tags": ["transcription factors evolution", "biology", "bioinformatics"]}}