{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "### Data Description\nThis notebook processes the spRandom-seq dataset from FFPE tissues to perform quality control, clustering, and identification of key differentially expressed genes."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\n\n# Load the spRandom-seq dataset (replace 'spRandomseq_data.h5ad' with the real filename)\nadata = sc.read_h5ad('spRandomseq_data.h5ad')\n\n# Basic QC and filtering\nsc.pp.filter_cells(adata, min_genes=200)\nsc.pp.filter_genes(adata, min_cells=3)\n\n# Normalize and log-transform\nsc.pp.normalize_total(adata, target_sum=1e4)\nsc.pp.log1p(adata)\n\n# Highly variable genes and PCA\nsc.pp.highly_variable_genes(adata, min_mean=0.0125, max_mean=3, min_disp=0.5)\nadata = adata[:, adata.var.highly_variable]\nsc.pp.scale(adata, max_value=10)\nsc.tl.pca(adata, svd_solver='arpack')\n\n# Clustering using Leiden algorithm\nsc.pp.neighbors(adata, n_neighbors=10, n_pcs=40)\nsc.tl.leiden(adata)\n\n# UMAP visualization\nsc.tl.umap(adata)\nsc.pl.umap(adata, color=['leiden'], save='_spRandomseq_clusters.png')\n\n# Differential expression analysis\nsc.tl.rank_genes_groups(adata, 'leiden', method='t-test')\nsc.pl.rank_genes_groups(adata, n_genes=20, sharey=False, save='_spRandomseq_DE.png')\n\nprint('spRandom-seq analysis complete.')"}, {"cell_type": "markdown", "metadata": {}, "source": "### Analysis Discussion\nThe above code performs essential preprocessing of the spRandom-seq dataset. It includes normalization, PCA, clustering, UMAP visualization, and differential gene expression analysis to identify spatially variable genes."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Additional analysis can be incorporated based on dataset specifics\nimport matplotlib.pyplot as plt\n\n# Display UMAP\nsc.pl.umap(adata, color=['leiden'], title='spRandom-seq Leiden Clusters')\nplt.show()\n\n# Save the final AnnData object for future use\nadata.write('spRandomseq_processed.h5ad')"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20and%20processes%20the%20spRandom-seq%20dataset%20for%20quality%20assessment%20and%20clustering%20analysis.%0A%0AInclude%20integration%20with%20spatial%20coordinates%20and%20incorporate%20CNV%2Fmutation%20analysis%20modules%20for%20a%20more%20comprehensive%20pipeline.%0A%0ASpatial%20RNA%20sequencing%20FFPE%20tissues%20spRandom-seq%20review%0A%0A%23%23%23%20Data%20Description%0AThis%20notebook%20processes%20the%20spRandom-seq%20dataset%20from%20FFPE%20tissues%20to%20perform%20quality%20control%2C%20clustering%2C%20and%20identification%20of%20key%20differentially%20expressed%20genes.%0A%0Aimport%20scanpy%20as%20sc%0A%0A%23%20Load%20the%20spRandom-seq%20dataset%20%28replace%20%27spRandomseq_data.h5ad%27%20with%20the%20real%20filename%29%0Aadata%20%3D%20sc.read_h5ad%28%27spRandomseq_data.h5ad%27%29%0A%0A%23%20Basic%20QC%20and%20filtering%0Asc.pp.filter_cells%28adata%2C%20min_genes%3D200%29%0Asc.pp.filter_genes%28adata%2C%20min_cells%3D3%29%0A%0A%23%20Normalize%20and%20log-transform%0Asc.pp.normalize_total%28adata%2C%20target_sum%3D1e4%29%0Asc.pp.log1p%28adata%29%0A%0A%23%20Highly%20variable%20genes%20and%20PCA%0Asc.pp.highly_variable_genes%28adata%2C%20min_mean%3D0.0125%2C%20max_mean%3D3%2C%20min_disp%3D0.5%29%0Aadata%20%3D%20adata%5B%3A%2C%20adata.var.highly_variable%5D%0Asc.pp.scale%28adata%2C%20max_value%3D10%29%0Asc.tl.pca%28adata%2C%20svd_solver%3D%27arpack%27%29%0A%0A%23%20Clustering%20using%20Leiden%20algorithm%0Asc.pp.neighbors%28adata%2C%20n_neighbors%3D10%2C%20n_pcs%3D40%29%0Asc.tl.leiden%28adata%29%0A%0A%23%20UMAP%20visualization%0Asc.tl.umap%28adata%29%0Asc.pl.umap%28adata%2C%20color%3D%5B%27leiden%27%5D%2C%20save%3D%27_spRandomseq_clusters.png%27%29%0A%0A%23%20Differential%20expression%20analysis%0Asc.tl.rank_genes_groups%28adata%2C%20%27leiden%27%2C%20method%3D%27t-test%27%29%0Asc.pl.rank_genes_groups%28adata%2C%20n_genes%3D20%2C%20sharey%3DFalse%2C%20save%3D%27_spRandomseq_DE.png%27%29%0A%0Aprint%28%27spRandom-seq%20analysis%20complete.%27%29%0A%0A%23%23%23%20Analysis%20Discussion%0AThe%20above%20code%20performs%20essential%20preprocessing%20of%20the%20spRandom-seq%20dataset.%20It%20includes%20normalization%2C%20PCA%2C%20clustering%2C%20UMAP%20visualization%2C%20and%20differential%20gene%20expression%20analysis%20to%20identify%20spatially%20variable%20genes.%0A%0A%23%20Additional%20analysis%20can%20be%20incorporated%20based%20on%20dataset%20specifics%0Aimport%20matplotlib.pyplot%20as%20plt%0A%0A%23%20Display%20UMAP%0Asc.pl.umap%28adata%2C%20color%3D%5B%27leiden%27%5D%2C%20title%3D%27spRandom-seq%20Leiden%20Clusters%27%29%0Aplt.show%28%29%0A%0A%23%20Save%20the%20final%20AnnData%20object%20for%20future%20use%0Aadata.write%28%27spRandomseq_processed.h5ad%27%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Spatial%20Total%20RNA%20Sequencing%20of%20Formalin-Fixed%20Paraffin%20Embedded%20Tissues%20by%20spRandom-seq.)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads and processes the spRandom-seq dataset for quality assessment and clustering analysis.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-03", "tags": ["spatial RNA sequencing FFPE", "biology", "bioinformatics"]}}