{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "Below, we download the relevant genomic datasets and use Biopython along with pandas and scipy to compute dN/dS ratios for candidate genes."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nfrom Bio import AlignIO\nfrom Bio.SeqRecord import SeqRecord\nimport numpy as np\n\n# Download dataset from URL_0 (placeholder for actual download code)\n# Assume dataset.csv contains columns: gene_id, nonsynonymous, synonymous\n\ndata = pd.read_csv('dataset.csv')\n\ndata['dn_ds'] = data['nonsynonymous'] / (data['synonymous'] + 1e-6)\n\nenriched_genes = data[data['dn_ds'] > data['dn_ds'].median()]\n\nprint('Top candidate genes for relaxed selection:')\nprint(enriched_genes.sort_values(by='dn_ds', ascending=False).head(10))"}, {"cell_type": "markdown", "metadata": {}, "source": "The following cell performs GO enrichment analysis using the gseapy package on the candidate gene list."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import gseapy as gp\n\n# Example: performing GO enrichment on candidate genes (gene list would be obtained from the previous step)\n\ngene_list = enriched_genes['gene_id'].tolist()\n\nenrich_res = gp.enrichr(gene_list=gene_list,\n                         gene_sets='GO_Biological_Process_2018',\n                         organism='Fish',\n                         cutoff=0.05)\n\nprint(enrich_res.results.head())"}, {"cell_type": "markdown", "metadata": {}, "source": "This notebook cell-by-cell analysis allows reproducible evaluation of the genomic signals of relaxed selection and helps in identifying functional pathways under constraint in adaptive evolution."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Further code can visualize these enrichments with Plotly\nimport plotly.express as px\n\nfig = px.bar(enrich_res.results, x='Term', y='Combined Score', title='GO Enrichment of Candidate Genes')\nfig.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20and%20processes%20Triplophysa%20genomic%20datasets%20to%20replicate%20relaxed%20selection%20analysis%20using%20dN%2FdS%20ratios%20and%20GO%20enrichment.%0A%0AIntegrate%20direct%20dataset%20download%20links%20and%20error%20handling%20for%20real-time%20data%20processing.%0A%0AWhole-genome%20relaxed%20selection%20Triplophysa%20Qinghai-Tibetan%20Plateau%20molecular%20constraints%0A%0ABelow%2C%20we%20download%20the%20relevant%20genomic%20datasets%20and%20use%20Biopython%20along%20with%20pandas%20and%20scipy%20to%20compute%20dN%2FdS%20ratios%20for%20candidate%20genes.%0A%0Aimport%20pandas%20as%20pd%0Afrom%20Bio%20import%20AlignIO%0Afrom%20Bio.SeqRecord%20import%20SeqRecord%0Aimport%20numpy%20as%20np%0A%0A%23%20Download%20dataset%20from%20URL_0%20%28placeholder%20for%20actual%20download%20code%29%0A%23%20Assume%20dataset.csv%20contains%20columns%3A%20gene_id%2C%20nonsynonymous%2C%20synonymous%0A%0Adata%20%3D%20pd.read_csv%28%27dataset.csv%27%29%0A%0Adata%5B%27dn_ds%27%5D%20%3D%20data%5B%27nonsynonymous%27%5D%20%2F%20%28data%5B%27synonymous%27%5D%20%2B%201e-6%29%0A%0Aenriched_genes%20%3D%20data%5Bdata%5B%27dn_ds%27%5D%20%3E%20data%5B%27dn_ds%27%5D.median%28%29%5D%0A%0Aprint%28%27Top%20candidate%20genes%20for%20relaxed%20selection%3A%27%29%0Aprint%28enriched_genes.sort_values%28by%3D%27dn_ds%27%2C%20ascending%3DFalse%29.head%2810%29%29%0A%0AThe%20following%20cell%20performs%20GO%20enrichment%20analysis%20using%20the%20gseapy%20package%20on%20the%20candidate%20gene%20list.%0A%0Aimport%20gseapy%20as%20gp%0A%0A%23%20Example%3A%20performing%20GO%20enrichment%20on%20candidate%20genes%20%28gene%20list%20would%20be%20obtained%20from%20the%20previous%20step%29%0A%0Agene_list%20%3D%20enriched_genes%5B%27gene_id%27%5D.tolist%28%29%0A%0Aenrich_res%20%3D%20gp.enrichr%28gene_list%3Dgene_list%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20gene_sets%3D%27GO_Biological_Process_2018%27%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20organism%3D%27Fish%27%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20cutoff%3D0.05%29%0A%0Aprint%28enrich_res.results.head%28%29%29%0A%0AThis%20notebook%20cell-by-cell%20analysis%20allows%20reproducible%20evaluation%20of%20the%20genomic%20signals%20of%20relaxed%20selection%20and%20helps%20in%20identifying%20functional%20pathways%20under%20constraint%20in%20adaptive%20evolution.%0A%0A%23%20Further%20code%20can%20visualize%20these%20enrichments%20with%20Plotly%0Aimport%20plotly.express%20as%20px%0A%0Afig%20%3D%20px.bar%28enrich_res.results%2C%20x%3D%27Term%27%2C%20y%3D%27Combined%20Score%27%2C%20title%3D%27GO%20Enrichment%20of%20Candidate%20Genes%27%29%0Afig.show%28%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Whole-genome%20relaxed%20selection%20and%20molecular%20constraints%20in%20Triplophysa%20under%20adapted%20Qinghai-Tibetan%20Plateau.)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads and processes Triplophysa genomic datasets to replicate relaxed selection analysis using dN/dS ratios and GO enrichment.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-24", "tags": ["Triplophysa genome selection", "biology", "bioinformatics"]}}