{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "The following notebook section describes how to download real sequencing data and simulated datasets to benchmark centroAnno."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\n# Download dataset from FigShare\nurl_sim = 'https://doi.org/10.6084/m9.figshare.26780017.v1'\ndata_sim = pd.read_csv(url_sim)\n\n# Placeholder for benchmarking centroAnno results\ncentroAnno_results = pd.read_csv('path_to_centroAnno_output.csv')\n\n# Compare annotation speeds and accuracies\nbenchmark_table = pd.merge(data_sim, centroAnno_results, on='chromosome_id', suffixes=('_sim', '_centro'))\nprint(benchmark_table.head())"}, {"cell_type": "markdown", "metadata": {}, "source": "This section outlines the steps for a detailed comparison of annotation performance and accuracy between centroAnno and existing methods."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import matplotlib.pyplot as plt\nimport seaborn as sns\n\nplt.figure(figsize=(10,6))\nax = sns.barplot(x='chromosome_id', y='annotation_speed', data=benchmark_table, color='#6A0C76')\nax.set_title('Comparison of Annotation Speeds')\nax.set_xlabel('Chromosome')\nax.set_ylabel('Speed (relative units)')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "The above plots help compare how effectively centroAnno outpaces other methods in processing time while maintaining accuracy in centromere annotation."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "from scipy.stats import pearsonr\n\n# Calculate correlation between predicted and simulated annotation accuracies\naccuracy_corr, p_value = pearsonr(benchmark_table['annotation_accuracy_sim'], benchmark_table['annotation_accuracy_centro'])\nprint('Correlation:', accuracy_corr, 'P-value:', p_value)"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20centromere%20annotation%20datasets%20and%20benchmarks%20centroAnno%20performance%20against%20alternative%20tools.%0A%0AImplement%20exception%20handling%20for%20file%20downloads%20and%20incorporate%20additional%20statistical%20analyses%20for%20robustness.%0A%0Acentromere%20annotation%20centroAnno%202025%20review%0A%0AThe%20following%20notebook%20section%20describes%20how%20to%20download%20real%20sequencing%20data%20and%20simulated%20datasets%20to%20benchmark%20centroAnno.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0A%23%20Download%20dataset%20from%20FigShare%0Aurl_sim%20%3D%20%27https%3A%2F%2Fdoi.org%2F10.6084%2Fm9.figshare.26780017.v1%27%0Adata_sim%20%3D%20pd.read_csv%28url_sim%29%0A%0A%23%20Placeholder%20for%20benchmarking%20centroAnno%20results%0AcentroAnno_results%20%3D%20pd.read_csv%28%27path_to_centroAnno_output.csv%27%29%0A%0A%23%20Compare%20annotation%20speeds%20and%20accuracies%0Abenchmark_table%20%3D%20pd.merge%28data_sim%2C%20centroAnno_results%2C%20on%3D%27chromosome_id%27%2C%20suffixes%3D%28%27_sim%27%2C%20%27_centro%27%29%29%0Aprint%28benchmark_table.head%28%29%29%0A%0AThis%20section%20outlines%20the%20steps%20for%20a%20detailed%20comparison%20of%20annotation%20performance%20and%20accuracy%20between%20centroAnno%20and%20existing%20methods.%0A%0Aimport%20matplotlib.pyplot%20as%20plt%0Aimport%20seaborn%20as%20sns%0A%0Aplt.figure%28figsize%3D%2810%2C6%29%29%0Aax%20%3D%20sns.barplot%28x%3D%27chromosome_id%27%2C%20y%3D%27annotation_speed%27%2C%20data%3Dbenchmark_table%2C%20color%3D%27%236A0C76%27%29%0Aax.set_title%28%27Comparison%20of%20Annotation%20Speeds%27%29%0Aax.set_xlabel%28%27Chromosome%27%29%0Aax.set_ylabel%28%27Speed%20%28relative%20units%29%27%29%0Aplt.show%28%29%0A%0AThe%20above%20plots%20help%20compare%20how%20effectively%20centroAnno%20outpaces%20other%20methods%20in%20processing%20time%20while%20maintaining%20accuracy%20in%20centromere%20annotation.%0A%0Afrom%20scipy.stats%20import%20pearsonr%0A%0A%23%20Calculate%20correlation%20between%20predicted%20and%20simulated%20annotation%20accuracies%0Aaccuracy_corr%2C%20p_value%20%3D%20pearsonr%28benchmark_table%5B%27annotation_accuracy_sim%27%5D%2C%20benchmark_table%5B%27annotation_accuracy_centro%27%5D%29%0Aprint%28%27Correlation%3A%27%2C%20accuracy_corr%2C%20%27P-value%3A%27%2C%20p_value%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20De%20novo%20annotation%20of%20centromere%20with%20centroAnno%20%5B2025%5D)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads centromere annotation datasets and benchmarks centroAnno performance against alternative tools.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-03", "tags": ["centromere annotation 2025", "biology", "bioinformatics"]}}