{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "In this notebook we import mutation data from the paper and create a DataFrame summarizing the mutations. This table can then be visualized using Plotly and pandas."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport plotly.express as px\n\n# Define mutation data as provided\ndata = [\n    {'Gene': 'FOXI1', 'Mutation': 'c.517C > A', 'Protein_Change': 'p.His173Asn', 'Impact': 'Moderate', 'Read_Depth': '366/324'},\n    {'Gene': 'KRT6C', 'Mutation': 'c.913G > C', 'Protein_Change': 'p.Glu305Gln', 'Impact': 'Moderate', 'Read_Depth': '498/420'},\n    {'Gene': 'GJD2', 'Mutation': 'c.517G > T', 'Protein_Change': 'p.Glu173*', 'Impact': 'High', 'Read_Depth': '187/164'}\n]\n\ndf = pd.DataFrame(data)\n\n# Display the DataFrame\nprint(df)\n\n# Plot a simple bar plot for variant allele frequencies (if available) or visualize mutation types\nfig = px.bar(df, x='Gene', y=['Read_Depth'], title='Mutation Read Depths', labels={'value': 'Read Depth', 'variable': 'Metric'})\nfig.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "The above code constructs a table from the mutation data and uses Plotly to create a bar plot. This analysis is useful to visually compare the read depths and infer reliability of detection."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20section%20downloads%20relevant%20mutation%20data%20and%20generates%20a%20detailed%20mutation%20table%20to%20aid%20visualization%20and%20further%20analysis.%0A%0AIntegrate%20actual%20VAF%20values%20and%20additional%20mutation%20predictors%20from%20the%20dataset%20to%20extend%20the%20analysis.%0A%0AWhole%20exome%20sequencing%20FOXI1%20KRT6C%20GJD2%20low-grade%20oncocytic%20tumor%20kidney%20case%20report%0A%0AIn%20this%20notebook%20we%20import%20mutation%20data%20from%20the%20paper%20and%20create%20a%20DataFrame%20summarizing%20the%20mutations.%20This%20table%20can%20then%20be%20visualized%20using%20Plotly%20and%20pandas.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20plotly.express%20as%20px%0A%0A%23%20Define%20mutation%20data%20as%20provided%0Adata%20%3D%20%5B%0A%20%20%20%20%7B%27Gene%27%3A%20%27FOXI1%27%2C%20%27Mutation%27%3A%20%27c.517C%20%3E%20A%27%2C%20%27Protein_Change%27%3A%20%27p.His173Asn%27%2C%20%27Impact%27%3A%20%27Moderate%27%2C%20%27Read_Depth%27%3A%20%27366%2F324%27%7D%2C%0A%20%20%20%20%7B%27Gene%27%3A%20%27KRT6C%27%2C%20%27Mutation%27%3A%20%27c.913G%20%3E%20C%27%2C%20%27Protein_Change%27%3A%20%27p.Glu305Gln%27%2C%20%27Impact%27%3A%20%27Moderate%27%2C%20%27Read_Depth%27%3A%20%27498%2F420%27%7D%2C%0A%20%20%20%20%7B%27Gene%27%3A%20%27GJD2%27%2C%20%27Mutation%27%3A%20%27c.517G%20%3E%20T%27%2C%20%27Protein_Change%27%3A%20%27p.Glu173%2A%27%2C%20%27Impact%27%3A%20%27High%27%2C%20%27Read_Depth%27%3A%20%27187%2F164%27%7D%0A%5D%0A%0Adf%20%3D%20pd.DataFrame%28data%29%0A%0A%23%20Display%20the%20DataFrame%0Aprint%28df%29%0A%0A%23%20Plot%20a%20simple%20bar%20plot%20for%20variant%20allele%20frequencies%20%28if%20available%29%20or%20visualize%20mutation%20types%0Afig%20%3D%20px.bar%28df%2C%20x%3D%27Gene%27%2C%20y%3D%5B%27Read_Depth%27%5D%2C%20title%3D%27Mutation%20Read%20Depths%27%2C%20labels%3D%7B%27value%27%3A%20%27Read%20Depth%27%2C%20%27variable%27%3A%20%27Metric%27%7D%29%0Afig.show%28%29%0A%0AThe%20above%20code%20constructs%20a%20table%20from%20the%20mutation%20data%20and%20uses%20Plotly%20to%20create%20a%20bar%20plot.%20This%20analysis%20is%20useful%20to%20visually%20compare%20the%20read%20depths%20and%20infer%20reliability%20of%20detection.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Whole%20exome%20sequencing%20identified%20mutations%20of%20forkhead%20box%20I%201%20%28FOXI1%29%2C%20keratin%206%C2%A0C%20%28KRT6C%29%20and%20gap%20junction%20protein%20delta%202%20%28GJD2%29%20in%20a%20low-grade%20oncocytic%20tumor%20of%20the%20kidney%3A%20a%20case%20report)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code section downloads relevant mutation data and generates a detailed mutation table to aid visualization and further analysis.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-24", "tags": ["FOXI1 mutations kidney tumor", "biology", "bioinformatics"]}}