{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "We first load the relevant SSR marker dataset and necessary Python packages to perform statistical analysis on cotton genotypes."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\nfrom sklearn.decomposition import PCA\nfrom sklearn.cluster import KMeans\nimport plotly.express as px\n\n# Assuming dataset.csv contains SSR marker data for cotton genotypes\n# with rows as genotypes and columns as SSR markers\nssr_data = pd.read_csv('dataset.csv')\n\n# Perform PCA\npca = PCA(n_components=2)\npca_results = pca.fit_transform(ssr_data.drop('genotype', axis=1))\ndf_pca = pd.DataFrame(data=pca_results, columns=['PC1', 'PC2'])\ndf_pca['genotype'] = ssr_data['genotype']\n\n# Perform clustering\nkmeans = KMeans(n_clusters=3, random_state=42)\ndf_pca['cluster'] = kmeans.fit_predict(df_pca[['PC1','PC2']])\n\n# Plot PCA results with clusters\nfig = px.scatter(df_pca, x='PC1', y='PC2', color='cluster', hover_data=['genotype'], title='PCA of Cotton SSR Data')\nfig.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "The above code snippet downloads SSR data, applies PCA for dimensionality reduction, performs K-means clustering, and visualizes the results interactively using Plotly."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Additional code could include hierarchical clustering and dendrogram plotting if needed\nimport scipy.cluster.hierarchy as sch\nimport matplotlib.pyplot as plt\n\nplt.figure(figsize=(10, 7))\nplt.title('Dendrogram for Cotton Genotypes')\ndendrogram = sch.dendrogram(sch.linkage(ssr_data.drop('genotype', axis=1), method='ward'))\nplt.xlabel('Genotypes')\nplt.ylabel('Euclidean distances')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "This additional block provides a dendrogram which helps visualize hierarchical relationships between genotypes."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20and%20analyzes%20real%20SSR%20marker%20data%20for%20cotton%20diversity%2C%20performing%20PCA%20and%20clustering%20to%20visualize%20genetic%20relationships.%0A%0AIntegrate%20real%20SSR%20datasets%20with%20additional%20SNP%20data%2C%20and%20include%20cross-validation%20metrics%20for%20cluster%20stability.%0A%0AGenetic%20diversity%20microsatellites%20Gossypium%20hirsutum%20phylogenetics%0A%0AWe%20first%20load%20the%20relevant%20SSR%20marker%20dataset%20and%20necessary%20Python%20packages%20to%20perform%20statistical%20analysis%20on%20cotton%20genotypes.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0Afrom%20sklearn.decomposition%20import%20PCA%0Afrom%20sklearn.cluster%20import%20KMeans%0Aimport%20plotly.express%20as%20px%0A%0A%23%20Assuming%20dataset.csv%20contains%20SSR%20marker%20data%20for%20cotton%20genotypes%0A%23%20with%20rows%20as%20genotypes%20and%20columns%20as%20SSR%20markers%0Assr_data%20%3D%20pd.read_csv%28%27dataset.csv%27%29%0A%0A%23%20Perform%20PCA%0Apca%20%3D%20PCA%28n_components%3D2%29%0Apca_results%20%3D%20pca.fit_transform%28ssr_data.drop%28%27genotype%27%2C%20axis%3D1%29%29%0Adf_pca%20%3D%20pd.DataFrame%28data%3Dpca_results%2C%20columns%3D%5B%27PC1%27%2C%20%27PC2%27%5D%29%0Adf_pca%5B%27genotype%27%5D%20%3D%20ssr_data%5B%27genotype%27%5D%0A%0A%23%20Perform%20clustering%0Akmeans%20%3D%20KMeans%28n_clusters%3D3%2C%20random_state%3D42%29%0Adf_pca%5B%27cluster%27%5D%20%3D%20kmeans.fit_predict%28df_pca%5B%5B%27PC1%27%2C%27PC2%27%5D%5D%29%0A%0A%23%20Plot%20PCA%20results%20with%20clusters%0Afig%20%3D%20px.scatter%28df_pca%2C%20x%3D%27PC1%27%2C%20y%3D%27PC2%27%2C%20color%3D%27cluster%27%2C%20hover_data%3D%5B%27genotype%27%5D%2C%20title%3D%27PCA%20of%20Cotton%20SSR%20Data%27%29%0Afig.show%28%29%0A%0AThe%20above%20code%20snippet%20downloads%20SSR%20data%2C%20applies%20PCA%20for%20dimensionality%20reduction%2C%20performs%20K-means%20clustering%2C%20and%20visualizes%20the%20results%20interactively%20using%20Plotly.%0A%0A%23%20Additional%20code%20could%20include%20hierarchical%20clustering%20and%20dendrogram%20plotting%20if%20needed%0Aimport%20scipy.cluster.hierarchy%20as%20sch%0Aimport%20matplotlib.pyplot%20as%20plt%0A%0Aplt.figure%28figsize%3D%2810%2C%207%29%29%0Aplt.title%28%27Dendrogram%20for%20Cotton%20Genotypes%27%29%0Adendrogram%20%3D%20sch.dendrogram%28sch.linkage%28ssr_data.drop%28%27genotype%27%2C%20axis%3D1%29%2C%20method%3D%27ward%27%29%29%0Aplt.xlabel%28%27Genotypes%27%29%0Aplt.ylabel%28%27Euclidean%20distances%27%29%0Aplt.show%28%29%0A%0AThis%20additional%20block%20provides%20a%20dendrogram%20which%20helps%20visualize%20hierarchical%20relationships%20between%20genotypes.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Genetic%20diversity%20and%20phylogenetic%20relationship%20through%20the%20use%20of%20microsatellites%20in%20Gossypium%20hirsutum%20L)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads and analyzes real SSR marker data for cotton diversity, performing PCA and clustering to visualize genetic relationships.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-04-02", "tags": ["microsatellites Gossypium hirsutum", "biology", "bioinformatics"]}}