{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "This section explains the download and integration of multi-omics data extracted from the study, focusing on RNA-seq and proteomics to compute Pearson correlations and visualize the relationships."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\nfrom scipy.stats import pearsonr\n\n# Load pre-processed dataframes for transcriptomics and proteomics (df_tx and df_prot)\n# common_samples would be determined based on overlapping identifiers\ncommon_samples = set(df_tx.columns).intersection(df_prot.columns)\nresults = []\nfor gene in df_tx.index:\n    if gene in df_prot.index:\n        corr, pval = pearsonr(df_tx.loc[gene, list(common_samples)], df_prot.loc[gene, list(common_samples)])\n        results.append({'Gene': gene, 'Correlation': corr, 'p-value': pval})\ndf_results = pd.DataFrame(results)\ndf_results.sort_values(by='Correlation', ascending=False, inplace=True)\n# Display top correlations\nprint(df_results.head())"}, {"cell_type": "markdown", "metadata": {}, "source": "The above Python code calculates the correlation between RNA and protein expression levels across shared samples, which aids in identifying key genes associated with CIN phenotypes."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import plotly.express as px\nfig = px.scatter(df_results, x='Correlation', y='p-value', hover_data=['Gene'], title='Transcript-Protein Correlation in CRC PDOs')\nfig.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20notebook%20processes%20and%20integrates%20transcriptomics%20and%20proteomics%20data%20from%20CRC%20PDOs%20to%20correlate%20expression%20with%20CIN%20levels.%0A%0AIncorporate%20direct%20dataset%20URLs%20and%20extend%20analysis%20with%20network-based%20approaches%20to%20integrate%20additional%20genomic%20features.%0A%0AChromosomal%20instability%20CRC%20omics%20organoids%20review%0A%0AThis%20section%20explains%20the%20download%20and%20integration%20of%20multi-omics%20data%20extracted%20from%20the%20study%2C%20focusing%20on%20RNA-seq%20and%20proteomics%20to%20compute%20Pearson%20correlations%20and%20visualize%20the%20relationships.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0Afrom%20scipy.stats%20import%20pearsonr%0A%0A%23%20Load%20pre-processed%20dataframes%20for%20transcriptomics%20and%20proteomics%20%28df_tx%20and%20df_prot%29%0A%23%20common_samples%20would%20be%20determined%20based%20on%20overlapping%20identifiers%0Acommon_samples%20%3D%20set%28df_tx.columns%29.intersection%28df_prot.columns%29%0Aresults%20%3D%20%5B%5D%0Afor%20gene%20in%20df_tx.index%3A%0A%20%20%20%20if%20gene%20in%20df_prot.index%3A%0A%20%20%20%20%20%20%20%20corr%2C%20pval%20%3D%20pearsonr%28df_tx.loc%5Bgene%2C%20list%28common_samples%29%5D%2C%20df_prot.loc%5Bgene%2C%20list%28common_samples%29%5D%29%0A%20%20%20%20%20%20%20%20results.append%28%7B%27Gene%27%3A%20gene%2C%20%27Correlation%27%3A%20corr%2C%20%27p-value%27%3A%20pval%7D%29%0Adf_results%20%3D%20pd.DataFrame%28results%29%0Adf_results.sort_values%28by%3D%27Correlation%27%2C%20ascending%3DFalse%2C%20inplace%3DTrue%29%0A%23%20Display%20top%20correlations%0Aprint%28df_results.head%28%29%29%0A%0AThe%20above%20Python%20code%20calculates%20the%20correlation%20between%20RNA%20and%20protein%20expression%20levels%20across%20shared%20samples%2C%20which%20aids%20in%20identifying%20key%20genes%20associated%20with%20CIN%20phenotypes.%0A%0Aimport%20plotly.express%20as%20px%0Afig%20%3D%20px.scatter%28df_results%2C%20x%3D%27Correlation%27%2C%20y%3D%27p-value%27%2C%20hover_data%3D%5B%27Gene%27%5D%2C%20title%3D%27Transcript-Protein%20Correlation%20in%20CRC%20PDOs%27%29%0Afig.show%28%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Decoding%20chromosomal%20instability%20insights%20in%20CRC%20by%20integrating%20omics%20and%20patient-derived%20organoids)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This notebook processes and integrates transcriptomics and proteomics data from CRC PDOs to correlate expression with CIN levels.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-11", "tags": ["chromosomal instability CRC", "biology", "bioinformatics"]}}