{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "In this notebook, we will load scRNA-seq datasets from the Broad Single Cell Portal (e.g., SCP2670) to analyze changes in T-cell activation markers following treatment with the bispecific antibody."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nadata = sc.read_h5ad('path_to_scRNAseq_data.h5ad')\n# Perform clustering and calculate marker expression differences\nsc.tl.rank_genes_groups(adata, 'treatment_group', method='wilcoxon')\nsc.pl.rank_genes_groups(adata, n_genes=20, sharey=False)"}, {"cell_type": "markdown", "metadata": {}, "source": "The code above highlights differential marker expression that can be tied to T-cell activation states, providing further insights into the antibody\u2019s mechanism of action."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Further analysis can include integration with known gene sets for T-cell function\nimport gseapy as gp\ngene_list = adata.uns['rank_genes_groups']['names']['treated'].tolist()\n# Perform enrichment analysis using gseapy\nenr = gp.enrichr(gene_list=gene_list, gene_sets='KEGG_2016', outdir='test/enrichr_kegg')\nprint(enr.results.head())"}, {"cell_type": "markdown", "metadata": {}, "source": "This analysis integrates scRNA-seq data with pathway enrichment to visualize impacts on immune signaling."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Plotting UMAP for visualization\nsc.pp.neighbors(adata, n_neighbors=10)\nsc.tl.umap(adata)\nsc.pl.umap(adata, color=['CD8A', 'GZMB', 'PRF1'], save='umap_activation.png')"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20Python3%20notebook%20code%20retrieves%20and%20analyzes%20single-cell%20RNA-seq%20data%20to%20assess%20immune%20cell%20infiltration%20differences%20in%20AGR2%20positive%20versus%20negative%20tumors%20based%20on%20bispecific%20antibody%20treatment.%0A%0AIntegrate%20additional%20datasets%20from%20multiple%20tumor%20types%20and%20refine%20cell%20type%20annotation%20methods%20for%20enhanced%20resolution.%0A%0ABispecific%20antibody%20AGR2%20PD1%20T-cell%20antitumor%20response%20PDL1%20upregulation%0A%0AIn%20this%20notebook%2C%20we%20will%20load%20scRNA-seq%20datasets%20from%20the%20Broad%20Single%20Cell%20Portal%20%28e.g.%2C%20SCP2670%29%20to%20analyze%20changes%20in%20T-cell%20activation%20markers%20following%20treatment%20with%20the%20bispecific%20antibody.%0A%0Aimport%20scanpy%20as%20sc%0Aadata%20%3D%20sc.read_h5ad%28%27path_to_scRNAseq_data.h5ad%27%29%0A%23%20Perform%20clustering%20and%20calculate%20marker%20expression%20differences%0Asc.tl.rank_genes_groups%28adata%2C%20%27treatment_group%27%2C%20method%3D%27wilcoxon%27%29%0Asc.pl.rank_genes_groups%28adata%2C%20n_genes%3D20%2C%20sharey%3DFalse%29%0A%0AThe%20code%20above%20highlights%20differential%20marker%20expression%20that%20can%20be%20tied%20to%20T-cell%20activation%20states%2C%20providing%20further%20insights%20into%20the%20antibody%E2%80%99s%20mechanism%20of%20action.%0A%0A%23%20Further%20analysis%20can%20include%20integration%20with%20known%20gene%20sets%20for%20T-cell%20function%0Aimport%20gseapy%20as%20gp%0Agene_list%20%3D%20adata.uns%5B%27rank_genes_groups%27%5D%5B%27names%27%5D%5B%27treated%27%5D.tolist%28%29%0A%23%20Perform%20enrichment%20analysis%20using%20gseapy%0Aenr%20%3D%20gp.enrichr%28gene_list%3Dgene_list%2C%20gene_sets%3D%27KEGG_2016%27%2C%20outdir%3D%27test%2Fenrichr_kegg%27%29%0Aprint%28enr.results.head%28%29%29%0A%0AThis%20analysis%20integrates%20scRNA-seq%20data%20with%20pathway%20enrichment%20to%20visualize%20impacts%20on%20immune%20signaling.%0A%0A%23%20Plotting%20UMAP%20for%20visualization%0Asc.pp.neighbors%28adata%2C%20n_neighbors%3D10%29%0Asc.tl.umap%28adata%29%0Asc.pl.umap%28adata%2C%20color%3D%5B%27CD8A%27%2C%20%27GZMB%27%2C%20%27PRF1%27%5D%2C%20save%3D%27umap_activation.png%27%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Bispecific%20antibody%20simultaneously%20targeting%20AGR2%20and%20PD1%20mediates%20cytotoxic%20T-cell-induced%20antitumor%20response%20in%20AGR2-dependent%20manner%20and%20inhibits%20AGR2-induced%20PDL1%20upregulation)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This Python3 notebook code retrieves and analyzes single-cell RNA-seq data to assess immune cell infiltration differences in AGR2 positive versus negative tumors based on bispecific antibody treatment.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-24", "tags": ["AGR2 PD1 bispecific antibody", "biology", "bioinformatics"]}}