{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "Step 1: Import libraries and load the proteomic dataset to assess unique proteins across sample groups."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport matplotlib.pyplot as plt\n\n# Load proteomic data (assumed file: proteomics_data.csv)\ndata = pd.read_csv('proteomics_data.csv')\n\n# Group by sample type and count unique proteins\nprotein_counts = data.groupby('sample_type')['protein_id'].nunique()\n\n# Plot unique proteins per sample type\nprotein_counts.plot(kind='bar', color='#6A0C76', title='Unique Proteins by Sample Type')\nplt.xlabel('Sample Type')\nplt.ylabel('Unique Protein Count')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "Step 2: Visualize the distribution of protein families across sample types to highlight differential expression."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import seaborn as sns\nsns.set_theme(style='whitegrid')\n\n# Aggregate counts of protein families by sample type\nfamily_distribution = data.groupby(['sample_type', 'protein_family']).size().reset_index(name='counts')\n\n# Create a bar plot using seaborn\nplt.figure(figsize=(10,6))\nax = sns.barplot(x='sample_type', y='counts', hue='protein_family', data=family_distribution, palette='deep')\nplt.title('Protein Family Distribution by Sample Type')\nplt.xlabel('Sample Type')\nplt.ylabel('Counts')\nplt.legend(title='Protein Family')\nplt.show()\n"}, {"cell_type": "markdown", "metadata": {}, "source": "This notebook illustrates key aspects of the proteomic dataset, reinforcing the study's discovery of tissue-specific expression patterns."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20analyzes%20proteomic%20datasets%20to%20compare%20protein%20abundances%20across%20sample%20types%20and%20visualizes%20key%20functional%20annotations.%0A%0AAdd%20error%20handling%2C%20data%20validation%20steps%2C%20and%20integrate%20functional%20enrichment%20analysis%20for%20deeper%20insights.%0A%0AProteomic%20analysis%20Actinia%20fragacea%20nematocyst%20venom%20mucus%20tissue%20profiles%0A%0AStep%201%3A%20Import%20libraries%20and%20load%20the%20proteomic%20dataset%20to%20assess%20unique%20proteins%20across%20sample%20groups.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20matplotlib.pyplot%20as%20plt%0A%0A%23%20Load%20proteomic%20data%20%28assumed%20file%3A%20proteomics_data.csv%29%0Adata%20%3D%20pd.read_csv%28%27proteomics_data.csv%27%29%0A%0A%23%20Group%20by%20sample%20type%20and%20count%20unique%20proteins%0Aprotein_counts%20%3D%20data.groupby%28%27sample_type%27%29%5B%27protein_id%27%5D.nunique%28%29%0A%0A%23%20Plot%20unique%20proteins%20per%20sample%20type%0Aprotein_counts.plot%28kind%3D%27bar%27%2C%20color%3D%27%236A0C76%27%2C%20title%3D%27Unique%20Proteins%20by%20Sample%20Type%27%29%0Aplt.xlabel%28%27Sample%20Type%27%29%0Aplt.ylabel%28%27Unique%20Protein%20Count%27%29%0Aplt.show%28%29%0A%0AStep%202%3A%20Visualize%20the%20distribution%20of%20protein%20families%20across%20sample%20types%20to%20highlight%20differential%20expression.%0A%0Aimport%20seaborn%20as%20sns%0Asns.set_theme%28style%3D%27whitegrid%27%29%0A%0A%23%20Aggregate%20counts%20of%20protein%20families%20by%20sample%20type%0Afamily_distribution%20%3D%20data.groupby%28%5B%27sample_type%27%2C%20%27protein_family%27%5D%29.size%28%29.reset_index%28name%3D%27counts%27%29%0A%0A%23%20Create%20a%20bar%20plot%20using%20seaborn%0Aplt.figure%28figsize%3D%2810%2C6%29%29%0Aax%20%3D%20sns.barplot%28x%3D%27sample_type%27%2C%20y%3D%27counts%27%2C%20hue%3D%27protein_family%27%2C%20data%3Dfamily_distribution%2C%20palette%3D%27deep%27%29%0Aplt.title%28%27Protein%20Family%20Distribution%20by%20Sample%20Type%27%29%0Aplt.xlabel%28%27Sample%20Type%27%29%0Aplt.ylabel%28%27Counts%27%29%0Aplt.legend%28title%3D%27Protein%20Family%27%29%0Aplt.show%28%29%0A%0A%0AThis%20notebook%20illustrates%20key%20aspects%20of%20the%20proteomic%20dataset%2C%20reinforcing%20the%20study%27s%20discovery%20of%20tissue-specific%20expression%20patterns.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Proteomic%20Diversity%20of%20the%20Sea%20Anemone%20Actinia%20fragacea%3A%20Comparative%20Analysis%20of%20Nematocyst%20Venom%2C%20Mucus%2C%20and%20Tissue-Specific%20Profiles)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code analyzes proteomic datasets to compare protein abundances across sample types and visualizes key functional annotations.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-19", "tags": ["Actinia fragacea proteomics", "biology", "bioinformatics"]}}