{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "### Step 1: Import Libraries\nImport necessary libraries for data analysis and visualization."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\nimport matplotlib.pyplot as plt\nimport seaborn as sns"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 2: Load OGM Data\nLoad the OGM dataset containing structural variants and HRD status."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "ogm_data = pd.read_csv('path_to_ogm_data.csv')\nprint(ogm_data.head())"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 3: Analyze Structural Variants\nAnalyze the structural variants to identify patterns associated with HRD."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "hrd_cases = ogm_data[ogm_data['HRD_status'] == 'HRD']\nnon_hrd_cases = ogm_data[ogm_data['HRD_status'] == 'non-HRD']\n\n# Count structural variants\nhrd_sv_count = hrd_cases['SV_type'].value_counts()\nnon_hrd_sv_count = non_hrd_cases['SV_type'].value_counts()"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 4: Visualize Results\nVisualize the distribution of structural variants in HRD vs non-HRD cases."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "plt.figure(figsize=(12,6))\nsns.barplot(x=hrd_sv_count.index, y=hrd_sv_count.values, color='blue', label='HRD')\nsns.barplot(x=non_hrd_sv_count.index, y=non_hrd_sv_count.values, color='red', label='non-HRD')\nplt.title('Structural Variants in HRD vs non-HRD Cases')\nplt.xlabel('Structural Variant Type')\nplt.ylabel('Count')\nplt.legend()\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 5: Discussion\nDiscuss the implications of the findings in the context of HRD detection."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Further analysis and discussion can be added here."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20analyzes%20structural%20variants%20from%20OGM%20data%20to%20identify%20HRD%20in%20low%20purity%20samples%20using%20relevant%20datasets.%0A%0AIncorporate%20additional%20datasets%20for%20validation%20and%20enhance%20the%20analysis%20with%20machine%20learning%20techniques%20to%20predict%20HRD%20status%20more%20accurately.%0A%0AOptical%20genome%20mapping%20vs%20NGS%20HRD%20detection%20low%20purity%0A%0A%23%23%23%20Step%201%3A%20Import%20Libraries%0AImport%20necessary%20libraries%20for%20data%20analysis%20and%20visualization.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0Aimport%20matplotlib.pyplot%20as%20plt%0Aimport%20seaborn%20as%20sns%0A%0A%23%23%23%20Step%202%3A%20Load%20OGM%20Data%0ALoad%20the%20OGM%20dataset%20containing%20structural%20variants%20and%20HRD%20status.%0A%0Aogm_data%20%3D%20pd.read_csv%28%27path_to_ogm_data.csv%27%29%0Aprint%28ogm_data.head%28%29%29%0A%0A%23%23%23%20Step%203%3A%20Analyze%20Structural%20Variants%0AAnalyze%20the%20structural%20variants%20to%20identify%20patterns%20associated%20with%20HRD.%0A%0Ahrd_cases%20%3D%20ogm_data%5Bogm_data%5B%27HRD_status%27%5D%20%3D%3D%20%27HRD%27%5D%0Anon_hrd_cases%20%3D%20ogm_data%5Bogm_data%5B%27HRD_status%27%5D%20%3D%3D%20%27non-HRD%27%5D%0A%0A%23%20Count%20structural%20variants%0Ahrd_sv_count%20%3D%20hrd_cases%5B%27SV_type%27%5D.value_counts%28%29%0Anon_hrd_sv_count%20%3D%20non_hrd_cases%5B%27SV_type%27%5D.value_counts%28%29%0A%0A%23%23%23%20Step%204%3A%20Visualize%20Results%0AVisualize%20the%20distribution%20of%20structural%20variants%20in%20HRD%20vs%20non-HRD%20cases.%0A%0Aplt.figure%28figsize%3D%2812%2C6%29%29%0Asns.barplot%28x%3Dhrd_sv_count.index%2C%20y%3Dhrd_sv_count.values%2C%20color%3D%27blue%27%2C%20label%3D%27HRD%27%29%0Asns.barplot%28x%3Dnon_hrd_sv_count.index%2C%20y%3Dnon_hrd_sv_count.values%2C%20color%3D%27red%27%2C%20label%3D%27non-HRD%27%29%0Aplt.title%28%27Structural%20Variants%20in%20HRD%20vs%20non-HRD%20Cases%27%29%0Aplt.xlabel%28%27Structural%20Variant%20Type%27%29%0Aplt.ylabel%28%27Count%27%29%0Aplt.legend%28%29%0Aplt.show%28%29%0A%0A%23%23%23%20Step%205%3A%20Discussion%0ADiscuss%20the%20implications%20of%20the%20findings%20in%20the%20context%20of%20HRD%20detection.%0A%0A%23%20Further%20analysis%20and%20discussion%20can%20be%20added%20here.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=What%20is%20the%20potential%20of%20optical%20genome%20mapping%20compared%20to%20NGS%20in%20detecting%20HRD%20in%20low%20purity%20samples%3F)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code analyzes structural variants from OGM data to identify HRD in low purity samples using relevant datasets.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-11", "tags": ["optical genome mapping HRD", "biology", "bioinformatics"]}}