{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "Below is a Python code snippet that uses the Plotly library to generate an interactive bar graph of sequencing depth across different genomic elements for the Bronze Age Y. pestis samples. This visualization facilitates comparison between chromosome and plasmid coverages."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "\nimport plotly.graph_objects as go\n\n# Data extracted from the study\nsamples = ['RISE00', 'RISE139', 'RISE386', 'RISE397', 'RISE505', 'RISE509', 'RISE511']\nco92 = [0.39, 0.14, 0.82, 0.25, 8.73, 29.45, 0.20]\npmt1 = [0.36, 0.24, 0.96, 0.40, 9.15, 16.96, 0.24]\npPCP1 = [1.40, 0.76, 1.12, 6.88, 34.09, 31.22, 1.19]\npCD1 = [0.66, 0.28, 1.60, 0.50, 17.46, 50.32, 0.60]\n\nfig = go.Figure(data=[\n    go.Bar(name='CO92 Chromosome', x=samples, y=co92),\n    go.Bar(name='pMT1 Plasmid', x=samples, y=pmt1),\n    go.Bar(name='pPCP1 Plasmid', x=samples, y=pPCP1),\n    go.Bar(name='pCD1 Plasmid', x=samples, y=pCD1)\n])\n\n# Change the bar mode\nfig.update_layout(barmode='group', title='Sequencing Depth Across Genomic Elements', \n                  xaxis_title='Sample', yaxis_title='Sequencing Depth (X)')\n\nfig.show()\n"}, {"cell_type": "markdown", "metadata": {}, "source": "The above code creates a grouped bar chart that displays the sequencing depth on the chromosome and three plasmids for each ancient sample. This helps in understanding the differences in genomic coverage that reflect various genetic decay patterns."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# For further analysis, one could integrate the coverage data with phylogenetic information\nimport pandas as pd\n\ndf = pd.DataFrame({\n    'Sample': samples,\n    'CO92': co92,\n    'pMT1': pmt1,\n    'pPCP1': pPCP1,\n    'pCD1': pCD1\n})\n\n# Display the data table\nprint(df)\n"}, {"cell_type": "markdown", "metadata": {}, "source": "This notebook not only visualizes the sequencing depth but also provides a concise table helpful for downstream comparative genomic analyses."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# End of the analysis code block"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20the%20extracted%20sequencing%20depth%20dataset%20and%20generates%20an%20interactive%20Plotly%20bar%20graph%20to%20visualize%20genomic%20coverage%20across%20samples%2C%20aiding%20in%20comparative%20analysis.%0A%0AInclude%20error%20handling%20for%20data%20input%20and%20expand%20to%20integrate%20temporal%20data%20for%20dynamic%20evolutionary%20modeling.%0A%0ABronze%20Age%20Yersinia%20pestis%20genome%20evolution%20hosts%20prehistoric%20plague%0A%0ABelow%20is%20a%20Python%20code%20snippet%20that%20uses%20the%20Plotly%20library%20to%20generate%20an%20interactive%20bar%20graph%20of%20sequencing%20depth%20across%20different%20genomic%20elements%20for%20the%20Bronze%20Age%20Y.%20pestis%20samples.%20This%20visualization%20facilitates%20comparison%20between%20chromosome%20and%20plasmid%20coverages.%0A%0A%0Aimport%20plotly.graph_objects%20as%20go%0A%0A%23%20Data%20extracted%20from%20the%20study%0Asamples%20%3D%20%5B%27RISE00%27%2C%20%27RISE139%27%2C%20%27RISE386%27%2C%20%27RISE397%27%2C%20%27RISE505%27%2C%20%27RISE509%27%2C%20%27RISE511%27%5D%0Aco92%20%3D%20%5B0.39%2C%200.14%2C%200.82%2C%200.25%2C%208.73%2C%2029.45%2C%200.20%5D%0Apmt1%20%3D%20%5B0.36%2C%200.24%2C%200.96%2C%200.40%2C%209.15%2C%2016.96%2C%200.24%5D%0ApPCP1%20%3D%20%5B1.40%2C%200.76%2C%201.12%2C%206.88%2C%2034.09%2C%2031.22%2C%201.19%5D%0ApCD1%20%3D%20%5B0.66%2C%200.28%2C%201.60%2C%200.50%2C%2017.46%2C%2050.32%2C%200.60%5D%0A%0Afig%20%3D%20go.Figure%28data%3D%5B%0A%20%20%20%20go.Bar%28name%3D%27CO92%20Chromosome%27%2C%20x%3Dsamples%2C%20y%3Dco92%29%2C%0A%20%20%20%20go.Bar%28name%3D%27pMT1%20Plasmid%27%2C%20x%3Dsamples%2C%20y%3Dpmt1%29%2C%0A%20%20%20%20go.Bar%28name%3D%27pPCP1%20Plasmid%27%2C%20x%3Dsamples%2C%20y%3DpPCP1%29%2C%0A%20%20%20%20go.Bar%28name%3D%27pCD1%20Plasmid%27%2C%20x%3Dsamples%2C%20y%3DpCD1%29%0A%5D%29%0A%0A%23%20Change%20the%20bar%20mode%0Afig.update_layout%28barmode%3D%27group%27%2C%20title%3D%27Sequencing%20Depth%20Across%20Genomic%20Elements%27%2C%20%0A%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20%20xaxis_title%3D%27Sample%27%2C%20yaxis_title%3D%27Sequencing%20Depth%20%28X%29%27%29%0A%0Afig.show%28%29%0A%0A%0AThe%20above%20code%20creates%20a%20grouped%20bar%20chart%20that%20displays%20the%20sequencing%20depth%20on%20the%20chromosome%20and%20three%20plasmids%20for%20each%20ancient%20sample.%20This%20helps%20in%20understanding%20the%20differences%20in%20genomic%20coverage%20that%20reflect%20various%20genetic%20decay%20patterns.%0A%0A%23%20For%20further%20analysis%2C%20one%20could%20integrate%20the%20coverage%20data%20with%20phylogenetic%20information%0Aimport%20pandas%20as%20pd%0A%0Adf%20%3D%20pd.DataFrame%28%7B%0A%20%20%20%20%27Sample%27%3A%20samples%2C%0A%20%20%20%20%27CO92%27%3A%20co92%2C%0A%20%20%20%20%27pMT1%27%3A%20pmt1%2C%0A%20%20%20%20%27pPCP1%27%3A%20pPCP1%2C%0A%20%20%20%20%27pCD1%27%3A%20pCD1%0A%7D%29%0A%0A%23%20Display%20the%20data%20table%0Aprint%28df%29%0A%0A%0AThis%20notebook%20not%20only%20visualizes%20the%20sequencing%20depth%20but%20also%20provides%20a%20concise%20table%20helpful%20for%20downstream%20comparative%20genomic%20analyses.%0A%0A%23%20End%20of%20the%20analysis%20code%20block%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Bronze%20AgeYersinia%20pestisgenome%20from%20sheep%20sheds%20light%20on%20hosts%20and%20evolution%20of%20a%20prehistoric%20plague%20lineage)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads the extracted sequencing depth dataset and generates an interactive Plotly bar graph to visualize genomic coverage across samples, aiding in comparative analysis.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-12", "tags": ["Yersinia pestis Bronze Age", "biology", "bioinformatics"]}}