{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "The following notebook section downloads the skin virome dataset and performs differential abundance and clustering analysis using Python libraries such as pandas, scipy, and seaborn."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\nimport seaborn as sns\nimport matplotlib.pyplot as plt\nfrom scipy.stats import f_oneway\n\n# Load the skin virome data (real dataset URL provided in research data links)\ndata = pd.read_csv('https://raw.githubusercontent.com/lizhiming11/skin_virus/main/skin_virome_data.csv')\n\n# Group data by skin microenvironment\ngrouped = data.groupby('microenvironment')\n\n# Perform ANOVA to test differences between microenvironments\ngroups = [grp['viral_abundance'].values for name, grp in grouped]\nanova_result = f_oneway(*groups)\nprint('ANOVA F-statistic:', anova_result.statistic, 'p-value:', anova_result.pvalue)\n\n# Create a boxplot for visual comparison\nplt.figure(figsize=(10,6))\nsns.boxplot(x='microenvironment', y='viral_abundance', data=data, palette='Purples')\nplt.title('Viral Abundance across Skin Microenvironments')\nplt.xlabel('Skin Microenvironment')\nplt.ylabel('Viral Abundance')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "This code block provides statistical analysis of viral abundance across different skin microenvironments, visualizing results to support the paper's claims."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Additional analysis: Clustering of viral operational taxonomic units (vOTUs)\nfrom sklearn.cluster import KMeans\n\n# Assume 'data_features' holds feature data for each vOTU\ndata_features = data.filter(regex='feature_')\n\nkmeans = KMeans(n_clusters=3, random_state=42).fit(data_features)\ndata['cluster'] = kmeans.labels_\n\n# Visualize clusters using a scatter plot of two selected features\na = plt.scatter(data['feature_1'], data['feature_2'], c=data['cluster'], cmap='viridis')\nplt.xlabel('Feature 1')\nplt.ylabel('Feature 2')\nplt.title('Clustering of vOTUs based on Feature Data')\nplt.colorbar(a)\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "This additional clustering analysis helps identify distinct groups within the virome which could correlate with microenvironmental adaptations."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20performs%20differential%20abundance%20analysis%20on%20skin%20virome%20datasets%20using%20real%20metagenomic%20samples%2C%20enabling%20cluster%20identification%20by%20microenvironment.%0A%0AIncorporate%20multi-omics%20data%20integration%20and%20time-series%20analysis%20to%20further%20refine%20the%20impact%20of%20host%20factors%20on%20virome%20dynamics.%0A%0AHuman%20skin%20DNA%20virome%20catalog%20viral%20diversity%20microenvironmental%20influences%0A%0AThe%20following%20notebook%20section%20downloads%20the%20skin%20virome%20dataset%20and%20performs%20differential%20abundance%20and%20clustering%20analysis%20using%20Python%20libraries%20such%20as%20pandas%2C%20scipy%2C%20and%20seaborn.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0Aimport%20seaborn%20as%20sns%0Aimport%20matplotlib.pyplot%20as%20plt%0Afrom%20scipy.stats%20import%20f_oneway%0A%0A%23%20Load%20the%20skin%20virome%20data%20%28real%20dataset%20URL%20provided%20in%20research%20data%20links%29%0Adata%20%3D%20pd.read_csv%28%27https%3A%2F%2Fraw.githubusercontent.com%2Flizhiming11%2Fskin_virus%2Fmain%2Fskin_virome_data.csv%27%29%0A%0A%23%20Group%20data%20by%20skin%20microenvironment%0Agrouped%20%3D%20data.groupby%28%27microenvironment%27%29%0A%0A%23%20Perform%20ANOVA%20to%20test%20differences%20between%20microenvironments%0Agroups%20%3D%20%5Bgrp%5B%27viral_abundance%27%5D.values%20for%20name%2C%20grp%20in%20grouped%5D%0Aanova_result%20%3D%20f_oneway%28%2Agroups%29%0Aprint%28%27ANOVA%20F-statistic%3A%27%2C%20anova_result.statistic%2C%20%27p-value%3A%27%2C%20anova_result.pvalue%29%0A%0A%23%20Create%20a%20boxplot%20for%20visual%20comparison%0Aplt.figure%28figsize%3D%2810%2C6%29%29%0Asns.boxplot%28x%3D%27microenvironment%27%2C%20y%3D%27viral_abundance%27%2C%20data%3Ddata%2C%20palette%3D%27Purples%27%29%0Aplt.title%28%27Viral%20Abundance%20across%20Skin%20Microenvironments%27%29%0Aplt.xlabel%28%27Skin%20Microenvironment%27%29%0Aplt.ylabel%28%27Viral%20Abundance%27%29%0Aplt.show%28%29%0A%0AThis%20code%20block%20provides%20statistical%20analysis%20of%20viral%20abundance%20across%20different%20skin%20microenvironments%2C%20visualizing%20results%20to%20support%20the%20paper%27s%20claims.%0A%0A%23%20Additional%20analysis%3A%20Clustering%20of%20viral%20operational%20taxonomic%20units%20%28vOTUs%29%0Afrom%20sklearn.cluster%20import%20KMeans%0A%0A%23%20Assume%20%27data_features%27%20holds%20feature%20data%20for%20each%20vOTU%0Adata_features%20%3D%20data.filter%28regex%3D%27feature_%27%29%0A%0Akmeans%20%3D%20KMeans%28n_clusters%3D3%2C%20random_state%3D42%29.fit%28data_features%29%0Adata%5B%27cluster%27%5D%20%3D%20kmeans.labels_%0A%0A%23%20Visualize%20clusters%20using%20a%20scatter%20plot%20of%20two%20selected%20features%0Aa%20%3D%20plt.scatter%28data%5B%27feature_1%27%5D%2C%20data%5B%27feature_2%27%5D%2C%20c%3Ddata%5B%27cluster%27%5D%2C%20cmap%3D%27viridis%27%29%0Aplt.xlabel%28%27Feature%201%27%29%0Aplt.ylabel%28%27Feature%202%27%29%0Aplt.title%28%27Clustering%20of%20vOTUs%20based%20on%20Feature%20Data%27%29%0Aplt.colorbar%28a%29%0Aplt.show%28%29%0A%0AThis%20additional%20clustering%20analysis%20helps%20identify%20distinct%20groups%20within%20the%20virome%20which%20could%20correlate%20with%20microenvironmental%20adaptations.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20A%20Comprehensive%20Reference%20Catalog%20of%20Human%20Skin%20DNA%20Virome%20Reveals%20Novel%20Viral%20Diversity%20and%20Microenvironmental%20Influences%29)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code performs differential abundance analysis on skin virome datasets using real metagenomic samples, enabling cluster identification by microenvironment.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-26", "tags": ["human skin DNA virome", "biology", "bioinformatics"]}}