{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "### Single-Cell RNA-Seq Analysis of X Chromosome Duplication\nAnalyzing the impact of the identified X chromosome duplication on gene expression in affected individuals."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nimport pandas as pd\n\n# Load the single-cell data\nadata = sc.read_10x_h5('path_to_scRNAseq_data.h5ad')\n\n# Preprocess the data\nsc.pp.filter_cells(adata, min_genes=200)\nsc.pp.filter_genes(adata, min_cells=3)\nsc.pp.normalize_total(adata, target_sum=1e4)\nsc.pp.log1p(adata)\n\n# Identify highly variable genes\nsc.pp.highly_variable_genes(adata, min_mean=0.0125, max_mean=3, min_disp=0.5)\nadata = adata[:, adata.var.highly_variable]\n\n# Scale the data\nsc.pp.scale(adata, max_value=10)\n\n# Perform PCA\nsc.tl.pca(adata, svd_solver='arpack')\n\n# Plot PCA\nsc.pl.pca(adata, color=['genotype'])"}, {"cell_type": "markdown", "metadata": {}, "source": "### Differential Expression Analysis\nIdentifying genes with significant expression changes between duplicated and non-duplicated regions."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "sc.tl.rank_genes_groups(adata, 'genotype', method='wilcoxon')\nsc.pl.rank_genes_groups(adata, n_genes=25, sharey=False)"}, {"cell_type": "markdown", "metadata": {}, "source": "### Results Interpretation\nAnalyzing the ranked genes to understand the molecular pathways affected by the duplication."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "de_genes = adata.uns['rank_genes_groups']['names']\nprint(de_genes)"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20Python3%20code%20analyzes%20gene%20expression%20data%20from%20single-cell%20RNA%20sequencing%20to%20identify%20differentially%20expressed%20genes%20due%20to%20X%20chromosome%20duplication.%0A%0AIncorporate%20additional%20normalization%20methods%20and%20validate%20findings%20with%20independent%20datasets%20to%20enhance%20the%20robustness%20of%20the%20differential%20expression%20analysis.%0A%0AX%20Chromosome%20duplication%20familial%20generalized%20dystonia%20case%20report%0A%0A%23%23%23%20Single-Cell%20RNA-Seq%20Analysis%20of%20X%20Chromosome%20Duplication%0AAnalyzing%20the%20impact%20of%20the%20identified%20X%20chromosome%20duplication%20on%20gene%20expression%20in%20affected%20individuals.%0A%0Aimport%20scanpy%20as%20sc%0Aimport%20pandas%20as%20pd%0A%0A%23%20Load%20the%20single-cell%20data%0Aadata%20%3D%20sc.read_10x_h5%28%27path_to_scRNAseq_data.h5ad%27%29%0A%0A%23%20Preprocess%20the%20data%0Asc.pp.filter_cells%28adata%2C%20min_genes%3D200%29%0Asc.pp.filter_genes%28adata%2C%20min_cells%3D3%29%0Asc.pp.normalize_total%28adata%2C%20target_sum%3D1e4%29%0Asc.pp.log1p%28adata%29%0A%0A%23%20Identify%20highly%20variable%20genes%0Asc.pp.highly_variable_genes%28adata%2C%20min_mean%3D0.0125%2C%20max_mean%3D3%2C%20min_disp%3D0.5%29%0Aadata%20%3D%20adata%5B%3A%2C%20adata.var.highly_variable%5D%0A%0A%23%20Scale%20the%20data%0Asc.pp.scale%28adata%2C%20max_value%3D10%29%0A%0A%23%20Perform%20PCA%0Asc.tl.pca%28adata%2C%20svd_solver%3D%27arpack%27%29%0A%0A%23%20Plot%20PCA%0Asc.pl.pca%28adata%2C%20color%3D%5B%27genotype%27%5D%29%0A%0A%23%23%23%20Differential%20Expression%20Analysis%0AIdentifying%20genes%20with%20significant%20expression%20changes%20between%20duplicated%20and%20non-duplicated%20regions.%0A%0Asc.tl.rank_genes_groups%28adata%2C%20%27genotype%27%2C%20method%3D%27wilcoxon%27%29%0Asc.pl.rank_genes_groups%28adata%2C%20n_genes%3D25%2C%20sharey%3DFalse%29%0A%0A%23%23%23%20Results%20Interpretation%0AAnalyzing%20the%20ranked%20genes%20to%20understand%20the%20molecular%20pathways%20affected%20by%20the%20duplication.%0A%0Ade_genes%20%3D%20adata.uns%5B%27rank_genes_groups%27%5D%5B%27names%27%5D%0Aprint%28de_genes%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20A%20Novel%20Large%20Duplication%20on%20the%20X%20Chromosome%20as%20a%20Cause%20of%20Familial%20Generalized%20Dystonia%3A%20A%20Case%20Report)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This Python3 code analyzes gene expression data from single-cell RNA sequencing to identify differentially expressed genes due to X chromosome duplication.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-01-29", "tags": ["X Chromosome duplication dystonia", "biology", "bioinformatics"]}}