{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "We begin by downloading the dataset from GEO and loading it using Scanpy. This dataset contains nuclei from PVAT samples."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nadata = sc.read_10x_mtx('path_to_PVAT_data/', var_names='gene_symbols', cache=True)\nadata.var_names_make_unique()\nsc.pp.filter_cells(adata, min_genes=200)\nsc.pp.filter_genes(adata, min_cells=3)\nsc.pp.normalize_total(adata, target_sum=1e4)\nsc.pp.log1p(adata)\nsc.pp.highly_variable_genes(adata, min_mean=0.0125, max_mean=3, min_disp=0.5)\nadata = adata[:, adata.var['highly_variable']]\nsc.pp.scale(adata, max_value=10)\nsc.tl.pca(adata, svd_solver='arpack')\nsc.pp.neighbors(adata, n_neighbors=10, n_pcs=40)\nsc.tl.umap(adata)\nsc.tl.leiden(adata)\nsc.pl.umap(adata, color=['leiden'])\n"}, {"cell_type": "markdown", "metadata": {}, "source": "Next, differential expression analysis is performed to identify cluster-specific marker genes that could indicate key transcription factors."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "sc.tl.rank_genes_groups(adata, 'leiden', method='wilcoxon')\nsc.pl.rank_genes_groups(adata, n_genes=20, sharey=False)\n"}, {"cell_type": "markdown", "metadata": {}, "source": "This notebook demonstrates a streamlined and reproducible pipeline to analyze single-nucleus RNA-seq data, which can help verify the paper's claims."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Further code could integrate CellChat or NicheNet analysis tools to evaluate ligand-receptor interactions\n# This section would require installation and setup of these packages.\n"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20and%20processes%20single-nucleus%20RNA-seq%20data%20from%20PVAT%2C%20performs%20clustering%2C%20and%20visualizes%20differential%20cell%20compositions%20for%20hypothesis%20testing.%0A%0AIntegrate%20spatial%20transcriptomics%20data%20and%20perform%20pseudotime%20trajectory%20analysis%20for%20enhanced%20insights.%0A%0ASingle-nucleus%20analysis%20thoracic%20perivascular%20adipose%20tissue%20high%20fat%20diet%20cell%20composition%20gene%20regulatory%20networks%0A%0AWe%20begin%20by%20downloading%20the%20dataset%20from%20GEO%20and%20loading%20it%20using%20Scanpy.%20This%20dataset%20contains%20nuclei%20from%20PVAT%20samples.%0A%0Aimport%20scanpy%20as%20sc%0Aadata%20%3D%20sc.read_10x_mtx%28%27path_to_PVAT_data%2F%27%2C%20var_names%3D%27gene_symbols%27%2C%20cache%3DTrue%29%0Aadata.var_names_make_unique%28%29%0Asc.pp.filter_cells%28adata%2C%20min_genes%3D200%29%0Asc.pp.filter_genes%28adata%2C%20min_cells%3D3%29%0Asc.pp.normalize_total%28adata%2C%20target_sum%3D1e4%29%0Asc.pp.log1p%28adata%29%0Asc.pp.highly_variable_genes%28adata%2C%20min_mean%3D0.0125%2C%20max_mean%3D3%2C%20min_disp%3D0.5%29%0Aadata%20%3D%20adata%5B%3A%2C%20adata.var%5B%27highly_variable%27%5D%5D%0Asc.pp.scale%28adata%2C%20max_value%3D10%29%0Asc.tl.pca%28adata%2C%20svd_solver%3D%27arpack%27%29%0Asc.pp.neighbors%28adata%2C%20n_neighbors%3D10%2C%20n_pcs%3D40%29%0Asc.tl.umap%28adata%29%0Asc.tl.leiden%28adata%29%0Asc.pl.umap%28adata%2C%20color%3D%5B%27leiden%27%5D%29%0A%0A%0ANext%2C%20differential%20expression%20analysis%20is%20performed%20to%20identify%20cluster-specific%20marker%20genes%20that%20could%20indicate%20key%20transcription%20factors.%0A%0Asc.tl.rank_genes_groups%28adata%2C%20%27leiden%27%2C%20method%3D%27wilcoxon%27%29%0Asc.pl.rank_genes_groups%28adata%2C%20n_genes%3D20%2C%20sharey%3DFalse%29%0A%0A%0AThis%20notebook%20demonstrates%20a%20streamlined%20and%20reproducible%20pipeline%20to%20analyze%20single-nucleus%20RNA-seq%20data%2C%20which%20can%20help%20verify%20the%20paper%27s%20claims.%0A%0A%23%20Further%20code%20could%20integrate%20CellChat%20or%20NicheNet%20analysis%20tools%20to%20evaluate%20ligand-receptor%20interactions%0A%23%20This%20section%20would%20require%20installation%20and%20setup%20of%20these%20packages.%0A%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Single-nucleus%20analysis%20of%20thoracic%20perivascular%20adipose%20tissue%20reveals%20critical%20changes%20in%20cell%20composition%2C%20communication%2C%20and%20gene%20regulatory%20networks%20induced%20by%20a%20high%20fat%20hypertensive%20diet)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads and processes single-nucleus RNA-seq data from PVAT, performs clustering, and visualizes differential cell compositions for hypothesis testing.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-19", "tags": ["single-nucleus analysis adipose tissue", "biology", "bioinformatics"]}}