{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "## Step-by-Step Analysis of DHT Impact on AR Activity\n\nIn this notebook, we will analyze how dihydrotestosterone (DHT) affects androgen receptor (AR) activity using transcriptomic and proteomic data from prostate cancer studies. This analysis will include loading datasets, performing differential expression analysis, pathway enrichment, and visualizing protein interaction networks."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport plotly.express as px\nimport seaborn as sns\nimport matplotlib.pyplot as plt\nfrom scipy.stats import ttest_ind\nfrom statsmodels.stats.multitest import multipletests\nimport networkx as nx\nimport matplotlib.pyplot as plt\nimport plotly.graph_objects as go\n\n# Load datasets\nil1b_data = pd.read_csv('il1b_data.csv')\nbergamottin_data = pd.read_csv('bergamottin_data.csv')\ntranscriptomic_data = pd.read_csv('transcriptomic_dht_data.csv')\nproteomic_data = pd.read_csv('proteomic_dht_data.csv')\n\n# Differential Gene Expression Analysis\n# Assuming 'control' and 'DHT_treated' are columns indicating treatment groups\ngene_counts = transcriptomic_data.groupby('gene')['expression'].agg(['mean', 'std']).reset_index()\ngene_counts['log2_mean_expression'] = np.log2(gene_counts['mean'] + 1)\n\n# T-test for differential expression\ngenes = transcriptomic_data['gene'].unique()\nt_test_results = []\nfor gene in genes:\n    control = transcriptomic_data[(transcriptomic_data['gene'] == gene) & (transcriptomic_data['group'] == 'control')]['expression']\n    dht = transcriptomic_data[(transcriptomic_data['gene'] == gene) & (transcriptomic_data['group'] == 'DHT_treated')]['expression']\n    t_stat, p_val = ttest_ind(control, dht, equal_var=False)\n    t_test_results.append({'gene': gene, 't_stat': t_stat, 'p_value': p_val})\n\n# Multiple testing correction\nttest_df = pd.DataFrame(t_test_results)\nttest_df['adj_p_value'] = multipletests(ttest_df['p_value'], method='fdr_bh')[1]\n\n# Identify significantly differentially expressed genes\nsig_genes = ttest_df[ttest_df['adj_p_value'] < 0.05]\n\n# Pathway Enrichment Analysis\n# Placeholder for pathway analysis using gseapy or similar\n# Example: import gseapy as gp\n# pathways = gp.enrichr(gene_list=sig_genes['gene'].tolist(), gene_sets='KEGG_2016', organism='Human')\n\n# Protein-Protein Interaction Network\n# Using proteomic_data to build interaction network\nG = nx.Graph()\nfor index, row in proteomic_data.iterrows():\n    G.add_edge(row['interactor1'], row['interactor2'])\n\n# Visualize the network with Plotly\npos = nx.spring_layout(G)\nedge_x = []\nedge_y = []\nfor edge in G.edges():\n    x0, y0 = pos[edge[0]]\n    x1, y1 = pos[edge[1]]\n    edge_x += [x0, x1, None]\n    edge_y += [y0, y1, None]\n\nedge_trace = go.Scatter(\n    x=edge_x, y=edge_y,\n    line=dict(width=0.5, color='#888'),\n    hoverinfo='none',\n    mode='lines'\n)\n\nnode_x = []\nnode_y = []\nfor node in G.nodes():\n    x, y = pos[node]\n    node_x.append(x)\n    node_y.append(y)\n\nnode_trace = go.Scatter(\n    x=node_x, y=node_y,\n    mode='markers+text',\n    text=[node for node in G.nodes()],\n    textposition=\"top center\",\n    marker=dict(\n        showscale=True,\n        colorscale='Viridis',\n        reversescale=True,\n        color=[],\n        size=10,\n        colorbar=dict(\n            thickness=15,\n            title='Node Connections',\n            xanchor='left',\n            titleside='right'\n        ),\n        line_width=2\n    )\n)\n\nnode_adjacencies = []\nnode_text = []\nfor node, adjacencies in G.adjacency():\n    node_adjacencies.append(len(adjacencies))\n    node_text.append('# of connections: '+str(len(adjacencies)))\n\nnode_trace.marker.color = node_adjacencies\nnode_trace.text = node_text\n\ndata = [edge_trace, node_trace]\nlayout = go.Layout(\n    title='<br>Protein-Protein Interaction Network of AR Interactors in Response to DHT',\n    titlefont=dict(size=16),\n    showlegend=False,\n    hovermode='closest',\n    margin=dict(b=20,l=5,r=5,t=40),\n    xaxis=dict(showgrid=False, zeroline=False, showticklabels=False),\n    yaxis=dict(showgrid=False, zeroline=False, showticklabels=False)\n)\nfig = go.Figure(data=data, layout=layout)\nfig.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [Evolve This Code](https://biologpt.com/?q=Evolve%20Code%3A%20This%20Python%20notebook%20performs%20a%20comprehensive%20analysis%20of%20the%20impact%20of%20DHT%20on%20AR%20activity%20using%20transcriptomic%20and%20proteomic%20datasets%20from%20prostate%20cancer%20studies.%0A%0AIncorporate%20machine%20learning%20models%20to%20predict%20AR%20activity%20based%20on%20gene%20expression%20profiles%20and%20validate%20findings%20with%20independent%20datasets.%0A%0AImpact%20of%20DHT%20on%20androgen%20receptor%20activity%20in%20prostate%20cancer%20studies%20with%20relevant%20datasets.%0A%0A%23%23%20Step-by-Step%20Analysis%20of%20DHT%20Impact%20on%20AR%20Activity%0A%0AIn%20this%20notebook%2C%20we%20will%20analyze%20how%20dihydrotestosterone%20%28DHT%29%20affects%20androgen%20receptor%20%28AR%29%20activity%20using%20transcriptomic%20and%20proteomic%20data%20from%20prostate%20cancer%20studies.%20This%20analysis%20will%20include%20loading%20datasets%2C%20performing%20differential%20expression%20analysis%2C%20pathway%20enrichment%2C%20and%20visualizing%20protein%20interaction%20networks.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20plotly.express%20as%20px%0Aimport%20seaborn%20as%20sns%0Aimport%20matplotlib.pyplot%20as%20plt%0Afrom%20scipy.stats%20import%20ttest_ind%0Afrom%20statsmodels.stats.multitest%20import%20multipletests%0Aimport%20networkx%20as%20nx%0Aimport%20matplotlib.pyplot%20as%20plt%0Aimport%20plotly.graph_objects%20as%20go%0A%0A%23%20Load%20datasets%0Ail1b_data%20%3D%20pd.read_csv%28%27il1b_data.csv%27%29%0Abergamottin_data%20%3D%20pd.read_csv%28%27bergamottin_data.csv%27%29%0Atranscriptomic_data%20%3D%20pd.read_csv%28%27transcriptomic_dht_data.csv%27%29%0Aproteomic_data%20%3D%20pd.read_csv%28%27proteomic_dht_data.csv%27%29%0A%0A%23%20Differential%20Gene%20Expression%20Analysis%0A%23%20Assuming%20%27control%27%20and%20%27DHT_treated%27%20are%20columns%20indicating%20treatment%20groups%0Agene_counts%20%3D%20transcriptomic_data.groupby%28%27gene%27%29%5B%27expression%27%5D.agg%28%5B%27mean%27%2C%20%27std%27%5D%29.reset_index%28%29%0Agene_counts%5B%27log2_mean_expression%27%5D%20%3D%20np.log2%28gene_counts%5B%27mean%27%5D%20%2B%201%29%0A%0A%23%20T-test%20for%20differential%20expression%0Agenes%20%3D%20transcriptomic_data%5B%27gene%27%5D.unique%28%29%0At_test_results%20%3D%20%5B%5D%0Afor%20gene%20in%20genes%3A%0A%20%20%20%20control%20%3D%20transcriptomic_data%5B%28transcriptomic_data%5B%27gene%27%5D%20%3D%3D%20gene%29%20%26%20%28transcriptomic_data%5B%27group%27%5D%20%3D%3D%20%27control%27%29%5D%5B%27expression%27%5D%0A%20%20%20%20dht%20%3D%20transcriptomic_data%5B%28transcriptomic_data%5B%27gene%27%5D%20%3D%3D%20gene%29%20%26%20%28transcriptomic_data%5B%27group%27%5D%20%3D%3D%20%27DHT_treated%27%29%5D%5B%27expression%27%5D%0A%20%20%20%20t_stat%2C%20p_val%20%3D%20ttest_ind%28control%2C%20dht%2C%20equal_var%3DFalse%29%0A%20%20%20%20t_test_results.append%28%7B%27gene%27%3A%20gene%2C%20%27t_stat%27%3A%20t_stat%2C%20%27p_value%27%3A%20p_val%7D%29%0A%0A%23%20Multiple%20testing%20correction%0Attest_df%20%3D%20pd.DataFrame%28t_test_results%29%0Attest_df%5B%27adj_p_value%27%5D%20%3D%20multipletests%28ttest_df%5B%27p_value%27%5D%2C%20method%3D%27fdr_bh%27%29%5B1%5D%0A%0A%23%20Identify%20significantly%20differentially%20expressed%20genes%0Asig_genes%20%3D%20ttest_df%5Bttest_df%5B%27adj_p_value%27%5D%20%3C%200.05%5D%0A%0A%23%20Pathway%20Enrichment%20Analysis%0A%23%20Placeholder%20for%20pathway%20analysis%20using%20gseapy%20or%20similar%0A%23%20Example%3A%20import%20gseapy%20as%20gp%0A%23%20pathways%20%3D%20gp.enrichr%28gene_list%3Dsig_genes%5B%27gene%27%5D.tolist%28%29%2C%20gene_sets%3D%27KEGG_2016%27%2C%20organism%3D%27Human%27%29%0A%0A%23%20Protein-Protein%20Interaction%20Network%0A%23%20Using%20proteomic_data%20to%20build%20interaction%20network%0AG%20%3D%20nx.Graph%28%29%0Afor%20index%2C%20row%20in%20proteomic_data.iterrows%28%29%3A%0A%20%20%20%20G.add_edge%28row%5B%27interactor1%27%5D%2C%20row%5B%27interactor2%27%5D%29%0A%0A%23%20Visualize%20the%20network%20with%20Plotly%0Apos%20%3D%20nx.spring_layout%28G%29%0Aedge_x%20%3D%20%5B%5D%0Aedge_y%20%3D%20%5B%5D%0Afor%20edge%20in%20G.edges%28%29%3A%0A%20%20%20%20x0%2C%20y0%20%3D%20pos%5Bedge%5B0%5D%5D%0A%20%20%20%20x1%2C%20y1%20%3D%20pos%5Bedge%5B1%5D%5D%0A%20%20%20%20edge_x%20%2B%3D%20%5Bx0%2C%20x1%2C%20None%5D%0A%20%20%20%20edge_y%20%2B%3D%20%5By0%2C%20y1%2C%20None%5D%0A%0Aedge_trace%20%3D%20go.Scatter%28%0A%20%20%20%20x%3Dedge_x%2C%20y%3Dedge_y%2C%0A%20%20%20%20line%3Ddict%28width%3D0.5%2C%20color%3D%27%23888%27%29%2C%0A%20%20%20%20hoverinfo%3D%27none%27%2C%0A%20%20%20%20mode%3D%27lines%27%0A%29%0A%0Anode_x%20%3D%20%5B%5D%0Anode_y%20%3D%20%5B%5D%0Afor%20node%20in%20G.nodes%28%29%3A%0A%20%20%20%20x%2C%20y%20%3D%20pos%5Bnode%5D%0A%20%20%20%20node_x.append%28x%29%0A%20%20%20%20node_y.append%28y%29%0A%0Anode_trace%20%3D%20go.Scatter%28%0A%20%20%20%20x%3Dnode_x%2C%20y%3Dnode_y%2C%0A%20%20%20%20mode%3D%27markers%2Btext%27%2C%0A%20%20%20%20text%3D%5Bnode%20for%20node%20in%20G.nodes%28%29%5D%2C%0A%20%20%20%20textposition%3D%22top%20center%22%2C%0A%20%20%20%20marker%3Ddict%28%0A%20%20%20%20%20%20%20%20showscale%3DTrue%2C%0A%20%20%20%20%20%20%20%20colorscale%3D%27Viridis%27%2C%0A%20%20%20%20%20%20%20%20reversescale%3DTrue%2C%0A%20%20%20%20%20%20%20%20color%3D%5B%5D%2C%0A%20%20%20%20%20%20%20%20size%3D10%2C%0A%20%20%20%20%20%20%20%20colorbar%3Ddict%28%0A%20%20%20%20%20%20%20%20%20%20%20%20thickness%3D15%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20title%3D%27Node%20Connections%27%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20xanchor%3D%27left%27%2C%0A%20%20%20%20%20%20%20%20%20%20%20%20titleside%3D%27right%27%0A%20%20%20%20%20%20%20%20%29%2C%0A%20%20%20%20%20%20%20%20line_width%3D2%0A%20%20%20%20%29%0A%29%0A%0Anode_adjacencies%20%3D%20%5B%5D%0Anode_text%20%3D%20%5B%5D%0Afor%20node%2C%20adjacencies%20in%20G.adjacency%28%29%3A%0A%20%20%20%20node_adjacencies.append%28len%28adjacencies%29%29%0A%20%20%20%20node_text.append%28%27%23%20of%20connections%3A%20%27%2Bstr%28len%28adjacencies%29%29%29%0A%0Anode_trace.marker.color%20%3D%20node_adjacencies%0Anode_trace.text%20%3D%20node_text%0A%0Adata%20%3D%20%5Bedge_trace%2C%20node_trace%5D%0Alayout%20%3D%20go.Layout%28%0A%20%20%20%20title%3D%27%3Cbr%3EProtein-Protein%20Interaction%20Network%20of%20AR%20Interactors%20in%20Response%20to%20DHT%27%2C%0A%20%20%20%20titlefont%3Ddict%28size%3D16%29%2C%0A%20%20%20%20showlegend%3DFalse%2C%0A%20%20%20%20hovermode%3D%27closest%27%2C%0A%20%20%20%20margin%3Ddict%28b%3D20%2Cl%3D5%2Cr%3D5%2Ct%3D40%29%2C%0A%20%20%20%20xaxis%3Ddict%28showgrid%3DFalse%2C%20zeroline%3DFalse%2C%20showticklabels%3DFalse%29%2C%0A%20%20%20%20yaxis%3Ddict%28showgrid%3DFalse%2C%20zeroline%3DFalse%2C%20showticklabels%3DFalse%29%0A%29%0Afig%20%3D%20go.Figure%28data%3Ddata%2C%20layout%3Dlayout%29%0Afig.show%28%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Evolve%20Code%3A%20Analyze%20the%20impact%20of%20DHT%20on%20AR%20activity%20using%20relevant%20datasets%20from%20prostate%20cancer%20studies.%0D%0A%0D%0AInclude%20more%20specific%20datasets%20related%20to%20AR%20activity%20and%20DHT%20treatment%20for%20a%20comprehensive%20analysis.%0D%0A%0D%0AEffects%20of%20DHT%20treatment%20on%20AR%20activity%0D%0A%0D%0A%23%23%23%20Step%201%3A%20Load%20necessary%20libraries%20and%20datasets%0D%0A-%20Import%20libraries%20for%20data%20analysis%20and%20visualization.%0D%0A-%20Load%20datasets%20related%20to%20DHT%20and%20AR%20activity.%0D%0A%0D%0Aimport%20pandas%20as%20pd%0D%0Aimport%20matplotlib.pyplot%20as%20plt%0D%0A%0D%0A%23%20Load%20datasets%0D%0Ail1b_data%20%3D%20pd.read_csv%28%26%23x27%3Bil1b_data.csv%26%23x27%3B%29%0D%0Abergamottin_data%20%3D%20pd.read_csv%28%26%23x27%3Bbergamottin_data.csv%26%23x27%3B%29%0D%0A%0D%0A%23%20Analyze%20AR%20activity%20changes%20with%20DHT%0D%0A%23%20Example%20analysis%20code%20here)\n***"}], "metadata": {"title": "This Python notebook performs a comprehensive analysis of the impact of DHT on AR activity using transcriptomic and proteomic datasets from prostate cancer studies.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-01-14", "tags": ["DHT effects on AR activity", "biology", "bioinformatics"]}}