{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "Below, we load the sample data and utilize Plotly to create an interactive bar chart highlighting the difference in MAG recovery between the two sample types."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import plotly.graph_objs as go\nimport plotly.offline as pyo\n\n# Data extracted from the paper\nsample_types = ['Fecal', 'Cloacal']\nMAGs_recovered = [123, 4]\n\n# Create bar chart\ntrace = go.Bar(\n    x=sample_types,\n    y=MAGs_recovered,\n    marker={'color': ['#6A0C76', '#B04B9A']}\n)\nlayout = go.Layout(\n    title='MAG Recovery by Sample Type',\n    xaxis={'title': 'Sample Type'},\n    yaxis={'title': 'Number of Recovered MAGs'}\n)\nfig = go.Figure(data=[trace], layout=layout)\npyo.plot(fig, filename='mag_recovery.html')"}, {"cell_type": "markdown", "metadata": {}, "source": "The code above creates an interactive visualization that clearly depicts the stark difference in MAG recovery between fecal and cloacal sampling."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20visualizes%20the%20differential%20MAG%20recovery%20across%20fecal%20and%20cloacal%20samples%20using%20real%20study%20data.%0A%0AIncorporate%20statistical%20significance%20indicators%20%28error%20bars%2Fconfidence%20intervals%29%20and%20additional%20replicates%20to%20enhance%20the%20visualization.%0A%0AMetagenome-assembled%20genomes%20lizard%20fecal%20cloacal%20samples%20bacterial%20communities%20functional%20profiles%0A%0ABelow%2C%20we%20load%20the%20sample%20data%20and%20utilize%20Plotly%20to%20create%20an%20interactive%20bar%20chart%20highlighting%20the%20difference%20in%20MAG%20recovery%20between%20the%20two%20sample%20types.%0A%0Aimport%20plotly.graph_objs%20as%20go%0Aimport%20plotly.offline%20as%20pyo%0A%0A%23%20Data%20extracted%20from%20the%20paper%0Asample_types%20%3D%20%5B%27Fecal%27%2C%20%27Cloacal%27%5D%0AMAGs_recovered%20%3D%20%5B123%2C%204%5D%0A%0A%23%20Create%20bar%20chart%0Atrace%20%3D%20go.Bar%28%0A%20%20%20%20x%3Dsample_types%2C%0A%20%20%20%20y%3DMAGs_recovered%2C%0A%20%20%20%20marker%3D%7B%27color%27%3A%20%5B%27%236A0C76%27%2C%20%27%23B04B9A%27%5D%7D%0A%29%0Alayout%20%3D%20go.Layout%28%0A%20%20%20%20title%3D%27MAG%20Recovery%20by%20Sample%20Type%27%2C%0A%20%20%20%20xaxis%3D%7B%27title%27%3A%20%27Sample%20Type%27%7D%2C%0A%20%20%20%20yaxis%3D%7B%27title%27%3A%20%27Number%20of%20Recovered%20MAGs%27%7D%0A%29%0Afig%20%3D%20go.Figure%28data%3D%5Btrace%5D%2C%20layout%3Dlayout%29%0Apyo.plot%28fig%2C%20filename%3D%27mag_recovery.html%27%29%0A%0AThe%20code%20above%20creates%20an%20interactive%20visualization%20that%20clearly%20depicts%20the%20stark%20difference%20in%20MAG%20recovery%20between%20fecal%20and%20cloacal%20sampling.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Contrasting%20recovery%20of%20metagenome%E2%80%91assembled%20genomes%20and%20derived%20bacterial%20communities%20and%20functional%20profiles%20from%20lizard%20fecal%20and%20cloacal%20samples)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code visualizes the differential MAG recovery across fecal and cloacal samples using real study data.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-11", "tags": ["metagenome assembled genomes", "biology", "bioinformatics"]}}