{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "This notebook outlines the process of downloading scRNA-seq datasets from GEO, processing them with Scanpy, and comparing gene expression profiles of NK cell markers with simulation parameters predicted by ABMACT."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nimport pandas as pd\n\n# Download scRNA-seq dataset GEO accession GSE190976\nadata = sc.read_10x_mtx('path_to_GSE190976_directory/', var_names='gene_symbols', cache=True)\n\n# Basic preprocessing\nsc.pp.filter_cells(adata, min_genes=200)\nsc.pp.filter_genes(adata, min_cells=3)\nadata.var['mt'] = adata.var_names.str.startswith('MT-')\nsc.pp.calculate_qc_metrics(adata, qc_vars=['mt'], percent_top=None, log1p=False, inplace=True)\n\n# Normalize and log transform\nsc.pp.normalize_total(adata, target_sum=1e4)\nsc.pp.log1p(adata)\n\n# Identify highly variable genes\nsc.pp.highly_variable_genes(adata, min_mean=0.0125, max_mean=3, min_disp=0.5)\nadata = adata[:, adata.var.highly_variable]\n\n# Scale data and perform PCA\nsc.pp.scale(adata, max_value=10)\nsc.tl.pca(adata, svd_solver='arpack')\n\n# Cluster cells and visualize expression of key NK markers\nsc.pp.neighbors(adata, n_neighbors=10, n_pcs=40)\nsc.tl.umap(adata)\nsc.pl.umap(adata, color=['NCAM1', 'KLRD1'])\n\n# Compare this to simulation parameters from ABMACT (to be integrated via additional analysis)\nprint('NK cell marker expression profile successfully generated.')"}, {"cell_type": "markdown", "metadata": {}, "source": "This code serves as a basis for integrating experimental single-cell data with simulation predictions to validate and refine the ABMACT model outputs."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Additional analysis can include computing differential expression between predicted high-activity and low-activity NK cell clusters\nsc.tl.rank_genes_groups(adata, 'louvain', method='t-test')\nsc.pl.rank_genes_groups(adata, n_genes=20, sharey=False)"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20and%20analyzes%20scRNA-seq%20datasets%20to%20validate%20NK%20cell%20behavior%20predictions%20from%20ABMACT.%0A%0AIntegrate%20simulation%20output%20data%20from%20ABMACT%20directly%20into%20the%20analysis%20pipeline%20to%20allow%20for%20quantitative%20correlation%20analyses%20between%20in%20silico%20predictions%20and%20actual%20gene%20expression%20profiles.%0A%0AAgent-based%20modeling%20cellular%20dynamics%20adoptive%20cell%20therapy%20review%0A%0AThis%20notebook%20outlines%20the%20process%20of%20downloading%20scRNA-seq%20datasets%20from%20GEO%2C%20processing%20them%20with%20Scanpy%2C%20and%20comparing%20gene%20expression%20profiles%20of%20NK%20cell%20markers%20with%20simulation%20parameters%20predicted%20by%20ABMACT.%0A%0Aimport%20scanpy%20as%20sc%0Aimport%20pandas%20as%20pd%0A%0A%23%20Download%20scRNA-seq%20dataset%20GEO%20accession%20GSE190976%0Aadata%20%3D%20sc.read_10x_mtx%28%27path_to_GSE190976_directory%2F%27%2C%20var_names%3D%27gene_symbols%27%2C%20cache%3DTrue%29%0A%0A%23%20Basic%20preprocessing%0Asc.pp.filter_cells%28adata%2C%20min_genes%3D200%29%0Asc.pp.filter_genes%28adata%2C%20min_cells%3D3%29%0Aadata.var%5B%27mt%27%5D%20%3D%20adata.var_names.str.startswith%28%27MT-%27%29%0Asc.pp.calculate_qc_metrics%28adata%2C%20qc_vars%3D%5B%27mt%27%5D%2C%20percent_top%3DNone%2C%20log1p%3DFalse%2C%20inplace%3DTrue%29%0A%0A%23%20Normalize%20and%20log%20transform%0Asc.pp.normalize_total%28adata%2C%20target_sum%3D1e4%29%0Asc.pp.log1p%28adata%29%0A%0A%23%20Identify%20highly%20variable%20genes%0Asc.pp.highly_variable_genes%28adata%2C%20min_mean%3D0.0125%2C%20max_mean%3D3%2C%20min_disp%3D0.5%29%0Aadata%20%3D%20adata%5B%3A%2C%20adata.var.highly_variable%5D%0A%0A%23%20Scale%20data%20and%20perform%20PCA%0Asc.pp.scale%28adata%2C%20max_value%3D10%29%0Asc.tl.pca%28adata%2C%20svd_solver%3D%27arpack%27%29%0A%0A%23%20Cluster%20cells%20and%20visualize%20expression%20of%20key%20NK%20markers%0Asc.pp.neighbors%28adata%2C%20n_neighbors%3D10%2C%20n_pcs%3D40%29%0Asc.tl.umap%28adata%29%0Asc.pl.umap%28adata%2C%20color%3D%5B%27NCAM1%27%2C%20%27KLRD1%27%5D%29%0A%0A%23%20Compare%20this%20to%20simulation%20parameters%20from%20ABMACT%20%28to%20be%20integrated%20via%20additional%20analysis%29%0Aprint%28%27NK%20cell%20marker%20expression%20profile%20successfully%20generated.%27%29%0A%0AThis%20code%20serves%20as%20a%20basis%20for%20integrating%20experimental%20single-cell%20data%20with%20simulation%20predictions%20to%20validate%20and%20refine%20the%20ABMACT%20model%20outputs.%0A%0A%23%20Additional%20analysis%20can%20include%20computing%20differential%20expression%20between%20predicted%20high-activity%20and%20low-activity%20NK%20cell%20clusters%0Asc.tl.rank_genes_groups%28adata%2C%20%27louvain%27%2C%20method%3D%27t-test%27%29%0Asc.pl.rank_genes_groups%28adata%2C%20n_genes%3D20%2C%20sharey%3DFalse%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Agent-based%20modeling%20of%20cellular%20dynamics%20in%20adoptive%20cell%20therapy)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads and analyzes scRNA-seq datasets to validate NK cell behavior predictions from ABMACT.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-03", "tags": ["agent-based modeling cellular therapy", "biology", "bioinformatics"]}}