{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "We first download and parse the mass spectrometry and transcriptomic datasets related to TBR1 interactome. We then construct a network graph using networkx to highlight interactors and annotate those with ASD/ID links."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport networkx as nx\nimport matplotlib.pyplot as plt\n\n# Download datasets (replace URLs with actual provided endpoints if available)\ndf_apms = pd.read_csv('https://biologpt.com/datasets/TBR1_AP-MS.csv')\ndf_transcript = pd.read_csv('https://biologpt.com/datasets/TBR1_transcriptomics.csv')\n\n# Construct network graph\nG = nx.Graph()\n\n# Add nodes and edges from AP-MS dataset\nfor _, row in df_apms.iterrows():\n    G.add_node(row['Interactor'], type=row['Type'], relevance=row['ASD_ID_link'])\n    G.add_edge('TBR1', row['Interactor'], weight=row['InteractionScore'])\n\n# Visualize the network\npos = nx.spring_layout(G)\nnode_colors = ['red' if G.nodes[node]['relevance'] == 'ASD/ID' else 'blue' for node in G.nodes()]\n\nplt.figure(figsize=(10,10))\nnx.draw_networkx_nodes(G, pos, node_color=node_colors, alpha=0.8)\nnx.draw_networkx_edges(G, pos, width=1.0, alpha=0.5)\nnx.draw_networkx_labels(G, pos, font_size=10)\nplt.title('TBR1 Interactome Network')\nplt.axis('off')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "The above code provides a visualization of the TBR1 interactome, highlighting key ASD/ID-related interactors. By integrating AP-MS and transcriptomic data, we aim to uncover potential regulatory disruptions."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Further analysis: Identify interactors with significant transcriptomic deregulation\nasg_interactors = df_transcript[df_transcript['p_value'] < 0.05]['Gene'].tolist()\ndisrupted_nodes = [node for node in G.nodes() if node in asg_interactors]\n\nprint('Disrupted key interactors:', disrupted_nodes)"}, {"cell_type": "markdown", "metadata": {}, "source": "This additional analysis checks for overlap between interactors with altered expression and those identified in the AP-MS experiment, offering insights into which interactions are most disrupted in NDD models."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import seaborn as sns\n\n# Plot interaction score distribution for all interactors\nplt.figure(figsize=(8,6))\nsns.histplot(df_apms['InteractionScore'], kde=True, color='green')\nplt.title('Distribution of TBR1 Interaction Scores')\nplt.xlabel('Interaction Score')\nplt.ylabel('Frequency')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20integrates%20AP-MS%20data%20with%20transcriptome%20profiles%20to%20build%20and%20visualize%20the%20TBR1%20interactome%20network%2C%20aiding%20in%20identifying%20disrupted%20interactions%20in%20NDD%20models.%0A%0AIntegrate%20real-time%20data%20fetching%20from%20protein%20databases%20and%20include%20statistical%20tests%20to%20evaluate%20interaction%20disruptions%20more%20robustly.%0A%0ATBR1%20interactome%20neurodevelopmental%20disorders%20protein%20interactions%0A%0AWe%20first%20download%20and%20parse%20the%20mass%20spectrometry%20and%20transcriptomic%20datasets%20related%20to%20TBR1%20interactome.%20We%20then%20construct%20a%20network%20graph%20using%20networkx%20to%20highlight%20interactors%20and%20annotate%20those%20with%20ASD%2FID%20links.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20networkx%20as%20nx%0Aimport%20matplotlib.pyplot%20as%20plt%0A%0A%23%20Download%20datasets%20%28replace%20URLs%20with%20actual%20provided%20endpoints%20if%20available%29%0Adf_apms%20%3D%20pd.read_csv%28%27https%3A%2F%2Fbiologpt.com%2Fdatasets%2FTBR1_AP-MS.csv%27%29%0Adf_transcript%20%3D%20pd.read_csv%28%27https%3A%2F%2Fbiologpt.com%2Fdatasets%2FTBR1_transcriptomics.csv%27%29%0A%0A%23%20Construct%20network%20graph%0AG%20%3D%20nx.Graph%28%29%0A%0A%23%20Add%20nodes%20and%20edges%20from%20AP-MS%20dataset%0Afor%20_%2C%20row%20in%20df_apms.iterrows%28%29%3A%0A%20%20%20%20G.add_node%28row%5B%27Interactor%27%5D%2C%20type%3Drow%5B%27Type%27%5D%2C%20relevance%3Drow%5B%27ASD_ID_link%27%5D%29%0A%20%20%20%20G.add_edge%28%27TBR1%27%2C%20row%5B%27Interactor%27%5D%2C%20weight%3Drow%5B%27InteractionScore%27%5D%29%0A%0A%23%20Visualize%20the%20network%0Apos%20%3D%20nx.spring_layout%28G%29%0Anode_colors%20%3D%20%5B%27red%27%20if%20G.nodes%5Bnode%5D%5B%27relevance%27%5D%20%3D%3D%20%27ASD%2FID%27%20else%20%27blue%27%20for%20node%20in%20G.nodes%28%29%5D%0A%0Aplt.figure%28figsize%3D%2810%2C10%29%29%0Anx.draw_networkx_nodes%28G%2C%20pos%2C%20node_color%3Dnode_colors%2C%20alpha%3D0.8%29%0Anx.draw_networkx_edges%28G%2C%20pos%2C%20width%3D1.0%2C%20alpha%3D0.5%29%0Anx.draw_networkx_labels%28G%2C%20pos%2C%20font_size%3D10%29%0Aplt.title%28%27TBR1%20Interactome%20Network%27%29%0Aplt.axis%28%27off%27%29%0Aplt.show%28%29%0A%0AThe%20above%20code%20provides%20a%20visualization%20of%20the%20TBR1%20interactome%2C%20highlighting%20key%20ASD%2FID-related%20interactors.%20By%20integrating%20AP-MS%20and%20transcriptomic%20data%2C%20we%20aim%20to%20uncover%20potential%20regulatory%20disruptions.%0A%0A%23%20Further%20analysis%3A%20Identify%20interactors%20with%20significant%20transcriptomic%20deregulation%0Aasg_interactors%20%3D%20df_transcript%5Bdf_transcript%5B%27p_value%27%5D%20%3C%200.05%5D%5B%27Gene%27%5D.tolist%28%29%0Adisrupted_nodes%20%3D%20%5Bnode%20for%20node%20in%20G.nodes%28%29%20if%20node%20in%20asg_interactors%5D%0A%0Aprint%28%27Disrupted%20key%20interactors%3A%27%2C%20disrupted_nodes%29%0A%0AThis%20additional%20analysis%20checks%20for%20overlap%20between%20interactors%20with%20altered%20expression%20and%20those%20identified%20in%20the%20AP-MS%20experiment%2C%20offering%20insights%20into%20which%20interactions%20are%20most%20disrupted%20in%20NDD%20models.%0A%0Aimport%20seaborn%20as%20sns%0A%0A%23%20Plot%20interaction%20score%20distribution%20for%20all%20interactors%0Aplt.figure%28figsize%3D%288%2C6%29%29%0Asns.histplot%28df_apms%5B%27InteractionScore%27%5D%2C%20kde%3DTrue%2C%20color%3D%27green%27%29%0Aplt.title%28%27Distribution%20of%20TBR1%20Interaction%20Scores%27%29%0Aplt.xlabel%28%27Interaction%20Score%27%29%0Aplt.ylabel%28%27Frequency%27%29%0Aplt.show%28%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Characterization%20of%20the%20TBR1%20interactome%3A%20variants%20associated%20with%20neurodevelopmental%20disorders%20disrupt%20novel%20protein%20interactions)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code integrates AP-MS data with transcriptome profiles to build and visualize the TBR1 interactome network, aiding in identifying disrupted interactions in NDD models.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-08", "tags": ["TBR1 interactome variants", "biology", "bioinformatics"]}}