{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "Below we describe a step-by-step Jupyter notebook for downloading IGAC scRNA-seq datasets and performing clustering, differential expression analysis, and cell-cell communication inference."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nadata = sc.read_10x_mtx('path_to_data_directory/')\nsc.pp.filter_cells(adata, min_genes=200)\nsc.pp.filter_genes(adata, min_cells=3)\nsc.pp.normalize_total(adata, target_sum=1e4)\nsc.pp.log1p(adata)\nsc.tl.pca(adata, svd_solver='arpack')\nsc.pp.neighbors(adata, n_neighbors=10, n_pcs=40)\nsc.tl.umap(adata)\nsc.tl.leiden(adata, resolution=0.5)\nsc.pl.umap(adata, color=['leiden'], save='_clusters.png')"}, {"cell_type": "markdown", "metadata": {}, "source": "The above code standardizes scRNA-seq data and performs clustering using Leiden algorithm to identify cell subpopulations. Further analysis can annotate T cell populations and infer cell-cell interactions."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import seaborn as sns\nimport matplotlib.pyplot as plt\n# Example: visualize expression of HAVCR2 and LAYN\nsc.pl.umap(adata, color=['HAVCR2', 'LAYN'], save='_marker_expression.png')"}, {"cell_type": "markdown", "metadata": {}, "source": "This step visualizes the expression profiles of important markers, aiding in the identification of exhausted T cells and Tregs."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Further code can be developed to analyze cell-cell communication using tools such as CellPhoneDB or pytc, integrating dataset-specific parameters."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20and%20analyzes%20scRNA-seq%20datasets%20from%20IGAC%20samples%20to%20identify%20immunosuppressive%20cell%20subpopulations%20and%20their%20interaction%20networks.%0A%0AInclude%20direct%20links%20to%20published%20IGAC%20datasets%20and%20integrate%20advanced%20cell-cell%20communication%20inference%20modules%20for%20deeper%20insights.%0A%0AImmunosuppressive%20microenvironment%20Lauren%20intestinal-type%20gastric%20adenocarcinoma%20profiles%0A%0ABelow%20we%20describe%20a%20step-by-step%20Jupyter%20notebook%20for%20downloading%20IGAC%20scRNA-seq%20datasets%20and%20performing%20clustering%2C%20differential%20expression%20analysis%2C%20and%20cell-cell%20communication%20inference.%0A%0Aimport%20scanpy%20as%20sc%0Aadata%20%3D%20sc.read_10x_mtx%28%27path_to_data_directory%2F%27%29%0Asc.pp.filter_cells%28adata%2C%20min_genes%3D200%29%0Asc.pp.filter_genes%28adata%2C%20min_cells%3D3%29%0Asc.pp.normalize_total%28adata%2C%20target_sum%3D1e4%29%0Asc.pp.log1p%28adata%29%0Asc.tl.pca%28adata%2C%20svd_solver%3D%27arpack%27%29%0Asc.pp.neighbors%28adata%2C%20n_neighbors%3D10%2C%20n_pcs%3D40%29%0Asc.tl.umap%28adata%29%0Asc.tl.leiden%28adata%2C%20resolution%3D0.5%29%0Asc.pl.umap%28adata%2C%20color%3D%5B%27leiden%27%5D%2C%20save%3D%27_clusters.png%27%29%0A%0AThe%20above%20code%20standardizes%20scRNA-seq%20data%20and%20performs%20clustering%20using%20Leiden%20algorithm%20to%20identify%20cell%20subpopulations.%20Further%20analysis%20can%20annotate%20T%20cell%20populations%20and%20infer%20cell-cell%20interactions.%0A%0Aimport%20seaborn%20as%20sns%0Aimport%20matplotlib.pyplot%20as%20plt%0A%23%20Example%3A%20visualize%20expression%20of%20HAVCR2%20and%20LAYN%0Asc.pl.umap%28adata%2C%20color%3D%5B%27HAVCR2%27%2C%20%27LAYN%27%5D%2C%20save%3D%27_marker_expression.png%27%29%0A%0AThis%20step%20visualizes%20the%20expression%20profiles%20of%20important%20markers%2C%20aiding%20in%20the%20identification%20of%20exhausted%20T%20cells%20and%20Tregs.%0A%0A%23%20Further%20code%20can%20be%20developed%20to%20analyze%20cell-cell%20communication%20using%20tools%20such%20as%20CellPhoneDB%20or%20pytc%2C%20integrating%20dataset-specific%20parameters.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20The%20profiles%20of%20immunosuppressive%20microenvironment%20in%20the%20Lauren%20intestinal-type%20gastric%20adenocarcinoma.)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads and analyzes scRNA-seq datasets from IGAC samples to identify immunosuppressive cell subpopulations and their interaction networks.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-11", "tags": ["Lauren gastric adenocarcinoma profiles", "biology", "bioinformatics"]}}