{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "The following section outlines how to download metagenomic datasets from the European Nucleotide Archive and process them with Kraken 2 and AMR++ pipeline outputs."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\n# Download dataset metadata using ENA API (simulated placeholder for accession PRJEB81356)\nurl = 'https://www.ebi.ac.uk/ena/browser/api/xml/PRJEB81356'\ndata = pd.read_xml(url)\n\n# Process sample information\nprint(data[['sample_accession', 'collection_date', 'geographic_location']].head())"}, {"cell_type": "markdown", "metadata": {}, "source": "Next, we visualize the normalized AMR gene abundance using Plotly."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import plotly.express as px\n# Example data, replace with actual processed AMR gene abundance values\ndf = pd.DataFrame({\n    'Sample': ['Sample1', 'Sample2', 'Sample3', 'Sample4'],\n    'AMR_abundance': [0.030, 0.035, 0.032, 0.038]\n})\nfig = px.bar(df, x='Sample', y='AMR_abundance', title='Normalized AMR Gene Abundance')\nfig.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "This analysis supports further investigations into inter-sample variability of AMR genes using custom scripts and deposited datasets."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20analyzes%20metagenomic%20datasets%20to%20extract%20microbial%20taxonomic%20profiles%20and%20AMR%20gene%20abundance%20using%20real%20dataset%20identifiers%20from%20the%20study.%0A%0AIntegrate%20real-time%20fetching%20of%20dataset%20updates%20and%20advanced%20statistical%20testing%20using%20scipy%20and%20statsmodels.%0A%0AMetagenomic%20analysis%20faecal%20microbiota%20AMR%20roe%20deer%20Western%20Pomerania%0A%0AThe%20following%20section%20outlines%20how%20to%20download%20metagenomic%20datasets%20from%20the%20European%20Nucleotide%20Archive%20and%20process%20them%20with%20Kraken%202%20and%20AMR%2B%2B%20pipeline%20outputs.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0A%23%20Download%20dataset%20metadata%20using%20ENA%20API%20%28simulated%20placeholder%20for%20accession%20PRJEB81356%29%0Aurl%20%3D%20%27https%3A%2F%2Fwww.ebi.ac.uk%2Fena%2Fbrowser%2Fapi%2Fxml%2FPRJEB81356%27%0Adata%20%3D%20pd.read_xml%28url%29%0A%0A%23%20Process%20sample%20information%0Aprint%28data%5B%5B%27sample_accession%27%2C%20%27collection_date%27%2C%20%27geographic_location%27%5D%5D.head%28%29%29%0A%0ANext%2C%20we%20visualize%20the%20normalized%20AMR%20gene%20abundance%20using%20Plotly.%0A%0Aimport%20plotly.express%20as%20px%0A%23%20Example%20data%2C%20replace%20with%20actual%20processed%20AMR%20gene%20abundance%20values%0Adf%20%3D%20pd.DataFrame%28%7B%0A%20%20%20%20%27Sample%27%3A%20%5B%27Sample1%27%2C%20%27Sample2%27%2C%20%27Sample3%27%2C%20%27Sample4%27%5D%2C%0A%20%20%20%20%27AMR_abundance%27%3A%20%5B0.030%2C%200.035%2C%200.032%2C%200.038%5D%0A%7D%29%0Afig%20%3D%20px.bar%28df%2C%20x%3D%27Sample%27%2C%20y%3D%27AMR_abundance%27%2C%20title%3D%27Normalized%20AMR%20Gene%20Abundance%27%29%0Afig.show%28%29%0A%0AThis%20analysis%20supports%20further%20investigations%20into%20inter-sample%20variability%20of%20AMR%20genes%20using%20custom%20scripts%20and%20deposited%20datasets.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Metagenomic%20analysis%20of%20the%20faecal%20microbiota%20and%20AMR%20in%20roe%20deer%20in%20Western%20Pomerania)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code analyzes metagenomic datasets to extract microbial taxonomic profiles and AMR gene abundance using real dataset identifiers from the study.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-27", "tags": ["faecal microbiota roe deer", "biology", "bioinformatics"]}}