{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "This notebook demonstrates the analysis workflow for assessing the causal impact of \u03b1-Klotho on VTE using real GWAS summary data. It integrates data merging, basic MR ratio estimation, and visualizations."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport matplotlib.pyplot as plt\n\n# Download GWAS summary data (replace 'alpha_klotho_gwas.csv' and 'vte_gwas.csv' with actual file paths)\nalpha_klotho = pd.read_csv('alpha_klotho_gwas.csv')\nvte_data = pd.read_csv('vte_gwas.csv')\n\n# Merge datasets on SNP identifier\nmerged_data = pd.merge(alpha_klotho, vte_data, on='SNP')\n\n# Compute MR estimates using a simple ratio method\nmerged_data['MR_estimate'] = merged_data['beta_vte'] / merged_data['beta_klotho']\n\n# Plot the histogram of MR estimates\nplt.figure(figsize=(8, 5))\nplt.hist(merged_data['MR_estimate'], bins=20, color='skyblue', edgecolor='black')\nplt.title('Distribution of MR Estimates for \u03b1-Klotho on VTE')\nplt.xlabel('MR Estimate (Odds Ratio)')\nplt.ylabel('Frequency')\nplt.show()\n\nmerged_data.head()"}, {"cell_type": "markdown", "metadata": {}, "source": "Additionally, a forest plot can be generated to compare the odds ratios and confidence intervals across different VTE outcomes."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import numpy as np\n\ndef forest_plot(data):\n    fig, ax = plt.subplots(figsize=(8, len(data)*0.5))\n    y_positions = np.arange(len(data))\n    ax.errorbar(data['OR'], y_positions, xerr=[data['OR']-data['LowerCI'], data['UpperCI']-data['OR']], fmt='o', color='darkblue')\n    ax.set_yticks(y_positions)\n    ax.set_yticklabels(data['Outcome'])\n    ax.invert_yaxis()\n    ax.set_xlabel('Odds Ratio')\n    ax.set_title('Forest Plot of MR Analysis Results')\n    plt.show()\n\n# Example dataset\nexample_data = pd.DataFrame({\n    'Outcome': ['DVT', 'PE', 'DVT+PE'],\n    'OR': [0.992, 0.474, 0.984],\n    'LowerCI': [0.986, 0.255, 0.971],\n    'UpperCI': [0.998, 0.881, 0.997]\n})\n\nforest_plot(example_data)"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20Python%20code%20downloads%20and%20merges%20GWAS%20summary%20data%20on%20%CE%B1-Klotho%20and%20VTE%2C%20performs%20a%20basic%20MR%20analysis%2C%20and%20visualizes%20the%20distribution%20of%20odds%20ratios%20to%20replicate%20study%20findings.%0A%0AIntegrate%20established%20MR%20Python%20packages%20and%20include%20comprehensive%20sensitivity%20analysis%20for%20enhanced%20robustness.%0A%0ACausal%20association%20%CE%B1-Klotho%20venous%20thromboembolism%20Mendelian%20randomization%0A%0AThis%20notebook%20demonstrates%20the%20analysis%20workflow%20for%20assessing%20the%20causal%20impact%20of%20%CE%B1-Klotho%20on%20VTE%20using%20real%20GWAS%20summary%20data.%20It%20integrates%20data%20merging%2C%20basic%20MR%20ratio%20estimation%2C%20and%20visualizations.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20matplotlib.pyplot%20as%20plt%0A%0A%23%20Download%20GWAS%20summary%20data%20%28replace%20%27alpha_klotho_gwas.csv%27%20and%20%27vte_gwas.csv%27%20with%20actual%20file%20paths%29%0Aalpha_klotho%20%3D%20pd.read_csv%28%27alpha_klotho_gwas.csv%27%29%0Avte_data%20%3D%20pd.read_csv%28%27vte_gwas.csv%27%29%0A%0A%23%20Merge%20datasets%20on%20SNP%20identifier%0Amerged_data%20%3D%20pd.merge%28alpha_klotho%2C%20vte_data%2C%20on%3D%27SNP%27%29%0A%0A%23%20Compute%20MR%20estimates%20using%20a%20simple%20ratio%20method%0Amerged_data%5B%27MR_estimate%27%5D%20%3D%20merged_data%5B%27beta_vte%27%5D%20%2F%20merged_data%5B%27beta_klotho%27%5D%0A%0A%23%20Plot%20the%20histogram%20of%20MR%20estimates%0Aplt.figure%28figsize%3D%288%2C%205%29%29%0Aplt.hist%28merged_data%5B%27MR_estimate%27%5D%2C%20bins%3D20%2C%20color%3D%27skyblue%27%2C%20edgecolor%3D%27black%27%29%0Aplt.title%28%27Distribution%20of%20MR%20Estimates%20for%20%CE%B1-Klotho%20on%20VTE%27%29%0Aplt.xlabel%28%27MR%20Estimate%20%28Odds%20Ratio%29%27%29%0Aplt.ylabel%28%27Frequency%27%29%0Aplt.show%28%29%0A%0Amerged_data.head%28%29%0A%0AAdditionally%2C%20a%20forest%20plot%20can%20be%20generated%20to%20compare%20the%20odds%20ratios%20and%20confidence%20intervals%20across%20different%20VTE%20outcomes.%0A%0Aimport%20numpy%20as%20np%0A%0Adef%20forest_plot%28data%29%3A%0A%20%20%20%20fig%2C%20ax%20%3D%20plt.subplots%28figsize%3D%288%2C%20len%28data%29%2A0.5%29%29%0A%20%20%20%20y_positions%20%3D%20np.arange%28len%28data%29%29%0A%20%20%20%20ax.errorbar%28data%5B%27OR%27%5D%2C%20y_positions%2C%20xerr%3D%5Bdata%5B%27OR%27%5D-data%5B%27LowerCI%27%5D%2C%20data%5B%27UpperCI%27%5D-data%5B%27OR%27%5D%5D%2C%20fmt%3D%27o%27%2C%20color%3D%27darkblue%27%29%0A%20%20%20%20ax.set_yticks%28y_positions%29%0A%20%20%20%20ax.set_yticklabels%28data%5B%27Outcome%27%5D%29%0A%20%20%20%20ax.invert_yaxis%28%29%0A%20%20%20%20ax.set_xlabel%28%27Odds%20Ratio%27%29%0A%20%20%20%20ax.set_title%28%27Forest%20Plot%20of%20MR%20Analysis%20Results%27%29%0A%20%20%20%20plt.show%28%29%0A%0A%23%20Example%20dataset%0Aexample_data%20%3D%20pd.DataFrame%28%7B%0A%20%20%20%20%27Outcome%27%3A%20%5B%27DVT%27%2C%20%27PE%27%2C%20%27DVT%2BPE%27%5D%2C%0A%20%20%20%20%27OR%27%3A%20%5B0.992%2C%200.474%2C%200.984%5D%2C%0A%20%20%20%20%27LowerCI%27%3A%20%5B0.986%2C%200.255%2C%200.971%5D%2C%0A%20%20%20%20%27UpperCI%27%3A%20%5B0.998%2C%200.881%2C%200.997%5D%0A%7D%29%0A%0Aforest_plot%28example_data%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Causal%20association%20between%20circulating%20%CE%B1-Klotho%20levels%20and%20venous%20thromboembolism%3A%20a%20two-sample%20Mendelian%20randomization%20study.)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This Python code downloads and merges GWAS summary data on \u03b1-Klotho and VTE, performs a basic MR analysis, and visualizes the distribution of odds ratios to replicate study findings.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-05", "tags": ["\u03b1-Klotho levels venous thromboembolism", "biology", "bioinformatics"]}}