{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "Below is a step-by-step Jupyter Notebook snippet that loads CFPS screening data, preprocesses it, trains a regression model using scikit-learn, and identifies top variants based on kinetic activity scores."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\nfrom sklearn.model_selection import train_test_split\nfrom sklearn.ensemble import RandomForestRegressor\nimport matplotlib.pyplot as plt\n\n# Load CFPS screening data (assuming data file 'cfps_data.csv' available from supplementary information)\ndata = pd.read_csv('cfps_data.csv')\n\n# Preprocess data: features as one-hot encoding of amino acid mutations, and target as kinetic activity score\nfeatures = data.drop(columns=['variant_id', 'activity_score'])\ntarget = data['activity_score']\n\n# Split the data\nX_train, X_test, y_train, y_test = train_test_split(features, target, test_size=0.2, random_state=42)\n\n# Train a Random Forest Regressor\nmodel = RandomForestRegressor(n_estimators=100, random_state=42)\nmodel.fit(X_train, y_train)\n\n# Evaluate the model\nscore = model.score(X_test, y_test)\nprint('Model R^2 Score:', score)\n\n# Predict and highlight top 10 variants\ndata['predicted_score'] = model.predict(features)\ntop_variants = data.sort_values(by='predicted_score', ascending=False).head(10)\nprint(top_variants[['variant_id', 'predicted_score']])\n\n# Plot the distribution of predicted scores\nplt.figure(figsize=(8,4))\nplt.hist(data['predicted_score'], bins=20, color='skyblue', edgecolor='black')\nplt.title('Distribution of Predicted Protease Activity Scores')\nplt.xlabel('Predicted Activity Score')\nplt.ylabel('Frequency')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "This notebook helps in visualizing the scoring distribution and identifying the leading candidates for further experimental validation."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Additional analysis: Feature importance plot\nimportances = model.feature_importances_\nfeature_names = features.columns\nindices = np.argsort(importances)[::-1]\n\nplt.figure(figsize=(10,6))\nplt.title('Feature Importances from CFPS Screening Model')\nplt.bar(range(len(importances)), importances[indices], color='lightgreen', align='center')\nplt.xticks(range(len(importances)), feature_names[indices], rotation=90)\nplt.tight_layout()\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20experimental%20CFPS%20screening%20data%20and%20applies%20ML%20analysis%20to%20identify%20top%20protease%20variants%2C%20aiding%20in%20protein%20engineering%20decision-making.%0A%0AInclude%20integration%20with%20live%20CFPS%20data%20pipelines%20and%20validation%20against%20in%20vivo%20results%20for%20enhanced%20predictive%20power.%0A%0ACell-free%20protein%20synthesis%20machine%20learning%20protease%20variants%20review%0A%0ABelow%20is%20a%20step-by-step%20Jupyter%20Notebook%20snippet%20that%20loads%20CFPS%20screening%20data%2C%20preprocesses%20it%2C%20trains%20a%20regression%20model%20using%20scikit-learn%2C%20and%20identifies%20top%20variants%20based%20on%20kinetic%20activity%20scores.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0Afrom%20sklearn.model_selection%20import%20train_test_split%0Afrom%20sklearn.ensemble%20import%20RandomForestRegressor%0Aimport%20matplotlib.pyplot%20as%20plt%0A%0A%23%20Load%20CFPS%20screening%20data%20%28assuming%20data%20file%20%27cfps_data.csv%27%20available%20from%20supplementary%20information%29%0Adata%20%3D%20pd.read_csv%28%27cfps_data.csv%27%29%0A%0A%23%20Preprocess%20data%3A%20features%20as%20one-hot%20encoding%20of%20amino%20acid%20mutations%2C%20and%20target%20as%20kinetic%20activity%20score%0Afeatures%20%3D%20data.drop%28columns%3D%5B%27variant_id%27%2C%20%27activity_score%27%5D%29%0Atarget%20%3D%20data%5B%27activity_score%27%5D%0A%0A%23%20Split%20the%20data%0AX_train%2C%20X_test%2C%20y_train%2C%20y_test%20%3D%20train_test_split%28features%2C%20target%2C%20test_size%3D0.2%2C%20random_state%3D42%29%0A%0A%23%20Train%20a%20Random%20Forest%20Regressor%0Amodel%20%3D%20RandomForestRegressor%28n_estimators%3D100%2C%20random_state%3D42%29%0Amodel.fit%28X_train%2C%20y_train%29%0A%0A%23%20Evaluate%20the%20model%0Ascore%20%3D%20model.score%28X_test%2C%20y_test%29%0Aprint%28%27Model%20R%5E2%20Score%3A%27%2C%20score%29%0A%0A%23%20Predict%20and%20highlight%20top%2010%20variants%0Adata%5B%27predicted_score%27%5D%20%3D%20model.predict%28features%29%0Atop_variants%20%3D%20data.sort_values%28by%3D%27predicted_score%27%2C%20ascending%3DFalse%29.head%2810%29%0Aprint%28top_variants%5B%5B%27variant_id%27%2C%20%27predicted_score%27%5D%5D%29%0A%0A%23%20Plot%20the%20distribution%20of%20predicted%20scores%0Aplt.figure%28figsize%3D%288%2C4%29%29%0Aplt.hist%28data%5B%27predicted_score%27%5D%2C%20bins%3D20%2C%20color%3D%27skyblue%27%2C%20edgecolor%3D%27black%27%29%0Aplt.title%28%27Distribution%20of%20Predicted%20Protease%20Activity%20Scores%27%29%0Aplt.xlabel%28%27Predicted%20Activity%20Score%27%29%0Aplt.ylabel%28%27Frequency%27%29%0Aplt.show%28%29%0A%0AThis%20notebook%20helps%20in%20visualizing%20the%20scoring%20distribution%20and%20identifying%20the%20leading%20candidates%20for%20further%20experimental%20validation.%0A%0A%23%20Additional%20analysis%3A%20Feature%20importance%20plot%0Aimportances%20%3D%20model.feature_importances_%0Afeature_names%20%3D%20features.columns%0Aindices%20%3D%20np.argsort%28importances%29%5B%3A%3A-1%5D%0A%0Aplt.figure%28figsize%3D%2810%2C6%29%29%0Aplt.title%28%27Feature%20Importances%20from%20CFPS%20Screening%20Model%27%29%0Aplt.bar%28range%28len%28importances%29%29%2C%20importances%5Bindices%5D%2C%20color%3D%27lightgreen%27%2C%20align%3D%27center%27%29%0Aplt.xticks%28range%28len%28importances%29%29%2C%20feature_names%5Bindices%5D%2C%20rotation%3D90%29%0Aplt.tight_layout%28%29%0Aplt.show%28%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Cell-free%20protein%20synthesis%20as%20a%20method%20to%20rapidly%20screen%20machine%20learning-directed%20protease%20variants)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads experimental CFPS screening data and applies ML analysis to identify top protease variants, aiding in protein engineering decision-making.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-13", "tags": ["cell-free protein synthesis", "biology", "bioinformatics"]}}