{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "### Step 1: Import Libraries\nImport necessary libraries for data analysis and visualization."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\nimport matplotlib.pyplot as plt\nimport seaborn as sns\nfrom statsmodels.stats.multitest import multipletests"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 2: Load GWAS Data\nLoad the GWAS dataset containing SNPs and their associated p-values."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "gwas_data = pd.read_csv('gwas_results.csv')\n\n# Display the first few rows of the dataset\nprint(gwas_data.head())"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 3: Adjust p-values for Multiple Testing\nUse the Benjamini-Hochberg method to control the false discovery rate."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "gwas_data['adjusted_p'] = multipletests(gwas_data['p_value'], method='fdr_bh')[1]"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 4: Visualize Significant SNPs\nCreate a Manhattan plot to visualize SNP significance across the genome."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "plt.figure(figsize=(10, 6))\nplt.scatter(gwas_data['position'], -np.log10(gwas_data['adjusted_p']), color='blue')\nplt.axhline(y=-np.log10(0.05), color='red', linestyle='--')\nplt.title('Manhattan Plot of GWAS Results')\nplt.xlabel('Genomic Position')\nplt.ylabel('-log10(Adjusted p-value)')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 5: Identify Significant SNPs\nFilter the dataset for SNPs with adjusted p-values below the threshold."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "significant_snps = gwas_data[gwas_data['adjusted_p'] < 0.05]\nprint(significant_snps)"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20analyzes%20GWAS%20data%20to%20identify%20significant%20SNPs%20associated%20with%20specific%20traits%20using%20Python%20libraries%20for%20statistical%20analysis%20and%20visualization.%0A%0AIncorporate%20additional%20datasets%20for%20validation%20and%20explore%20the%20integration%20of%20epigenetic%20data%20to%20enhance%20the%20analysis.%0A%0AGenome-wide%20association%20studies%20overview%0A%0A%23%23%23%20Step%201%3A%20Import%20Libraries%0AImport%20necessary%20libraries%20for%20data%20analysis%20and%20visualization.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0Aimport%20matplotlib.pyplot%20as%20plt%0Aimport%20seaborn%20as%20sns%0Afrom%20statsmodels.stats.multitest%20import%20multipletests%0A%0A%23%23%23%20Step%202%3A%20Load%20GWAS%20Data%0ALoad%20the%20GWAS%20dataset%20containing%20SNPs%20and%20their%20associated%20p-values.%0A%0Agwas_data%20%3D%20pd.read_csv%28%27gwas_results.csv%27%29%0A%0A%23%20Display%20the%20first%20few%20rows%20of%20the%20dataset%0Aprint%28gwas_data.head%28%29%29%0A%0A%23%23%23%20Step%203%3A%20Adjust%20p-values%20for%20Multiple%20Testing%0AUse%20the%20Benjamini-Hochberg%20method%20to%20control%20the%20false%20discovery%20rate.%0A%0Agwas_data%5B%27adjusted_p%27%5D%20%3D%20multipletests%28gwas_data%5B%27p_value%27%5D%2C%20method%3D%27fdr_bh%27%29%5B1%5D%0A%0A%23%23%23%20Step%204%3A%20Visualize%20Significant%20SNPs%0ACreate%20a%20Manhattan%20plot%20to%20visualize%20SNP%20significance%20across%20the%20genome.%0A%0Aplt.figure%28figsize%3D%2810%2C%206%29%29%0Aplt.scatter%28gwas_data%5B%27position%27%5D%2C%20-np.log10%28gwas_data%5B%27adjusted_p%27%5D%29%2C%20color%3D%27blue%27%29%0Aplt.axhline%28y%3D-np.log10%280.05%29%2C%20color%3D%27red%27%2C%20linestyle%3D%27--%27%29%0Aplt.title%28%27Manhattan%20Plot%20of%20GWAS%20Results%27%29%0Aplt.xlabel%28%27Genomic%20Position%27%29%0Aplt.ylabel%28%27-log10%28Adjusted%20p-value%29%27%29%0Aplt.show%28%29%0A%0A%23%23%23%20Step%205%3A%20Identify%20Significant%20SNPs%0AFilter%20the%20dataset%20for%20SNPs%20with%20adjusted%20p-values%20below%20the%20threshold.%0A%0Asignificant_snps%20%3D%20gwas_data%5Bgwas_data%5B%27adjusted_p%27%5D%20%3C%200.05%5D%0Aprint%28significant_snps%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=GWAS)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code analyzes GWAS data to identify significant SNPs associated with specific traits using Python libraries for statistical analysis and visualization.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-18", "tags": ["GWAS methodology and applications", "biology", "bioinformatics"]}}