{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "This notebook details the data ingestion, preprocessing of RNA-seq files with Fastp, alignment using HISAT2, and BAM to BED conversion for feature extraction."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import subprocess\nimport os\n\n# Download dataset metadata\nsubprocess.run(['wget', 'https://github.com/BioinformaticsLabAtMUN/OpDetect/raw/main/dataset_info.csv'])\n\n# Preprocess RNA-seq data using Fastp\nos.system('fastp -i sample_R1.fastq -I sample_R2.fastq -o trimmed_R1.fastq -O trimmed_R2.fastq')\n\n# Align reads using HISAT2\nos.system('hisat2 -x genome_index -1 trimmed_R1.fastq -2 trimmed_R2.fastq -S aligned.sam')\n\n# Convert SAM to BAM using SAMtools\nos.system('samtools view -Sb aligned.sam > aligned.bam')\n\n# Generate coverage data using BEDtools\nos.system('bedtools genomecov -ibam aligned.bam -bg > coverage.bedgraph')\n\nprint('Data preprocessing complete.')"}, {"cell_type": "markdown", "metadata": {}, "source": "Next, we define a simple CNN-LSTM model using TensorFlow/Keras to demonstrate the model architecture described in the paper. Replace placeholder data with actual processed tensors."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import tensorflow as tf\nfrom tensorflow.keras.models import Sequential\nfrom tensorflow.keras.layers import Conv1D, LSTM, Dense, Dropout\n\n# Define the CNN-LSTM model\nmodel = Sequential([\n    Conv1D(filters=32, kernel_size=3, activation='relu', input_shape=(150, 6)),\n    Dropout(0.2),\n    LSTM(50, return_sequences=False),\n    Dense(1, activation='sigmoid')\n])\n\nmodel.compile(optimizer='adam', loss='binary_crossentropy', metrics=['accuracy'])\n\n# Placeholder for training step: model.fit(X_train, y_train, epochs=10, batch_size=32, validation_split=0.2)\n\nprint('CNN-LSTM model defined and ready for training.')"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20raw%20RNA-seq%20datasets%2C%20preprocesses%20them%20using%20Fastp%20and%20HISAT2%2C%20and%20prepares%20input%20tensors%20for%20a%20CNN-LSTM%20model%20to%20replicate%20operon%20detection%20as%20presented%20in%20the%20paper.%0A%0AIncorporate%20hyperparameter%20tuning%2C%20error%20checking%2C%20and%20use%20real%20dataset%20paths%20for%20enhanced%20reproducibility.%0A%0AOpDetect%20convolutional%20recurrent%20neural%20network%20operon%20detection%20RNA-seq%0A%0AThis%20notebook%20details%20the%20data%20ingestion%2C%20preprocessing%20of%20RNA-seq%20files%20with%20Fastp%2C%20alignment%20using%20HISAT2%2C%20and%20BAM%20to%20BED%20conversion%20for%20feature%20extraction.%0A%0Aimport%20subprocess%0Aimport%20os%0A%0A%23%20Download%20dataset%20metadata%0Asubprocess.run%28%5B%27wget%27%2C%20%27https%3A%2F%2Fgithub.com%2FBioinformaticsLabAtMUN%2FOpDetect%2Fraw%2Fmain%2Fdataset_info.csv%27%5D%29%0A%0A%23%20Preprocess%20RNA-seq%20data%20using%20Fastp%0Aos.system%28%27fastp%20-i%20sample_R1.fastq%20-I%20sample_R2.fastq%20-o%20trimmed_R1.fastq%20-O%20trimmed_R2.fastq%27%29%0A%0A%23%20Align%20reads%20using%20HISAT2%0Aos.system%28%27hisat2%20-x%20genome_index%20-1%20trimmed_R1.fastq%20-2%20trimmed_R2.fastq%20-S%20aligned.sam%27%29%0A%0A%23%20Convert%20SAM%20to%20BAM%20using%20SAMtools%0Aos.system%28%27samtools%20view%20-Sb%20aligned.sam%20%3E%20aligned.bam%27%29%0A%0A%23%20Generate%20coverage%20data%20using%20BEDtools%0Aos.system%28%27bedtools%20genomecov%20-ibam%20aligned.bam%20-bg%20%3E%20coverage.bedgraph%27%29%0A%0Aprint%28%27Data%20preprocessing%20complete.%27%29%0A%0ANext%2C%20we%20define%20a%20simple%20CNN-LSTM%20model%20using%20TensorFlow%2FKeras%20to%20demonstrate%20the%20model%20architecture%20described%20in%20the%20paper.%20Replace%20placeholder%20data%20with%20actual%20processed%20tensors.%0A%0Aimport%20tensorflow%20as%20tf%0Afrom%20tensorflow.keras.models%20import%20Sequential%0Afrom%20tensorflow.keras.layers%20import%20Conv1D%2C%20LSTM%2C%20Dense%2C%20Dropout%0A%0A%23%20Define%20the%20CNN-LSTM%20model%0Amodel%20%3D%20Sequential%28%5B%0A%20%20%20%20Conv1D%28filters%3D32%2C%20kernel_size%3D3%2C%20activation%3D%27relu%27%2C%20input_shape%3D%28150%2C%206%29%29%2C%0A%20%20%20%20Dropout%280.2%29%2C%0A%20%20%20%20LSTM%2850%2C%20return_sequences%3DFalse%29%2C%0A%20%20%20%20Dense%281%2C%20activation%3D%27sigmoid%27%29%0A%5D%29%0A%0Amodel.compile%28optimizer%3D%27adam%27%2C%20loss%3D%27binary_crossentropy%27%2C%20metrics%3D%5B%27accuracy%27%5D%29%0A%0A%23%20Placeholder%20for%20training%20step%3A%20model.fit%28X_train%2C%20y_train%2C%20epochs%3D10%2C%20batch_size%3D32%2C%20validation_split%3D0.2%29%0A%0Aprint%28%27CNN-LSTM%20model%20defined%20and%20ready%20for%20training.%27%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20OpDetect%3A%20A%20convolutional%20and%20recurrent%20neural%20network%20classifier%20for%20precise%20and%20sensitive%20operon%20detection%20from%20RNA-seq%20data)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads raw RNA-seq datasets, preprocesses them using Fastp and HISAT2, and prepares input tensors for a CNN-LSTM model to replicate operon detection as presented in the paper.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-04-03", "tags": ["OpDetect RNA-seq classifier", "biology", "bioinformatics"]}}