{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "Loading and quality filtering the PDAC single-cell RNA-seq dataset."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nadata = sc.read_10x_mtx('path_to_data/', var_names='gene_symbols', cache=True)\nsc.pp.filter_cells(adata, min_genes=200)\nsc.pp.filter_genes(adata, min_cells=3)\nadata.raw = adata.copy()\nprint(adata)"}, {"cell_type": "markdown", "metadata": {}, "source": "Normalizing, performing PCA, and clustering cells to identify distinct cell populations."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "sc.pp.normalize_total(adata, target_sum=1e4)\nsc.pp.log1p(adata)\nsc.tl.pca(adata, svd_solver='arpack')\nsc.pp.neighbors(adata, n_neighbors=10, n_pcs=40)\nsc.tl.leiden(adata, resolution=0.6)\nsc.tl.umap(adata)\nsc.pl.umap(adata, color=['leiden'], save='_clusters.png')"}, {"cell_type": "markdown", "metadata": {}, "source": "Extracting and analyzing the prognostic 14-gene signature from the dataset."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nprognostic_genes = ['RPS10', 'MIF', 'MT-ATP6', 'CSTB', 'IFI30', 'NPC2', 'BTG1', 'CTSD', 'FCGR2A', 'SEC61G', 'IER3', 'HSPB1', 'HMOX1', 'ZFP36L1']\ngene_exp = adata[:, prognostic_genes].to_df()\nprint(gene_exp.head())\n# Further integration with clinical metadata could enable survival analysis using Cox regression."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20processes%20and%20analyzes%20PDAC%20scRNA-seq%20data%20to%20validate%20the%20prognostic%20gene%20signature%20and%20explore%20therapeutic%20targets.%0A%0AIntegrate%20clinical%20metadata%20and%20run%20a%20multivariate%20Cox%20regression%20to%20validate%20the%20prognostic%20efficacy%20of%20the%20gene%20signature.%0A%0ASingle-cell%20transcriptomics%20PDAC%20tumor%20microenvironment%20prognostic%20signature%0A%0ALoading%20and%20quality%20filtering%20the%20PDAC%20single-cell%20RNA-seq%20dataset.%0A%0Aimport%20scanpy%20as%20sc%0Aadata%20%3D%20sc.read_10x_mtx%28%27path_to_data%2F%27%2C%20var_names%3D%27gene_symbols%27%2C%20cache%3DTrue%29%0Asc.pp.filter_cells%28adata%2C%20min_genes%3D200%29%0Asc.pp.filter_genes%28adata%2C%20min_cells%3D3%29%0Aadata.raw%20%3D%20adata.copy%28%29%0Aprint%28adata%29%0A%0ANormalizing%2C%20performing%20PCA%2C%20and%20clustering%20cells%20to%20identify%20distinct%20cell%20populations.%0A%0Asc.pp.normalize_total%28adata%2C%20target_sum%3D1e4%29%0Asc.pp.log1p%28adata%29%0Asc.tl.pca%28adata%2C%20svd_solver%3D%27arpack%27%29%0Asc.pp.neighbors%28adata%2C%20n_neighbors%3D10%2C%20n_pcs%3D40%29%0Asc.tl.leiden%28adata%2C%20resolution%3D0.6%29%0Asc.tl.umap%28adata%29%0Asc.pl.umap%28adata%2C%20color%3D%5B%27leiden%27%5D%2C%20save%3D%27_clusters.png%27%29%0A%0AExtracting%20and%20analyzing%20the%20prognostic%2014-gene%20signature%20from%20the%20dataset.%0A%0Aimport%20pandas%20as%20pd%0Aprognostic_genes%20%3D%20%5B%27RPS10%27%2C%20%27MIF%27%2C%20%27MT-ATP6%27%2C%20%27CSTB%27%2C%20%27IFI30%27%2C%20%27NPC2%27%2C%20%27BTG1%27%2C%20%27CTSD%27%2C%20%27FCGR2A%27%2C%20%27SEC61G%27%2C%20%27IER3%27%2C%20%27HSPB1%27%2C%20%27HMOX1%27%2C%20%27ZFP36L1%27%5D%0Agene_exp%20%3D%20adata%5B%3A%2C%20prognostic_genes%5D.to_df%28%29%0Aprint%28gene_exp.head%28%29%29%0A%23%20Further%20integration%20with%20clinical%20metadata%20could%20enable%20survival%20analysis%20using%20Cox%20regression.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Single-cell%20transcriptomics%20analysis%20reveals%20dynamic%20changes%20and%20prognostic%20signature%20in%20tumor%20microenvironment%20of%20PDAC.)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code processes and analyzes PDAC scRNA-seq data to validate the prognostic gene signature and explore therapeutic targets.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-08", "tags": ["single-cell transcriptomics PDAC", "biology", "bioinformatics"]}}