{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "We begin by loading the scRNAseq dataset associated with the SST+ neuron atlas. This dataset includes cells from various developmental stages which are then preprocessed, normalized, and clustered to reveal distinct neuronal subtypes."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nadata = sc.read_h5ad('Dev-SST_v1.h5ad')\nsc.pp.filter_cells(adata, min_genes=200)\nsc.pp.filter_genes(adata, min_cells=3)\nsc.pp.normalize_total(adata, target_sum=1e4)\nsc.pp.log1p(adata)\nsc.tl.pca(adata, svd_solver='arpack')\nsc.pp.neighbors(adata, n_neighbors=10, n_pcs=40)\nsc.tl.umap(adata)\nsc.tl.leiden(adata, resolution=0.5)\nsc.pl.umap(adata, color=['leiden'])"}, {"cell_type": "markdown", "metadata": {}, "source": "The above code snippet pre-processes the data and computes a UMAP embedding to visualize clusters, which represent the distinct diversification modes of SST+ neurons."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import matplotlib.pyplot as plt\nsc.pl.umap(adata, color=['developmental_stage', 'leiden'], save='_sst_neurons.pdf')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "These visualizations help validate the existence of early-committing Martinotti cells versus the gradual diversification seen in non-Martinotti cells, as well as the transient clusters in long-range projecting neurons."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Further downstream analysis can include trajectory inference\nsc.tl.dpt(adata)\nsc.pl.dpt_groups_pseudotime(adata, color='dpt_pseudotime')"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20and%20processes%20scRNAseq%20datasets%20to%20replicate%20clustering%20analysis%20of%20SST%2B%20neurons%2C%20providing%20quantitative%20support%20for%20diversification%20modes.%0A%0AIntegrate%20more%20robust%20batch%20correction%20and%20incorporate%20alternative%20trajectory%20inference%20methods%20to%20refine%20subpopulation%20dynamics.%0A%0ASingle-cell%20transcriptomic%20atlas%20developing%20inhibitory%20neurons%20diversification%20modes%0A%0AWe%20begin%20by%20loading%20the%20scRNAseq%20dataset%20associated%20with%20the%20SST%2B%20neuron%20atlas.%20This%20dataset%20includes%20cells%20from%20various%20developmental%20stages%20which%20are%20then%20preprocessed%2C%20normalized%2C%20and%20clustered%20to%20reveal%20distinct%20neuronal%20subtypes.%0A%0Aimport%20scanpy%20as%20sc%0Aadata%20%3D%20sc.read_h5ad%28%27Dev-SST_v1.h5ad%27%29%0Asc.pp.filter_cells%28adata%2C%20min_genes%3D200%29%0Asc.pp.filter_genes%28adata%2C%20min_cells%3D3%29%0Asc.pp.normalize_total%28adata%2C%20target_sum%3D1e4%29%0Asc.pp.log1p%28adata%29%0Asc.tl.pca%28adata%2C%20svd_solver%3D%27arpack%27%29%0Asc.pp.neighbors%28adata%2C%20n_neighbors%3D10%2C%20n_pcs%3D40%29%0Asc.tl.umap%28adata%29%0Asc.tl.leiden%28adata%2C%20resolution%3D0.5%29%0Asc.pl.umap%28adata%2C%20color%3D%5B%27leiden%27%5D%29%0A%0AThe%20above%20code%20snippet%20pre-processes%20the%20data%20and%20computes%20a%20UMAP%20embedding%20to%20visualize%20clusters%2C%20which%20represent%20the%20distinct%20diversification%20modes%20of%20SST%2B%20neurons.%0A%0Aimport%20matplotlib.pyplot%20as%20plt%0Asc.pl.umap%28adata%2C%20color%3D%5B%27developmental_stage%27%2C%20%27leiden%27%5D%2C%20save%3D%27_sst_neurons.pdf%27%29%0Aplt.show%28%29%0A%0AThese%20visualizations%20help%20validate%20the%20existence%20of%20early-committing%20Martinotti%20cells%20versus%20the%20gradual%20diversification%20seen%20in%20non-Martinotti%20cells%2C%20as%20well%20as%20the%20transient%20clusters%20in%20long-range%20projecting%20neurons.%0A%0A%23%20Further%20downstream%20analysis%20can%20include%20trajectory%20inference%0Asc.tl.dpt%28adata%29%0Asc.pl.dpt_groups_pseudotime%28adata%2C%20color%3D%27dpt_pseudotime%27%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20A%20single-cell%20transcriptomic%20atlas%20of%20developing%20inhibitory%20neurons%20reveals%20expanding%20and%20contracting%20modes%20of%20diversification)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads and processes scRNAseq datasets to replicate clustering analysis of SST+ neurons, providing quantitative support for diversification modes.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-24", "tags": ["single-cell transcriptomics", "biology", "bioinformatics"]}}