{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "### Data Preparation and Model Implementation\nThis block describes downloading a real spatial omics dataset and applying SIMVI to extract intrinsic and spatial-induced latent variables."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nimport simvi\n\n# Download and load the dataset (assuming proper access to a specific spatial omics dataset)\nadata = sc.datasets.visium_sge()\n\n# Preprocess and prepare the dataset\nsc.pp.filter_cells(adata, min_genes=200)\nsc.pp.normalize_total(adata, target_sum=1e4)\nsc.pp.log1p(adata)\n\n# Apply SIMVI modeling (this is a placeholder for actual SIMVI function calls)\nmodel = simvi.SIMVI(adata)\nmodel.train(epochs=50)\nlatent_intrinsic, latent_spatial = model.extract_latents()\n\n# Evaluate and visualize the latent variables\nsc.pp.neighbors(adata, use_rep=latent_intrinsic)\nsc.tl.umap(adata)\nsc.pl.umap(adata, color=['cluster'])"}, {"cell_type": "markdown", "metadata": {}, "source": "### Analysis and Downstream Interpretation\nThe following steps include differential expression analysis based on the SIMVI-derived latent spaces to link spatial context with gene expression."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import matplotlib.pyplot as plt\n\n# Differential expression analysis between groups defined by spatial clusters\nsc.tl.rank_genes_groups(adata, 'cluster', method='wilcoxon')\nsc.pl.rank_genes_groups(adata, n_genes=20, sharey=False)\n\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "This notebook illustrates a concise, reproducible pipeline to leverage SIMVI in disentangling and understanding spatial omics data, integrating both preprocessing and downstream analytical steps."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20spatial%20omics%20datasets%2C%20applies%20SIMVI%20for%20latent%20variable%20extraction%2C%20and%20evaluates%20gene%20expression%20shifts%20related%20to%20spatial%20context.%0A%0AInclude%20parameter%20tuning%20for%20latent%20dimension%20selection%20and%20integrate%20additional%20visualization%20modules%20for%20spatial%20effect%20mapping.%0A%0ASIMVI%20cellular%20states%20spatial%20omics%20review%0A%0A%23%23%23%20Data%20Preparation%20and%20Model%20Implementation%0AThis%20block%20describes%20downloading%20a%20real%20spatial%20omics%20dataset%20and%20applying%20SIMVI%20to%20extract%20intrinsic%20and%20spatial-induced%20latent%20variables.%0A%0Aimport%20scanpy%20as%20sc%0Aimport%20simvi%0A%0A%23%20Download%20and%20load%20the%20dataset%20%28assuming%20proper%20access%20to%20a%20specific%20spatial%20omics%20dataset%29%0Aadata%20%3D%20sc.datasets.visium_sge%28%29%0A%0A%23%20Preprocess%20and%20prepare%20the%20dataset%0Asc.pp.filter_cells%28adata%2C%20min_genes%3D200%29%0Asc.pp.normalize_total%28adata%2C%20target_sum%3D1e4%29%0Asc.pp.log1p%28adata%29%0A%0A%23%20Apply%20SIMVI%20modeling%20%28this%20is%20a%20placeholder%20for%20actual%20SIMVI%20function%20calls%29%0Amodel%20%3D%20simvi.SIMVI%28adata%29%0Amodel.train%28epochs%3D50%29%0Alatent_intrinsic%2C%20latent_spatial%20%3D%20model.extract_latents%28%29%0A%0A%23%20Evaluate%20and%20visualize%20the%20latent%20variables%0Asc.pp.neighbors%28adata%2C%20use_rep%3Dlatent_intrinsic%29%0Asc.tl.umap%28adata%29%0Asc.pl.umap%28adata%2C%20color%3D%5B%27cluster%27%5D%29%0A%0A%23%23%23%20Analysis%20and%20Downstream%20Interpretation%0AThe%20following%20steps%20include%20differential%20expression%20analysis%20based%20on%20the%20SIMVI-derived%20latent%20spaces%20to%20link%20spatial%20context%20with%20gene%20expression.%0A%0Aimport%20matplotlib.pyplot%20as%20plt%0A%0A%23%20Differential%20expression%20analysis%20between%20groups%20defined%20by%20spatial%20clusters%0Asc.tl.rank_genes_groups%28adata%2C%20%27cluster%27%2C%20method%3D%27wilcoxon%27%29%0Asc.pl.rank_genes_groups%28adata%2C%20n_genes%3D20%2C%20sharey%3DFalse%29%0A%0Aplt.show%28%29%0A%0AThis%20notebook%20illustrates%20a%20concise%2C%20reproducible%20pipeline%20to%20leverage%20SIMVI%20in%20disentangling%20and%20understanding%20spatial%20omics%20data%2C%20integrating%20both%20preprocessing%20and%20downstream%20analytical%20steps.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20SIMVI%20disentangles%20intrinsic%20and%20spatial-induced%20cellular%20states%20in%20spatial%20omics%20data)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads spatial omics datasets, applies SIMVI for latent variable extraction, and evaluates gene expression shifts related to spatial context.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-04-03", "tags": ["SIMVI cellular states", "biology", "bioinformatics"]}}