{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "This notebook segment downloads TCGA RNA-seq datasets related to EAC and BE, processes the data using pandas, and visualizes gene expression patterns with seaborn heatmaps."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport seaborn as sns\nimport matplotlib.pyplot as plt\n\ndata = pd.read_csv('path_to_TCGA_EAC_BE_data.csv')\nheatmap_data = data.pivot('gene', 'sample', 'expression')\nplt.figure(figsize=(10,8))\nsns.heatmap(heatmap_data, cmap='viridis')\nplt.title('Gene Expression Heatmap: BE-to-EAC Transition')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "Next, we generate a PCA plot to observe transcriptomic shifts in BE PDOs with CRISPRi perturbations."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "from sklearn.decomposition import PCA\npca = PCA(n_components=2)\nexpression_data = data.drop(['gene'], axis=1).T\ntransformed = pca.fit_transform(expression_data)\npca_df = pd.DataFrame(transformed, columns=['PC1', 'PC2'])\nplt.figure(figsize=(8,6))\nplt.scatter(pca_df['PC1'], pca_df['PC2'], c='teal')\nplt.xlabel('Principal Component 1')\nplt.ylabel('Principal Component 2')\nplt.title('PCA of BE PDO CRISPRi Transcriptomes')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20and%20visualizes%20gene%20expression%20changes%20in%20PDO%20co-culture%20models%20to%20support%20analysis%20of%20the%20BE-to-EAC%20transition.%0A%0AIntegrate%20additional%20datasets%20from%20single-cell%20RNA-seq%20and%20include%20batch%20correction%20methods%20for%20enhanced%20resolution.%0A%0ABE-to-EAC%20transition%20PDO%20co-culture%20model%20organoid%20study%0A%0AThis%20notebook%20segment%20downloads%20TCGA%20RNA-seq%20datasets%20related%20to%20EAC%20and%20BE%2C%20processes%20the%20data%20using%20pandas%2C%20and%20visualizes%20gene%20expression%20patterns%20with%20seaborn%20heatmaps.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20seaborn%20as%20sns%0Aimport%20matplotlib.pyplot%20as%20plt%0A%0Adata%20%3D%20pd.read_csv%28%27path_to_TCGA_EAC_BE_data.csv%27%29%0Aheatmap_data%20%3D%20data.pivot%28%27gene%27%2C%20%27sample%27%2C%20%27expression%27%29%0Aplt.figure%28figsize%3D%2810%2C8%29%29%0Asns.heatmap%28heatmap_data%2C%20cmap%3D%27viridis%27%29%0Aplt.title%28%27Gene%20Expression%20Heatmap%3A%20BE-to-EAC%20Transition%27%29%0Aplt.show%28%29%0A%0ANext%2C%20we%20generate%20a%20PCA%20plot%20to%20observe%20transcriptomic%20shifts%20in%20BE%20PDOs%20with%20CRISPRi%20perturbations.%0A%0Afrom%20sklearn.decomposition%20import%20PCA%0Apca%20%3D%20PCA%28n_components%3D2%29%0Aexpression_data%20%3D%20data.drop%28%5B%27gene%27%5D%2C%20axis%3D1%29.T%0Atransformed%20%3D%20pca.fit_transform%28expression_data%29%0Apca_df%20%3D%20pd.DataFrame%28transformed%2C%20columns%3D%5B%27PC1%27%2C%20%27PC2%27%5D%29%0Aplt.figure%28figsize%3D%288%2C6%29%29%0Aplt.scatter%28pca_df%5B%27PC1%27%5D%2C%20pca_df%5B%27PC2%27%5D%2C%20c%3D%27teal%27%29%0Aplt.xlabel%28%27Principal%20Component%201%27%29%0Aplt.ylabel%28%27Principal%20Component%202%27%29%0Aplt.title%28%27PCA%20of%20BE%20PDO%20CRISPRi%20Transcriptomes%27%29%0Aplt.show%28%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3AInvestigating%20BE-to-EAC%20Transition%20Using%20a%20PDO%20Co-Culture%20Model%0D%0AI.%20Summary%0D%0AEsophageal%20adenocarcinoma%20%28EAC%29%20is%20a%20highly%20aggressive%20malignancy%20often%20arising%20from%20Barrett%E2%80%99s%20esophagus%20%28BE%29%2C%20a%20premalignant%20condition.%20Understanding%20the%20cellular%20and%20molecular%20mechanisms%20driving%20BE-to-EAC%20progression%20is%20critical%20for%20early%20intervention%20and%20therapeutic%20development.%20This%20project%20proposes%20a%20novel%20organoid-based%20approach%20to%20model%20BE-to-EAC%20transition%20by%20co-culturing%20GFP-labeled%20EAC%20patient-derived%20organoids%20%28PDOs%29%20with%20CRISPR-ready%20dCas9-mCherry-labeled%20BE%20PDOs.%20By%20using%20live%20imaging%20and%20transcriptomics%2C%20we%20aim%20to%20identify%20key%20regulatory%20pathways%20that%20facilitate%20EAC%20invasion%20and%20progression.%0D%0ASeveral%20studies%20have%20demonstrated%20the%20utility%20of%20PDO%20models%20in%20studying%20cancer%20progression%2C%20maintaining%20genetic%20heterogeneity%2C%20and%20allowing%20functional%20genomic%20manipulation%20%28Li%20et%20al.%2C%202018%3B%20Drost%20%26amp%3B%20Clevers%2C%202018%29.%20Furthermore%2C%20CRISPR-based%20approaches%20have%20successfully%20identified%20key%20regulators%20of%20cancer%20metastasis%20and%20tumor%20progression%20in%20organoid%20models%20%28Koppen%20et%20al.%2C%202022%29.%0D%0ASpecific%20Aims%3A%0D%0A1.%09Generate%20and%20characterize%20labeled%20PDOs%3A%20Develop%20GFP-labeled%20EAC%20PDOs%20and%20dCas9-mCherry-labeled%20BE%20PDOs%20for%20live-cell%20tracking%20%0D%0A2.%09Assess%20BE-to-EAC%20invasion%20dynamics%3A%20Co-culture%20PDOs%20at%20various%20BE%3AEAC%20ratios%20%281%3A1%2C%203%3A1%2C%201%3A3%29%20and%20visualize%20invasion%20using%20live-cell%20imaging%20with%20Keyence.%0D%0A3.%09Identify%20key%20drivers%20of%20progression%20using%20CRISPRi%3A%20Introduce%20a%20dCas9-sgRNA%20library%20to%20selectively%20inhibit%20oncogenes%20or%20tumor%20suppressors%20in%20BE%20PDOs%20and%20assess%20its%20impact%20on%20EAC%20invasion%0D%0A4.%09Characterize%20PDO%20histology%20and%20pathology%20over%20time%3A%20Perform%20H%26amp%3BE%20staining%2C%20pathology%20scoring%2C%20and%20growth%20kinetics%20analysis%20%28including%20organoid%20formation%20rate%20and%20average%20PDO%20size%29%20%0D%0AThis%20study%20will%20establish%20an%20advanced%20PDO-based%20platform%20to%20dissect%20the%20molecular%20basis%20of%20EAC%20progression%20and%20identify%20novel%20therapeutic%20targets.%0D%0A%20%0D%0A%0D%0A%0D%0A%0D%0AII.%20Aims%0D%0AAim%201%3A%20Establish%20and%20Characterize%20a%20PDO%20Co-Culture%20System%20to%20Model%20BE-to-EAC%20Transition%0D%0A%E2%80%A2%09Generate%20GFP-labeled%20EAC%20PDOs%20and%20dCas9-mCherry-labeled%20BE%20PDOs%20using%20lentiviral%20transduction%20%0D%0A%E2%80%A2%09Co-culture%20PDOs%20in%20various%20BE%3AEAC%20ratios%20%281%3A1%2C%203%3A1%2C%201%3A3%29%20to%20assess%20how%20BE%20cells%20influence%20EAC%20invasion%20and%20vice%20versa.%0D%0A%E2%80%A2%09Use%20live-cell%20imaging%20with%20Keyence%20to%20monitor%20real-time%20interactions%2C%20invasion%2C%20and%20PDO%20morphological%20changes%0D%0A%E2%80%A2%09Perform%20histology%20and%20pathology%20scoring%20on%20PDOs%20at%20multiple%20passages%20to%20assess%20tumorigenic%20transformation%20%0D%0A%E2%80%A2%09Analyze%20growth%20kinetics%20through%20organoid%20formation%20rate%20and%20average%20organoid%20size%20%0D%0AHypothesis%3A%20The%20BE-to-EAC%20transition%20is%20influenced%20by%20cell-cell%20interactions%20in%20the%20tumor%20microenvironment%2C%20altering%20growth%20patterns%20and%20invasion%20potential.%0D%0AMimicking%20Tumor%20Microenvironment%20Dynamics%0D%0A%E2%80%A2%091%3A1%20Ratio%20%28Equal%20BE%3AEAC%20Cells%29%3A%20This%20ratio%20represents%20the%20early%20stages%20of%20BE-to-EAC%20progression%2C%20where%20both%20normal%20BE%20cells%20and%20transformed%20EAC%20cells%20interact%20in%20an%20equal%20balance.%20This%20can%20help%20study%20how%20the%20presence%20of%20normal%20BE%20cells%20impacts%20the%20malignant%20behavior%20of%20EAC%20cells.%0D%0A%E2%80%A2%093%3A1%20Ratio%20%28More%20BE%20Cells%20than%20EAC%20Cells%29%3A%20This%20ratio%20can%20represent%20an%20early-phase%20tumor%20microenvironment%20where%20the%20majority%20of%20cells%20are%20still%20normal%20BE%20cells%2C%20but%20there%20is%20a%20significant%20presence%20of%20malignant%20EAC%20cells.%20Studying%20this%20ratio%20can%20help%20understand%20how%20normal%20cells%20influence%20the%20initiation%20of%20malignancy%2C%20possibly%20through%20cell%20signaling%2C%20communication%2C%20or%20inflammatory%20responses.%20It%20could%20also%20demonstrate%20how%20normal%20cells%20are%20progressively%20overtaken%20by%20malignant%20ones.%0D%0A%E2%80%A2%091%3A3%20Ratio%20%28More%20EAC%20Cells%20than%20BE%20Cells%29%3A%20This%20ratio%20may%20represent%20a%20later-stage%20scenario%20where%20the%20majority%20of%20the%20environment%20consists%20of%20transformed%20EAC%20cells.%20This%20will%20allow%20to%20investigate%20how%20the%20progression%20of%20EAC%20dominance%20impacts%20the%20remaining%20BE%20cells%2C%20as%20well%20as%20explore%20how%20malignant%20cells%20invade%20and%20outcompete%20normal%20cells.%20It%20also%20helps%20study%20how%20EAC%20cells%20may%20modify%20the%20behavior%20of%20normal%20cells%20as%20they%20transition%20from%20premalignant%20to%20malignant%20states.%0D%0A2.%20Understanding%20Cell-Cell%20Interactions%0D%0A%E2%80%A2%09The%20different%20ratios%20will%20enable%20you%20to%20investigate%20how%20BE%20cells%20influence%20the%20invasive%20properties%20of%20EAC%20cells%20and%20vice%20versa.%20This%20will%20help%20identify%20whether%20normal%20cells%20in%20the%20TME%20can%20act%20as%20either%20suppressors%20or%20promoters%20of%20EAC%20progression.%20For%20example%2C%20BE%20cells%20may%20secrete%20molecules%20or%20release%20cytokines%20that%20either%20suppress%20or%20promote%20invasion%20by%20EAC%20cells.%0D%0A%E2%80%A2%09Moreover%2C%20the%20ratios%20may%20show%20how%20cell%20density%20and%20the%20relative%20proportion%20of%20normal%20versus%20malignant%20cells%20affect%20key%20processes%20such%20as%20epithelial-mesenchymal%20transition%20%28EMT%29%2C%20invasion%2C%20and%20resistance%20to%20apoptosis.%0D%0A3.%20Replicating%20the%20Heterogeneity%20of%20Tumor%20Progression%0D%0A%E2%80%A2%09Using%20multiple%20ratios%20simulates%20the%20heterogeneity%20tumor%20microenvironment%2C%20offering%20insights%20into%20how%20the%20presence%20of%20normal%20cells%20or%20tumor-stroma%20interactions%20modulate%20the%20progression%20of%20cancer.%0D%0A4.%20Monitoring%20Temporal%20Changes%20in%20Cell%20Behavior%0D%0A%E2%80%A2%09Using%20live-cell%20imaging%20with%20Keyence%20allows%20you%20to%20capture%20real-time%20dynamics%20of%20the%20PDOs%2C%20enabling%20the%20monitoring%20of%20invasion%2C%20proliferation%2C%20and%20cell%20movement%20in%20response%20to%20different%20co-culture%20conditions.%0D%0A%E2%80%A2%09The%201-hour%20imaging%20intervals%20over%20a%206-day%20period%20%28or%20incubatorlu%20mikroskop%2024%20hr%20imaging%29%20will%20provide%20a%20detailed%20timeline%20of%20cell%20behavior%20changes%2C%20helping%20you%20track%20early%20signs%20of%20invasion%20and%20alterations%20in%20cell%20morphology%20over%20time%2C%20including%20how%20the%20growth%20patterns%20of%20both%20BE%20and%20EAC%20cells%20evolve.%0D%0A%E2%80%A2%09Invasion%20dynamics%20can%20be%20studied%20by%20assessing%20how%20EAC%20cells%20invade%20normal%20BE%20tissue%20and%20how%20the%20BE%20cells%20themselves%20are%20affected%2C%20which%20is%20critical%20for%20understanding%20how%20BE%20cells%20can%20transition%20to%20EAC%20through%20cell%20interaction.%0D%0A5.%20Real-Time%20Monitoring%20of%20Cellular%20Interactions%0D%0A%E2%80%A2%09The%20live-cell%20imaging%20system%20allows%20to%20visualize%20and%20track%20the%20morphological%20changes%20of%20cells%20and%20interactions%20in%20the%20co-culture%2C%20including%20cell-cell%20contact%2C%20migration%20patterns%2C%20and%20invasion%20into%20the%20surrounding%20matrix.%0D%0A%E2%80%A2%09Tracking%20the%20real-time%20interactions%20between%20BE%20and%20EAC%20cells%20can%20lead%20to%20insights%20on%20how%20cancerous%20cells%20invade%20healthy%20tissue%2C%20and%20which%20cellular%20characteristics%20%28such%20as%20motility%2C%20adhesion%2C%20and%20morphology%29%20may%20be%20modulated%20by%20these%20interactions.%0D%0A6.%20Insight%20into%20Invasion%20and%20Tumor%20Growth%20Mechanisms%0D%0A%E2%80%A2%09The%20co-culture%20with%20different%20BE%3AEAC%20ratios%20can%20help%20reveal%20key%20invasion%20pathways%20that%20may%20be%20triggered%20by%20the%20presence%20of%20normal%20BE%20cells%20in%20the%20TME.%20For%20example%2C%20this%20might%20uncover%20molecular%20signaling%20pathways%2C%20cytokines%2C%20and%20growth%20factors%20that%20are%20involved%20in%20the%20malignant%20transition%20from%20BE%20to%20EAC.%0D%0A%E2%80%A2%09The%20co-culture%20system%2C%20combined%20with%20imaging%2C%20can%20also%20help%20identify%20potential%20therapeutic%20targets%20by%20understanding%20how%20these%20cellular%20interactions%20influence%20EAC%20progression.%20By%20looking%20at%20which%20factors%20are%20upregulated%20in%20each%20ratio%2C%20you%20can%20explore%20whether%20targeting%20these%20factors%20can%20suppress%20invasion%20or%20stop%20progression%20to%20EAC.%0D%0A%0D%0A%0D%0AAim%202%3A%20Identify%20Critical%20Genes%20Driving%20EAC%20Progression%20Using%20CRISPRi%20in%20BE%20PDOs%0D%0A%E2%80%A2%09Generate%20dCas9-mCherry-labeled%20BE%20PDOs%20with%20a%20CRISPRi%20sgRNA%20library%20to%20selectively%20inhibit%20oncogenes%20%28e.g.%2C%20MYC%2C%20KRAS%29%20or%20tumor%20suppressors%20%28e.g.%2C%20TP53%2C%20CDKN2A%29%0D%0A%E2%80%A2%09Perform%20co-culture%20assays%20with%20genetically%20modified%20BE%20PDOs%20and%20monitor%20changes%20in%20EAC%20invasion%20rates.%0D%0A%E2%80%A2%09Use%20RNA-seq%20and%20qPCR%20to%20analyze%20transcriptomic%20shifts%20in%20BE%20PDOs%20with%20specific%20gene%20perturbations.%0D%0AHypothesis%3A%20Silencing%20tumor%20suppressors%20or%20activating%20oncogenes%20in%20BE%20PDOs%20will%20increase%20EAC%20invasion%20and%20alter%20molecular%20pathways%20associated%20with%20progression.%0D%0A%20%0D%0AIII.%20Research%20Design%20%26amp%3B%20Methods%0D%0A1.%20PDO%20Generation%20%26amp%3B%20Labeling%0D%0A%E2%80%A2%09GFP-EAC%20PDOs%3A%20Lentiviral%20transduction%20with%20a%20GFP-expressing%20vector%2C%20followed%20by%20antibiotic%20selection%20or%20FACS%0D%0A%E2%80%A2%09dCas9-mCherry%20BE%20PDOs%3A%20Lentiviral%20transduction%20with%20a%20dCas9-mCherry%20construct%2C%20followed%20by%20antibiotic%20selection%20or%20FACS.%0D%0A2.%20PDO%20Co-Culture%20%26amp%3B%20Invasion%20Assays%0D%0A%E2%80%A2%09Seed%20BE%20and%20EAC%20PDOs%20in%2050%20%C2%B5L%20Matrigel%20domes%20and%20culture%20in%20optimized%20media%20%28Boj%20et%20al.%2C%202015%29.%0D%0A%E2%80%A2%09Co-culture%20BE%3AEAC%20PDOs%20in%20ratios%20of%201%3A1%2C%203%3A1%2C%20and%201%3A3.%0D%0A%E2%80%A2%09Use%20live-cell%20imaging%20with%20Keyence%20to%20monitor%20PDO%20interactions%20and%20invasion%20over%20time%20%281%20hour%2F6-day%20period%29%20%28Liu%20et%20al.%2C%202018%29%0D%0A3.%20Histology%20%26amp%3B%20Growth%20Kinetics%20Analysis%0D%0A%E2%80%A2%09Collect%20PDOs%20at%20multiple%20passages%20%28P3%2C%20P6%2C%20P10%29%20and%20perform%20H%26amp%3BE%20staining%20for%20pathology%20scoring%0D%0A%E2%80%A2%09Measure%20organoid%20formation%20rate%20and%20average%20size%20over%20time%20%0D%0A4.%20CRISPRi%20Screening%0D%0A%E2%80%A2%09Introduce%20sgRNA%20libraries%20targeting%20oncogenes%20and%20tumor%20suppressors%20into%20BE%20PDOs%20%0D%0A%E2%80%A2%09Perform%20co-culture%20assays%20with%20modified%20BE%20PDOs%20and%20monitor%20changes%20in%20EAC%20invasion%20rates.%0D%0A%E2%80%A2%09Assess%20gene%20expression%20changes%20via%20qPCR%20and%20RNA-seq%20%0D%0A5.%20Functional%20Assays%20to%20Characterize%20Neoplastic%20Transformation%0D%0ATo%20comprehensively%20assess%20the%20phenotypic%20changes%20associated%20with%20BE-to-EAC%20transition%2C%20we%20will%20conduct%20a%20series%20of%20functional%20assays%20targeting%20key%20characteristics%20of%20neoplastic%20transformation%2C%20including%20proliferation%2C%20apoptosis%2C%20migration%2C%20and%20basement%20membrane%20integrity.%20These%20assays%20will%20be%20applied%20to%20both%20unmodified%20and%20genetically%20modified%20PDOs%20%28e.g.%2C%20dCas9-mCherry-labeled%20BE%20PDOs%20with%20CRISPRi%20perturbations%29.%0D%0Aa.%20Proliferation%20Assays%0D%0A%E2%80%A2%09Objective%3A%20Measure%20cell%20proliferation%20rates%20to%20assess%20neoplastic%20growth%20potential.%0D%0A%E2%80%A2%09Method%3A%20Perform%20EdU%20incorporation%20assays%20to%20quantify%20actively%20proliferating%20cells%20within%20PDOs.%20Live-cell%20imaging%20%28Keyence%29%20will%20also%20be%20used%20to%20monitor%20proliferation%20over%20time.%0D%0A%E2%80%A2%09Expected%20Outcome%3A%20Increased%20proliferation%20in%20PDOs%20with%20oncogene%20activation%20or%20tumor%20suppressor%20inhibition%2C%20particularly%20in%20CRISPR-modified%20BE%20PDOs.%0D%0Ab.%20Apoptosis%20Assays%0D%0A%E2%80%A2%09Objective%3A%20Determine%20apoptotic%20resistance%2C%20a%20hallmark%20of%20cancerous%20transformation.%0D%0A%E2%80%A2%09Method%3A%20Perform%20TUNEL%20staining%20and%20Caspase-3%2F7%20activity%20assays%20to%20quantify%20apoptotic%20cells%20in%20PDOs%20under%20various%20conditions.%0D%0A%E2%80%A2%09Expected%20Outcome%3A%20Reduced%20apoptosis%20in%20PDOs%20with%20neoplastic%20transformation%2C%20particularly%20those%20with%20Wnt%20pathway%20activation%20%28e.g.%2C%20APC%20knockout%29.%0D%0Ad.%20Basement%20Membrane%20Integrity%20Assays%0D%0A%E2%80%A2%09Objective%3A%20Assess%20the%20structural%20integrity%20of%20the%20basement%20membrane%2C%20which%20is%20often%20compromised%20during%20neoplastic%20progression.%0D%0A%E2%80%A2%09Method%3A%20Perform%20immunofluorescence%20staining%20for%20basement%20membrane%20markers%20%28e.g.%2C%20Laminin%2C%20Collagen%20IV%29%20and%20evaluate%20structural%20disruption%20using%20confocal%20microscopy.%0D%0A%E2%80%A2%09Expected%20Outcome%3A%20Increased%20basement%20membrane%20disruption%20and%20invasion%20in%20EAC%20PDOs%20or%20BE%20PDOs%20with%20relevant%20genetic%20perturbations.%0D%0Ad.%20Basement%20Membrane%20Integrity%20Assays%0D%0A%E2%80%A2%09Objective%3A%20Assess%20the%20structural%20integrity%20of%20the%20basement%20membrane%2C%20which%20is%20often%20compromised%20during%20neoplastic%20progression.%0D%0A%E2%80%A2%09Method%3A%20Perform%20immunofluorescence%20staining%20for%20basement%20membrane%20markers%20%28e.g.%2C%20Laminin%2C%20Collagen%20IV%29%20and%20evaluate%20structural%20disruption%20using%20confocal%20microscopy.%0D%0A%E2%80%A2%09Expected%20Outcome%3A%20Increased%20basement%20membrane%20disruption%20and%20invasion%20in%20EAC%20PDOs%20or%20BE%20PDOs%20with%20relevant%20genetic%20perturbations.%0D%0Ae.%20Cytokine%20Paneling%0D%0A%E2%80%A2%09Conditioned%20media%20collection%3A%20At%20various%20time%20points%20%28day%201%2C%203%2C%20and%206%29%2C%20collect%20the%20conditioned%20media%20from%20the%20co-cultured%20PDOs%20for%20cytokine%20analysis.%20This%20will%20allow%20the%20examination%20of%20secreted%20factors%20and%20the%20influence%20of%20BE%20cells%20on%20the%20tumor%20microenvironment%20%28TME%29.%0D%0A%E2%80%A2%09Cytokine%20profiling%20using%20multiplex%20assays%3A%20Employ%20Luminex%20multiplex%20assays%20to%20assess%20cytokine%20levels%20%28e.g.%2C%20IL-6%2C%20TNF-%CE%B1%2C%20VEGF%2C%20TGF-%CE%B2%29%20in%20the%20collected%20media.%20These%20factors%20will%20be%20correlated%20with%20EAC%20PDO%20behavior%2C%20invasion%2C%20and%20growth.%0D%0A%E2%80%A2%09Neutralization%20and%20stimulation%20assays%3A%20To%20functionally%20validate%20cytokines%20of%20interest%2C%20neutralizing%20antibodies%20or%20recombinant%20cytokines%20will%20be%20introduced%20into%20the%20co-culture%20systems%20to%20determine%20their%20role%20in%20EAC%20invasion%20and%20tumorigenesis.%20Neutralizing%20specific%20cytokines%20by%20blocking%20peptide%20will%20help%20establish%20their%20importance%20in%20EAC%20progression.%0D%0Ae.%20Organoid%20Viability%20and%20Growth%20Assays%0D%0A%E2%80%A2%09Objective%3A%20Assess%20overall%20viability%20and%20growth%20kinetics%20of%20PDOs%20under%20different%20experimental%20conditions.%0D%0A%E2%80%A2%09Method%3A%20Use%20CellTiter-Glo%203D%20assays%20for%20ATP%20quantification%20as%20a%20readout%20of%20cell%20viability%20and%20metabolic%20activity.%20Measure%20organoid%20size%20and%20formation%20rate%20over%20time.%0D%0A%E2%80%A2%09Expected%20Outcome%3A%20Increased%20viability%20and%20growth%20in%20PDOs%20with%20oncogenic%20mutations%2C%20especially%20under%20co-culture%20conditions%20with%20EAC%20PDOs.%0D%0A%20%0D%0ATimeline%20%26amp%3B%20Milestones%0D%0ATimepoint%09Milestone%0D%0AMonths%201-3%09Generate%20%26amp%3B%20validate%20labeled%20PDOs%2C%20establish%20co-culture%20conditions%2C%20perform%20initial%20histology%20images%20and%20scoring.%0D%0AMonths%204-6%09Conduct%20co-culture%20invasion%20assays%2C%20live-cell%20imaging%2C%20and%20initiate%20functional%20assays%20%28proliferation%2C%20apoptosis%2C%20migration%2C%20basement%20membrane%20integrity%2C%20viability%29.%0D%0AMonths%207-9%09Perform%20CRISPRi%20perturbation%2C%20transcriptomic%20profiling%2C%20and%20complete%20comprehensive%20functional%20assays.%20Analyze%20collected%20data.%0D%0AMonths%2010-12%09Finalize%20data%20analysis%2C%20validate%20findings%2C%20and%20prepare%20manuscript%20for%20publication.%0D%0A%20%0D%0A%0D%0A%20perform%20proofreading%20and%20control%20references%20if%20necessary%20add%20new%20reference)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads and visualizes gene expression changes in PDO co-culture models to support analysis of the BE-to-EAC transition.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-31", "tags": ["BE-to-EAC transition mechanisms", "biology", "bioinformatics"]}}