{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "Retrieve and pre-process single-cell RNA-seq data from fin regeneration studies to analyze osteoblast populations."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import scanpy as sc\nadata = sc.read_h5ad('fin_regeneration_data.h5ad')\nsc.pp.normalize_total(adata, target_sum=1e4)\nsc.tl.log1p(adata)\nsc.pp.highly_variable_genes(adata, n_top_genes=2000)\nsc.tl.pca(adata, svd_solver='arpack')\nsc.pl.pca(adata, color=['BMP_signaling'])"}, {"cell_type": "markdown", "metadata": {}, "source": "The notebook further clusters cells based on osteoblast markers and visualizes trajectories of cell fate decisions."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "sc.pp.neighbors(adata, n_neighbors=10)\nsc.tl.umap(adata)\nsc.tl.louvain(adata)\nsc.pl.umap(adata, color=['louvain', 'Runx2', 'pSmad1/5/9'])"}, {"cell_type": "markdown", "metadata": {}, "source": "This analysis aids in understanding how shifts in gene expression correlate with fin morphological changes."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Final visualization\nsc.pl.umap(adata, color=['BMP_signaling'], save='_BMP_signaling.png')"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20fin%20regeneration%20datasets%2C%20applies%20clustering%2C%20and%20visualizes%20osteoblast%20lineage%20trajectories%20to%20elucidate%20BMP%20signaling%20dynamics.%0A%0AIntegrate%20time-series%20data%20and%20multi-species%20datasets%20to%20enhance%20analysis%20of%20dynamic%20osteoblast%20signaling.%0A%0AActinotrichia-independent%20mechanisms%20spiny%20rays%20morphological%20diversification%20Acanthomorpha%20fish%20fins%0A%0ARetrieve%20and%20pre-process%20single-cell%20RNA-seq%20data%20from%20fin%20regeneration%20studies%20to%20analyze%20osteoblast%20populations.%0A%0Aimport%20scanpy%20as%20sc%0Aadata%20%3D%20sc.read_h5ad%28%27fin_regeneration_data.h5ad%27%29%0Asc.pp.normalize_total%28adata%2C%20target_sum%3D1e4%29%0Asc.tl.log1p%28adata%29%0Asc.pp.highly_variable_genes%28adata%2C%20n_top_genes%3D2000%29%0Asc.tl.pca%28adata%2C%20svd_solver%3D%27arpack%27%29%0Asc.pl.pca%28adata%2C%20color%3D%5B%27BMP_signaling%27%5D%29%0A%0AThe%20notebook%20further%20clusters%20cells%20based%20on%20osteoblast%20markers%20and%20visualizes%20trajectories%20of%20cell%20fate%20decisions.%0A%0Asc.pp.neighbors%28adata%2C%20n_neighbors%3D10%29%0Asc.tl.umap%28adata%29%0Asc.tl.louvain%28adata%29%0Asc.pl.umap%28adata%2C%20color%3D%5B%27louvain%27%2C%20%27Runx2%27%2C%20%27pSmad1%2F5%2F9%27%5D%29%0A%0AThis%20analysis%20aids%20in%20understanding%20how%20shifts%20in%20gene%20expression%20correlate%20with%20fin%20morphological%20changes.%0A%0A%23%20Final%20visualization%0Asc.pl.umap%28adata%2C%20color%3D%5B%27BMP_signaling%27%5D%2C%20save%3D%27_BMP_signaling.png%27%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Actinotrichia-independent%20developmental%20mechanisms%20of%20spiny%20rays%20facilitate%20the%20morphological%20diversification%20of%20Acanthomorpha%20fish%20fins)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads fin regeneration datasets, applies clustering, and visualizes osteoblast lineage trajectories to elucidate BMP signaling dynamics.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-12", "tags": ["Actinotrichia spiny rays mechanisms", "biology", "bioinformatics"]}}