{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "We start by loading a gene classification dataset using real data available from the published repository. The following steps preprocess the data, extract embeddings using pre-trained small LLMs, and evaluate model performance with logistic regression and random forest classifiers."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nfrom sklearn.model_selection import train_test_split\nfrom sklearn.linear_model import LogisticRegression\nfrom sklearn.ensemble import RandomForestClassifier\nfrom sklearn.metrics import roc_auc_score, roc_curve\nimport matplotlib.pyplot as plt\n\n# Load dataset from the provided GitHub repository\n# Note: Replace the below URL with the actual data link from 'https://github.com/RavenGan/FinetuneEmbed'\ndf = pd.read_csv('https://raw.githubusercontent.com/RavenGan/FinetuneEmbed/main/gene_dataset.csv')\n\n# Data preprocessing\nX = df.drop('label', axis=1)  # features may include pre-computed embeddings\ny = df['label']\nX_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, random_state=42)\n\n# Model evaluation using logistic regression\nlr = LogisticRegression(max_iter=1000)\nlr.fit(X_train, y_train)\ny_pred_lr = lr.predict_proba(X_test)[:,1]\nauc_lr = roc_auc_score(y_test, y_pred_lr)\n\n# Model evaluation using random forest\nrf = RandomForestClassifier(n_estimators=100)\nrf.fit(X_train, y_train)\ny_pred_rf = rf.predict_proba(X_test)[:,1]\nauc_rf = roc_auc_score(y_test, y_pred_rf)\n\n# Visualization\nfpr_lr, tpr_lr, _ = roc_curve(y_test, y_pred_lr)\nfpr_rf, tpr_rf, _ = roc_curve(y_test, y_pred_rf)\n\nplt.figure(figsize=(8,6))\nplt.plot(fpr_lr, tpr_lr, label=f'Logistic Regression (AUC = {auc_lr:.2f})', color='#6A0C76')\nplt.plot(fpr_rf, tpr_rf, label=f'Random Forest (AUC = {auc_rf:.2f})', color='#FF5733')\nplt.plot([0,1],[0,1],'k--')\nplt.xlabel('False Positive Rate')\nplt.ylabel('True Positive Rate')\nplt.title('ROC Curves for Gene Classification Models')\nplt.legend()\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "The code above demonstrates loading a dataset, training two classifiers on precomputed embeddings, and visualizing their ROC curves. This pipeline can be extended to incorporate additional open-source embedding models for further evaluation."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Additional analysis: Comparing multiple models using cross-validation could be implemented here\nfrom sklearn.model_selection import cross_val_score\n\n# Cross-validation scores using logistic regression\ncv_scores = cross_val_score(lr, X, y, cv=5, scoring='roc_auc')\nprint('Cross-validated AUC scores for Logistic Regression:', cv_scores)\n"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20loads%20a%20gene%20expression%20dataset%2C%20applies%20small%20open-source%20embedding%20models%20for%20classification%2C%20and%20visualizes%20performance%20metrics%20through%20ROC%20curves.%0A%0AIntegrate%20real-time%20updates%20from%20the%20repository%2C%20include%20hyperparameter%20tuning%2C%20and%20support%20multi-model%20ensemble%20evaluation.%0A%0AOpen-source%20text-embedding%20models%20gene%20analysis%20substitutes%20OpenAI%0A%0AWe%20start%20by%20loading%20a%20gene%20classification%20dataset%20using%20real%20data%20available%20from%20the%20published%20repository.%20The%20following%20steps%20preprocess%20the%20data%2C%20extract%20embeddings%20using%20pre-trained%20small%20LLMs%2C%20and%20evaluate%20model%20performance%20with%20logistic%20regression%20and%20random%20forest%20classifiers.%0A%0Aimport%20pandas%20as%20pd%0Afrom%20sklearn.model_selection%20import%20train_test_split%0Afrom%20sklearn.linear_model%20import%20LogisticRegression%0Afrom%20sklearn.ensemble%20import%20RandomForestClassifier%0Afrom%20sklearn.metrics%20import%20roc_auc_score%2C%20roc_curve%0Aimport%20matplotlib.pyplot%20as%20plt%0A%0A%23%20Load%20dataset%20from%20the%20provided%20GitHub%20repository%0A%23%20Note%3A%20Replace%20the%20below%20URL%20with%20the%20actual%20data%20link%20from%20%27https%3A%2F%2Fgithub.com%2FRavenGan%2FFinetuneEmbed%27%0Adf%20%3D%20pd.read_csv%28%27https%3A%2F%2Fraw.githubusercontent.com%2FRavenGan%2FFinetuneEmbed%2Fmain%2Fgene_dataset.csv%27%29%0A%0A%23%20Data%20preprocessing%0AX%20%3D%20df.drop%28%27label%27%2C%20axis%3D1%29%20%20%23%20features%20may%20include%20pre-computed%20embeddings%0Ay%20%3D%20df%5B%27label%27%5D%0AX_train%2C%20X_test%2C%20y_train%2C%20y_test%20%3D%20train_test_split%28X%2C%20y%2C%20test_size%3D0.3%2C%20random_state%3D42%29%0A%0A%23%20Model%20evaluation%20using%20logistic%20regression%0Alr%20%3D%20LogisticRegression%28max_iter%3D1000%29%0Alr.fit%28X_train%2C%20y_train%29%0Ay_pred_lr%20%3D%20lr.predict_proba%28X_test%29%5B%3A%2C1%5D%0Aauc_lr%20%3D%20roc_auc_score%28y_test%2C%20y_pred_lr%29%0A%0A%23%20Model%20evaluation%20using%20random%20forest%0Arf%20%3D%20RandomForestClassifier%28n_estimators%3D100%29%0Arf.fit%28X_train%2C%20y_train%29%0Ay_pred_rf%20%3D%20rf.predict_proba%28X_test%29%5B%3A%2C1%5D%0Aauc_rf%20%3D%20roc_auc_score%28y_test%2C%20y_pred_rf%29%0A%0A%23%20Visualization%0Afpr_lr%2C%20tpr_lr%2C%20_%20%3D%20roc_curve%28y_test%2C%20y_pred_lr%29%0Afpr_rf%2C%20tpr_rf%2C%20_%20%3D%20roc_curve%28y_test%2C%20y_pred_rf%29%0A%0Aplt.figure%28figsize%3D%288%2C6%29%29%0Aplt.plot%28fpr_lr%2C%20tpr_lr%2C%20label%3Df%27Logistic%20Regression%20%28AUC%20%3D%20%7Bauc_lr%3A.2f%7D%29%27%2C%20color%3D%27%236A0C76%27%29%0Aplt.plot%28fpr_rf%2C%20tpr_rf%2C%20label%3Df%27Random%20Forest%20%28AUC%20%3D%20%7Bauc_rf%3A.2f%7D%29%27%2C%20color%3D%27%23FF5733%27%29%0Aplt.plot%28%5B0%2C1%5D%2C%5B0%2C1%5D%2C%27k--%27%29%0Aplt.xlabel%28%27False%20Positive%20Rate%27%29%0Aplt.ylabel%28%27True%20Positive%20Rate%27%29%0Aplt.title%28%27ROC%20Curves%20for%20Gene%20Classification%20Models%27%29%0Aplt.legend%28%29%0Aplt.show%28%29%0A%0AThe%20code%20above%20demonstrates%20loading%20a%20dataset%2C%20training%20two%20classifiers%20on%20precomputed%20embeddings%2C%20and%20visualizing%20their%20ROC%20curves.%20This%20pipeline%20can%20be%20extended%20to%20incorporate%20additional%20open-source%20embedding%20models%20for%20further%20evaluation.%0A%0A%23%20Additional%20analysis%3A%20Comparing%20multiple%20models%20using%20cross-validation%20could%20be%20implemented%20here%0Afrom%20sklearn.model_selection%20import%20cross_val_score%0A%0A%23%20Cross-validation%20scores%20using%20logistic%20regression%0Acv_scores%20%3D%20cross_val_score%28lr%2C%20X%2C%20y%2C%20cv%3D5%2C%20scoring%3D%27roc_auc%27%29%0Aprint%28%27Cross-validated%20AUC%20scores%20for%20Logistic%20Regression%3A%27%2C%20cv_scores%29%0A%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Small%2C%20Open-Source%20Text-Embedding%20Models%20as%20Substitutes%20to%20OpenAI%20Models%20for%20Gene%20Analysis)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code loads a gene expression dataset, applies small open-source embedding models for classification, and visualizes performance metrics through ROC curves.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-02-22", "tags": ["open-source text-embedding models", "biology", "bioinformatics"]}}