{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "### Step 1: Import Required Libraries\nThis step involves importing necessary libraries for data manipulation and analysis."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\nfrom Bio import SeqIO\nfrom Bio.SeqUtils import GC"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 2: Load Genome Assembly Data\nLoad the genome assembly data and contaminant databases for analysis."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Load genome assembly\nassembly_data = SeqIO.to_dict(SeqIO.parse('genome_assembly.fasta', 'fasta'))\n\n# Load contaminant database\ncontaminants = pd.read_csv('contaminant_sequences.csv')"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 3: Identify Contaminants\nUsing taxonomic annotation and GC content to identify potential contaminants."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "def identify_contaminants(assembly_data, contaminants):\n    identified = []\n    for seq_id, seq_record in assembly_data.items():\n        gc_content = GC(seq_record.seq)\n        if gc_content < 30 or gc_content > 70:\n            identified.append(seq_id)\n    return identified\n\ncontaminant_ids = identify_contaminants(assembly_data, contaminants)"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 4: Remove Contaminants from Assembly\nRemove identified contaminants from the genome assembly."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "cleaned_assembly = {seq_id: seq_record for seq_id, seq_record in assembly_data.items() if seq_id not in contaminant_ids}\n\n# Save cleaned assembly\nwith open('cleaned_genome_assembly.fasta', 'w') as output:\n    SeqIO.write(cleaned_assembly.values(), output, 'fasta')"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 5: Summary of Contaminant Removal\nProvide a summary of the contaminant removal process."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "print(f'Removed {len(contaminant_ids)} contaminants from the genome assembly.')"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20implements%20a%20pipeline%20for%20contaminant%20removal%20in%20genome%20assemblies%20using%20taxonomic%20annotation%20and%20read%20coverage%20analysis.%0A%0AConsider%20integrating%20additional%20bioinformatics%20tools%20for%20more%20comprehensive%20contaminant%20detection%20and%20removal%2C%20such%20as%20Kraken%20or%20BlobTools.%0A%0AContaminant%20removal%20techniques%20in%20genome%20assembly%0A%0A%23%23%23%20Step%201%3A%20Import%20Required%20Libraries%0AThis%20step%20involves%20importing%20necessary%20libraries%20for%20data%20manipulation%20and%20analysis.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0Afrom%20Bio%20import%20SeqIO%0Afrom%20Bio.SeqUtils%20import%20GC%0A%0A%23%23%23%20Step%202%3A%20Load%20Genome%20Assembly%20Data%0ALoad%20the%20genome%20assembly%20data%20and%20contaminant%20databases%20for%20analysis.%0A%0A%23%20Load%20genome%20assembly%0Aassembly_data%20%3D%20SeqIO.to_dict%28SeqIO.parse%28%27genome_assembly.fasta%27%2C%20%27fasta%27%29%29%0A%0A%23%20Load%20contaminant%20database%0Acontaminants%20%3D%20pd.read_csv%28%27contaminant_sequences.csv%27%29%0A%0A%23%23%23%20Step%203%3A%20Identify%20Contaminants%0AUsing%20taxonomic%20annotation%20and%20GC%20content%20to%20identify%20potential%20contaminants.%0A%0Adef%20identify_contaminants%28assembly_data%2C%20contaminants%29%3A%0A%20%20%20%20identified%20%3D%20%5B%5D%0A%20%20%20%20for%20seq_id%2C%20seq_record%20in%20assembly_data.items%28%29%3A%0A%20%20%20%20%20%20%20%20gc_content%20%3D%20GC%28seq_record.seq%29%0A%20%20%20%20%20%20%20%20if%20gc_content%20%3C%2030%20or%20gc_content%20%3E%2070%3A%0A%20%20%20%20%20%20%20%20%20%20%20%20identified.append%28seq_id%29%0A%20%20%20%20return%20identified%0A%0Acontaminant_ids%20%3D%20identify_contaminants%28assembly_data%2C%20contaminants%29%0A%0A%23%23%23%20Step%204%3A%20Remove%20Contaminants%20from%20Assembly%0ARemove%20identified%20contaminants%20from%20the%20genome%20assembly.%0A%0Acleaned_assembly%20%3D%20%7Bseq_id%3A%20seq_record%20for%20seq_id%2C%20seq_record%20in%20assembly_data.items%28%29%20if%20seq_id%20not%20in%20contaminant_ids%7D%0A%0A%23%20Save%20cleaned%20assembly%0Awith%20open%28%27cleaned_genome_assembly.fasta%27%2C%20%27w%27%29%20as%20output%3A%0A%20%20%20%20SeqIO.write%28cleaned_assembly.values%28%29%2C%20output%2C%20%27fasta%27%29%0A%0A%23%23%23%20Step%205%3A%20Summary%20of%20Contaminant%20Removal%0AProvide%20a%20summary%20of%20the%20contaminant%20removal%20process.%0A%0Aprint%28f%27Removed%20%7Blen%28contaminant_ids%29%7D%20contaminants%20from%20the%20genome%20assembly.%27%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=contaminant%20removal%20in%20genome%20assembly)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code implements a pipeline for contaminant removal in genome assemblies using taxonomic annotation and read coverage analysis.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-27", "tags": ["genome assembly contaminant removal", "biology", "bioinformatics"]}}