{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "### Step 1: Import Libraries\nImport necessary libraries for data analysis."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\nimport seaborn as sns\nimport matplotlib.pyplot as plt\nfrom scipy import stats"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 2: Load Data\nLoad RNA-seq data from GEO or other repositories."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Load the dataset\nrna_seq_data = pd.read_csv('path_to_rna_seq_data.csv')"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 3: Data Preprocessing\nPreprocess the data to remove low-quality reads and normalize."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "# Normalize the data\nnormalized_data = (rna_seq_data - rna_seq_data.mean()) / rna_seq_data.std()"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 4: Differential Expression Analysis\nPerform differential expression analysis to identify genes affected by ASXL1 variants."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "from statsmodels.stats.multitest import multipletests\n# Example differential expression analysis\nresults = stats.ttest_ind(normalized_data[condition1], normalized_data[condition2])\np_values = results.pvalue\n# Adjust for multiple testing\nadjusted_p_values = multipletests(p_values, method='fdr_bh')[1]"}, {"cell_type": "markdown", "metadata": {}, "source": "### Step 5: Visualization\nVisualize the results using a volcano plot."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "plt.figure(figsize=(10,6))\nsns.scatterplot(x=log2_fold_change, y=-np.log10(adjusted_p_values))\nplt.title('Volcano Plot of Differential Expression')\nplt.xlabel('Log2 Fold Change')\nplt.ylabel('-Log10 Adjusted P-Value')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20analyzes%20RNA-seq%20data%20to%20identify%20differentially%20expressed%20genes%20associated%20with%20ASXL1%20truncating%20variants%20in%20leukemia.%0A%0AInclude%20additional%20datasets%20for%20validation%20and%20expand%20the%20analysis%20to%20include%20pathway%20enrichment.%0A%0ATest%20truncating%20variants%20of%20transcription%20factors%20in%20leukemia%20experiments%0A%0A%23%23%23%20Step%201%3A%20Import%20Libraries%0AImport%20necessary%20libraries%20for%20data%20analysis.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0Aimport%20seaborn%20as%20sns%0Aimport%20matplotlib.pyplot%20as%20plt%0Afrom%20scipy%20import%20stats%0A%0A%23%23%23%20Step%202%3A%20Load%20Data%0ALoad%20RNA-seq%20data%20from%20GEO%20or%20other%20repositories.%0A%0A%23%20Load%20the%20dataset%0Arna_seq_data%20%3D%20pd.read_csv%28%27path_to_rna_seq_data.csv%27%29%0A%0A%23%23%23%20Step%203%3A%20Data%20Preprocessing%0APreprocess%20the%20data%20to%20remove%20low-quality%20reads%20and%20normalize.%0A%0A%23%20Normalize%20the%20data%0Anormalized_data%20%3D%20%28rna_seq_data%20-%20rna_seq_data.mean%28%29%29%20%2F%20rna_seq_data.std%28%29%0A%0A%23%23%23%20Step%204%3A%20Differential%20Expression%20Analysis%0APerform%20differential%20expression%20analysis%20to%20identify%20genes%20affected%20by%20ASXL1%20variants.%0A%0Afrom%20statsmodels.stats.multitest%20import%20multipletests%0A%23%20Example%20differential%20expression%20analysis%0Aresults%20%3D%20stats.ttest_ind%28normalized_data%5Bcondition1%5D%2C%20normalized_data%5Bcondition2%5D%29%0Ap_values%20%3D%20results.pvalue%0A%23%20Adjust%20for%20multiple%20testing%0Aadjusted_p_values%20%3D%20multipletests%28p_values%2C%20method%3D%27fdr_bh%27%29%5B1%5D%0A%0A%23%23%23%20Step%205%3A%20Visualization%0AVisualize%20the%20results%20using%20a%20volcano%20plot.%0A%0Aplt.figure%28figsize%3D%2810%2C6%29%29%0Asns.scatterplot%28x%3Dlog2_fold_change%2C%20y%3D-np.log10%28adjusted_p_values%29%29%0Aplt.title%28%27Volcano%20Plot%20of%20Differential%20Expression%27%29%0Aplt.xlabel%28%27Log2%20Fold%20Change%27%29%0Aplt.ylabel%28%27-Log10%20Adjusted%20P-Value%27%29%0Aplt.show%28%29%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Design%20Experiments%3A%20test%20truncationg%20variants%20in%20transcription%20factor%20in%20leukemia)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code analyzes RNA-seq data to identify differentially expressed genes associated with ASXL1 truncating variants in leukemia.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-03-26", "tags": ["truncating variants leukemia", "biology", "bioinformatics"]}}