{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "### Data Download and Preparation\nThis section downloads proteomic datasets and prepares the data for differential analysis using Python libraries."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import pandas as pd\nimport numpy as np\n# Assume dataset URLs provided\npdsa_data = pd.read_csv('https://example.com/proteomics_PD_MSA.csv')\n# Display first few rows\ndf_head = pdsa_data.head()\nprint(df_head)"}, {"cell_type": "markdown", "metadata": {}, "source": "### Differential Abundance Analysis\nWe perform a t-test for each protein to identify significant differences between PD and MSA samples."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "from scipy.stats import ttest_ind\n\n# Separate datasets for PD and MSA\npd_data = pdsa_data[pdsa_data['disease']=='PD']\nmsa_data = pdsa_data[pdsa_data['disease']=='MSA']\n\n# Initialize list to hold results\nresults = []\n\nfor protein in pdsa_data.columns[2:]:\n    t_stat, p_value = ttest_ind(pd_data[protein], msa_data[protein], nan_policy='omit')\n    results.append({'protein': protein, 't_stat': t_stat, 'p_value': p_value})\n\nresults_df = pd.DataFrame(results)\n# Filter significant proteins\nsignificant = results_df[results_df['p_value'] < 0.05]\nprint(significant)"}, {"cell_type": "markdown", "metadata": {}, "source": "### Visualization of Differential Expression\nWe plot significant proteins showing differential expression using Plotly."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import plotly.express as px\n\nfig = px.bar(significant, x='protein', y='t_stat', color='p_value', title='Differential Protein Expression in PD vs MSA')\nfig.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "This notebook clearly outlines the differential analysis pipeline to interrogate proteomic differences between PD and MSA."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20code%20downloads%20post-mortem%20proteomic%20datasets%20and%20performs%20differential%20abundance%20analysis%20to%20highlight%20interactome%20differences%20between%20PD%20and%20MSA.%0A%0AInclude%20raw%20dataset%20links%20and%20more%20sophisticated%20multi-variate%20models%20to%20account%20for%20batch%20effects%20and%20confounders.%0A%0AProximity%20proteomics%20multiple%20system%20atrophy%20Parkinson%27s%20disease%20features%0A%0A%23%23%23%20Data%20Download%20and%20Preparation%0AThis%20section%20downloads%20proteomic%20datasets%20and%20prepares%20the%20data%20for%20differential%20analysis%20using%20Python%20libraries.%0A%0Aimport%20pandas%20as%20pd%0Aimport%20numpy%20as%20np%0A%23%20Assume%20dataset%20URLs%20provided%0Apdsa_data%20%3D%20pd.read_csv%28%27https%3A%2F%2Fexample.com%2Fproteomics_PD_MSA.csv%27%29%0A%23%20Display%20first%20few%20rows%0Adf_head%20%3D%20pdsa_data.head%28%29%0Aprint%28df_head%29%0A%0A%23%23%23%20Differential%20Abundance%20Analysis%0AWe%20perform%20a%20t-test%20for%20each%20protein%20to%20identify%20significant%20differences%20between%20PD%20and%20MSA%20samples.%0A%0Afrom%20scipy.stats%20import%20ttest_ind%0A%0A%23%20Separate%20datasets%20for%20PD%20and%20MSA%0Apd_data%20%3D%20pdsa_data%5Bpdsa_data%5B%27disease%27%5D%3D%3D%27PD%27%5D%0Amsa_data%20%3D%20pdsa_data%5Bpdsa_data%5B%27disease%27%5D%3D%3D%27MSA%27%5D%0A%0A%23%20Initialize%20list%20to%20hold%20results%0Aresults%20%3D%20%5B%5D%0A%0Afor%20protein%20in%20pdsa_data.columns%5B2%3A%5D%3A%0A%20%20%20%20t_stat%2C%20p_value%20%3D%20ttest_ind%28pd_data%5Bprotein%5D%2C%20msa_data%5Bprotein%5D%2C%20nan_policy%3D%27omit%27%29%0A%20%20%20%20results.append%28%7B%27protein%27%3A%20protein%2C%20%27t_stat%27%3A%20t_stat%2C%20%27p_value%27%3A%20p_value%7D%29%0A%0Aresults_df%20%3D%20pd.DataFrame%28results%29%0A%23%20Filter%20significant%20proteins%0Asignificant%20%3D%20results_df%5Bresults_df%5B%27p_value%27%5D%20%3C%200.05%5D%0Aprint%28significant%29%0A%0A%23%23%23%20Visualization%20of%20Differential%20Expression%0AWe%20plot%20significant%20proteins%20showing%20differential%20expression%20using%20Plotly.%0A%0Aimport%20plotly.express%20as%20px%0A%0Afig%20%3D%20px.bar%28significant%2C%20x%3D%27protein%27%2C%20y%3D%27t_stat%27%2C%20color%3D%27p_value%27%2C%20title%3D%27Differential%20Protein%20Expression%20in%20PD%20vs%20MSA%27%29%0Afig.show%28%29%0A%0AThis%20notebook%20clearly%20outlines%20the%20differential%20analysis%20pipeline%20to%20interrogate%20proteomic%20differences%20between%20PD%20and%20MSA.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Proximity%20proteomics%20reveals%20unique%20and%20shared%20pathological%20features%20between%20multiple%20system%20atrophy%20and%20Parkinson%E2%80%99s%20disease)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This code downloads post-mortem proteomic datasets and performs differential abundance analysis to highlight interactome differences between PD and MSA.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-04-03", "tags": ["multiple system atrophy features", "biology", "bioinformatics"]}}