{"nbformat": 4, "nbformat_minor": 5, "cells": [{"cell_type": "markdown", "metadata": {}, "source": "We begin by importing PPI network data extracted from the published omics analysis dataset of the paper. The goal is to compute centrality measures (Degree, Betweenness) using networkx, and visualize the network to highlight key targets such as AKT1 and FAM81A."}, {"cell_type": "code", "outputs": [], "metadata": {}, "execution_count": null, "source": "import networkx as nx\nimport matplotlib.pyplot as plt\n\n# Assume ppi_edges is a list of tuples (source, target, weight)\n# Example: ppi_edges = [('AKT1','TP53',0.9), ('AKT1','FAM81A',0.85), ('TP53','CASP3',0.8)]\n\n# Load actual dataset here\nppi_edges = []  # replace with real data from paper\n\nG = nx.Graph()\nG.add_weighted_edges_from(ppi_edges)\n\n# Calculate centrality metrics\ndegree_centrality = nx.degree_centrality(G)\nbetweenness_centrality = nx.betweenness_centrality(G)\n\n# Plot network\nplt.figure(figsize=(8,6))\npos = nx.spring_layout(G)\nnx.draw_networkx_nodes(G, pos, node_color='lightblue', node_size=[v*5000 for v in degree_centrality.values()])\nnx.draw_networkx_edges(G, pos, width=1.0, alpha=0.7)\nnx.draw_networkx_labels(G, pos, font_size=10)\nplt.title('PPI Network Highlighting Key Targets')\nplt.axis('off')\nplt.show()"}, {"cell_type": "markdown", "metadata": {}, "source": "This notebook serves to illustrate which proteins are most central in the network and could be driving the observed therapeutic effects. High degree and betweenness suggest potential key regulatory roles, guiding further experimental designs."}, {"cell_type": "markdown", "metadata": {}, "source": "\n\n\n\n***\n### [**Evolve This Code**](https://biologpt.com/?q=Evolve%20Code%3A%20This%20Python%20code%20performs%20network%20topology%20analysis%20on%20protein-protein%20interaction%20data%20derived%20from%20omics%20datasets%20to%20identify%20key%20targets%20in%20gastric%20cancer.%0A%0AInclude%20real%20omics-derived%20PPI%20dataset%20and%20integrate%20interactive%20network%20visualization%20libraries%20like%20vis-network%20for%20enhanced%20user%20interaction.%0A%0AIntegrative%20network%20pharmacology%20transcriptomics%20proteomics%20Shen%20Qing%20Weichang%20Formula%20gastric%20cancer%0A%0AWe%20begin%20by%20importing%20PPI%20network%20data%20extracted%20from%20the%20published%20omics%20analysis%20dataset%20of%20the%20paper.%20The%20goal%20is%20to%20compute%20centrality%20measures%20%28Degree%2C%20Betweenness%29%20using%20networkx%2C%20and%20visualize%20the%20network%20to%20highlight%20key%20targets%20such%20as%20AKT1%20and%20FAM81A.%0A%0Aimport%20networkx%20as%20nx%0Aimport%20matplotlib.pyplot%20as%20plt%0A%0A%23%20Assume%20ppi_edges%20is%20a%20list%20of%20tuples%20%28source%2C%20target%2C%20weight%29%0A%23%20Example%3A%20ppi_edges%20%3D%20%5B%28%27AKT1%27%2C%27TP53%27%2C0.9%29%2C%20%28%27AKT1%27%2C%27FAM81A%27%2C0.85%29%2C%20%28%27TP53%27%2C%27CASP3%27%2C0.8%29%5D%0A%0A%23%20Load%20actual%20dataset%20here%0Appi_edges%20%3D%20%5B%5D%20%20%23%20replace%20with%20real%20data%20from%20paper%0A%0AG%20%3D%20nx.Graph%28%29%0AG.add_weighted_edges_from%28ppi_edges%29%0A%0A%23%20Calculate%20centrality%20metrics%0Adegree_centrality%20%3D%20nx.degree_centrality%28G%29%0Abetweenness_centrality%20%3D%20nx.betweenness_centrality%28G%29%0A%0A%23%20Plot%20network%0Aplt.figure%28figsize%3D%288%2C6%29%29%0Apos%20%3D%20nx.spring_layout%28G%29%0Anx.draw_networkx_nodes%28G%2C%20pos%2C%20node_color%3D%27lightblue%27%2C%20node_size%3D%5Bv%2A5000%20for%20v%20in%20degree_centrality.values%28%29%5D%29%0Anx.draw_networkx_edges%28G%2C%20pos%2C%20width%3D1.0%2C%20alpha%3D0.7%29%0Anx.draw_networkx_labels%28G%2C%20pos%2C%20font_size%3D10%29%0Aplt.title%28%27PPI%20Network%20Highlighting%20Key%20Targets%27%29%0Aplt.axis%28%27off%27%29%0Aplt.show%28%29%0A%0AThis%20notebook%20serves%20to%20illustrate%20which%20proteins%20are%20most%20central%20in%20the%20network%20and%20could%20be%20driving%20the%20observed%20therapeutic%20effects.%20High%20degree%20and%20betweenness%20suggest%20potential%20key%20regulatory%20roles%2C%20guiding%20further%20experimental%20designs.%0A%0A)\n***\n\n### [Created with BioloGPT](https://biologpt.com/?q=Paper%20Review%3A%20Integrative%20network%20pharmacology%2C%20transcriptomics%2C%20and%20proteomics%20reveal%20the%20material%20basis%20and%20mechanism%20of%20the%20Shen%20Qing%20Weichang%20Formula%20against%20gastric%20cancer.)\n[![BioloGPT Logo](https://biologpt.com/static/icons/bioinformatics_wizard.png)](https://biologpt.com/)\n***"}], "metadata": {"title": "This Python code performs network topology analysis on protein-protein interaction data derived from omics datasets to identify key targets in gastric cancer.", "kernelspec": {"display_name": "Python 3", "language": "python", "name": "python3"}, "language_info": {"name": "python", "version": "3.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "file_extension": ".py", "pygments_lexer": "ipython3"}, "author": "BioloGPT", "creation_date": "2025-04-03", "tags": ["network pharmacology gastric cancer", "biology", "bioinformatics"]}}